---
title: "Anuran eye scaling and investment"
author: "Kate Thomas"
date: May 2020
output:
  html_document:
    theme: flatly
    toc: TRUE
    toc_float: TRUE
    code_fold: hide
---

```{r packages, eval = FALSE, include = FALSE}

#Install required packages
install.packages("tidyverse")
install.packages("smatr")
install.packages("plyr")
install.packages("nlme")
install.packages("ape")
install.packages("geiger")
install.packages("picante")
install.packages("caper")
install.packages("phytools")
install.packages("phylotools")
install.packages("plotly")
install.packages("knitr")
install.packages("kableExtra")
install.packages("gridExtra")
install.packages("ggimage")
install.packages("grid")
install.packages("gtable")
install.packages("cowplot")
install.packages("mvmorph")

#install ggtree package from Bioconductor
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("ggtree") 
```

```{r setup, include = FALSE}

# Load package libraries
library(plyr)
library(smatr)
library(picante)
library(nlme)
library(ape)
library(evobiR)
library(geiger)
library(caper)
library(phytools)
library(phylotools)
library(plotly)
library(kableExtra)
library(gridExtra)
library(ggimage)
library(ggtree)
library(grid)
library(gtable)
library(cowplot)
library(knitr)
library(mvMORPH)
library(evomap)
library(tidyverse)

# Markdown settings
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, fig.height = 4.5, fig.width=8)

```

# Data

## Morphological measurements

We recorded 6 morphological measurements from each frog specimen: snout-vent length, wet mass, corneal diameters (left and right), and transverse eye diameters (left and right). Symmetrical measures (e.g. eye diameters) were averaged for each individual prior to further analysis. 

```{r load-data}

# Load raw data for measured frog specimens
frogs_morph <- data.frame(read.csv("../Data/frogs_analysis.csv", header=TRUE, na.strings=c("", "NA", " ","<1"))) #load raw data

# Tidy dataset for analysis
frogs.subset <- frogs_morph %>%
  mutate(rootmass = Mass_g^(1/3)) %>% #adds cube root of mass
  mutate(eyemean = rowMeans(frogs_morph[c('ED_right_mm', 'ED_left_mm')], na.rm=TRUE)) %>% #adds mean of L/R eyes
  mutate(cormean = rowMeans(frogs_morph[c('CD_right_mm', 'CD_left_mm')], na.rm=TRUE)) %>%  #adds mean of L/R corneas
  mutate(eyemean = na_if(eyemean, "NaN")) %>% #remove missing values
  mutate(cormean = na_if(cormean, "NaN")) %>% #remove missing values
   select(Order, Suborder, Family, Subfamily, Genus, Species, genus_species, mod_tiplabel, SVL_mm, Mass_g, rootmass, ED_right_mm, ED_left_mm, eyemean, CD_right_mm, CD_left_mm, cormean) #keeps only columns of interest for analyses
```

## Ecological traits

We coded 5 ecological traits for each species from published literature, and then merged this data with our specimen measurements by genus and species. 

```{r trait-data, results = "hide"}

#Import ecological trait data
traits <- data.frame(read.csv("../Data/traits_analysis.csv"))

#Change "Unknown" trait categorizations to NA so they will be excluded from statistical comparisons
traits.subset <- traits %>%
  mutate(Adult_habitat = factor(na_if(Adult_habitat, "Unknown"))) %>%
  mutate(Activity_period = factor(na_if(Activity_period, "Unknown"))) %>%
  mutate(Mating_habitat = factor(na_if(Mating_habitat, "Unknown"))) %>%
  mutate(Life_history = factor(na_if(Life_history, "Unknown"))) %>%
  mutate(Larval_habitat = factor(na_if(Larval_habitat, "Unknown"))) %>%
  mutate(Sex_dichromatism = factor(na_if(Sex_dichromatism, "Unknown"))) %>%
  select(genus_species, Adult_habitat, Activity_period, Mating_habitat, 
         Life_history, Larval_habitat, Sex_dichromatism)

# Merge morphology dataset with ecological trait dataset
frogs <- merge(frogs.subset, traits.subset, by="genus_species", 
                   all.x = TRUE, all.y = FALSE)

# Check structure of final dataframe for analyses (numbers and factors in propoer columns)
str(frogs)
```

## Sampling

We sampled all 55 extant anuran families, and morphological sampling for interspecific analysis ranged from n = 1 to n = 7 individuals for each species (based on availability in the collections)

```{r sampling, results = "hide"}

#Number of species and individuals sampled 
counts <-ddply(frogs, .(frogs$Family, frogs$Subfamily, frogs$Genus, frogs$Species), nrow)
names(counts) <- c("Family","Subfamily", "Genus","Species","Sampled")

#create scrolling RMarkdown table of sampling
kable(counts[ , c("Family","Genus","Species","Sampled")], caption = "Species and sampling effort for morphological data from museum specimens.") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top") %>%
  scroll_box(height = "500px")

#export sampling data for supplement
write.csv(counts, file = "../Manuscript/Figures/sampling.csv")
```

## Molecular phylogeny

We downloaded a 309-species phylogeny from Feng et al. (2019) and made a list of all exact species matches and close relatives (substitutions) from our morphological dataset. We then dropped species from the phylogeny that were not present in our dataset and renamed tips to our morphological species (this essentially adds in polytomies at the genus level, but then since we drop unused species it's just a direct replacement of the Feng species with our sampled species in terms of tree topology). 

```{r data-phylogeny, fig.height=50, fig.width=10}

#Import list of species to remove and substitute in Feng tree
changes <- data.frame(read.csv("../Data/phylosubs.csv", header=TRUE))
drops <- as.character(changes$exclude) #create vector of taxa to drop from tree
subs <- changes[ , c("tips_original", "tips_sub")] #create a dataframe with column 1 with all original tip labels, and column 2 with corresponding desired tip labels

#Import full 309 species tree from  Feng et al. (2017) downloaded from Dryad
tree_orig <- read.tree(file = "../Data/Chronogram_309sp.txt") #reads tree

#Drop species not included (as matches or substitutions) in our study
tree_pruned <- drop.tip(phy = tree_orig, tip = drops) 

#Make substitutions of species on tips based on our sampling
tree_pruned$tip.label <- subs[[2]][match(tree_pruned$tip.label, subs[[1]])]

#Test whether tree is ultrametric (it should be)
is.ultrametric(tree_pruned)

#Make this ultrametric tree read as ultrametric
tree_pruned <- force.ultrametric(tree_pruned)

#add additional species to existing genera using phytools
tree_pruned <- add.species.to.genus(tree_pruned, "Kaloula_indochinensis_Microhylidae", where = "root")
tree_pruned <- add.species.to.genus(tree_pruned, "Spea_bombifrons_Scaphiopodidae", where = "root")
tree_pruned <- add.species.to.genus(tree_pruned, "Megophrys_intermedia_Megophryidae", where = "root")

#find node for most recent ancestor between two taxa, add species to closest relatives based on node
mfusca <- findMRCA(tree_pruned, tips = c("Microhyla_fissipes_Microhylidae", "Microhyla_heymonsi_Microhylidae"), type = "node")
tree_pruned <- bind.tip(tree_pruned, "Microhyla_fusca_Microhylidae", where = mfusca)

mpulv <- findMRCA(tree_pruned, tips = c("Microhyla_fissipes_Microhylidae", "Microhyla_heymonsi_Microhylidae"), type = "node")
tree_pruned <- bind.tip(tree_pruned, "Microhyla_pulverata_Microhylidae", where = mpulv)

mgerti <- findMRCA(tree_pruned, tips = c("Megophrys_microstoma_Megophryidae", "Megophrys_brachykolos_Megophryidae"), type = "node")
tree_pruned <- bind.tip(tree_pruned, "Megophrys_gerti_Megophryidae", where = mgerti)

gcarol <- findMRCA(tree_pruned, tips = c("Gastrophryne_olivacea_Microhylidae", "Elachistocleis_ovalis_Microhylidae"), type = "node")
tree_pruned <- bind.tip(tree_pruned, "Gastrophryne_carolinensis_Microhylidae", where = gcarol)

oquix <- findMRCA(tree_pruned, tips = c("Barycholos_pulcher_Strabomantidae", "Strabomantis_bufoniformis_Strabomantidae"), type = "node")
tree_pruned <- bind.tip(tree_pruned, "Oreobates_quixensis_Strabomantidae", where = oquix)

ldringi <- findMRCA(tree_pruned, tips = c("Scutiger_boulengeri_Megophryidae","Leptobrachella_bidoupensis_Megophryidae"), type = "node")
tree_pruned <- bind.tip(tree_pruned, "Leptobrachella_dringi_Megophryidae", where = ldringi)

#ladderize tree
frog.tree <- ladderize(tree_pruned)

#check tree
is.rooted(frog.tree) #tests whether tree is rooted 
is.binary.tree(frog.tree) #tests whether tree is dichotomous (no polytomies)

#make dichotomous
frog.tree <- multi2di(frog.tree)

#Plot resulting tree
plot.phylo(frog.tree, 
           type = "phylogram", 
           show.tip.label = TRUE, 
           cex = 0.7, #text size
           no.margin = TRUE, 
           use.edge.length = TRUE,
           edge.width = 1,
           label.offset = .02)
```

# Interspecific allometry

## Species means for morphological comparisons

We then found species means for various morphological parameters. In order to keep as many specimens as possible in our comparisons, we created a new dataframe for each set of morphological comparisons, so that the few specimens with missing data for one parameter could still be used in comparisons not including that parameter. 

```{r species-av}

## SPECIES AVERAGES (1-7 specimens per species) FOR VARIOUS COMPARISONS

##### eye diameter vs. snout vent length #####

# Take mean of replicate specimens for each species (and remove specimens with missing data for two traits of interest)
av.edsvl <- frogs %>% 
  mutate_if(is.character, as.factor) %>% 
  filter(!(is.na(eyemean) | is.na(SVL_mm))) %>% 
  group_by(genus_species) %>%
  summarise(eye_av = mean(eyemean), svl_av = mean(SVL_mm), n = n())

## Merge data means with other column info
av.edsvl <- merge(av.edsvl, frogs[match(unique(frogs$genus_species), frogs$genus_species), ], by="genus_species", all.x = TRUE, all.y = FALSE)

#Keep averages and associated traits only
av.edsvl <- av.edsvl %>% select(genus_species, eye_av, svl_av, n, Order, Suborder, Family, Subfamily, Genus, Species, mod_tiplabel,Adult_habitat, Activity_period, Mating_habitat, Life_history, Larval_habitat, Sex_dichromatism) 


##### eye diameter vs. the cube root of mass #####

# Take mean of replicate specimens for each species (and remove specimens with missing data for two traits of interest)
av.edmass <- frogs %>% 
  mutate_if(is.character, as.factor) %>% 
  filter(!(is.na(eyemean) | is.na(rootmass))) %>% 
  group_by(genus_species) %>%
  summarise(eye_av = mean(eyemean), rootmass_av = mean(rootmass), n = n())

## Merge data means with other column info
av.edmass <- merge(av.edmass, frogs[match(unique(frogs$genus_species), frogs$genus_species), ], by="genus_species", all.x = TRUE, all.y = FALSE)

#Keep averages and associated traits only
av.edmass <- av.edmass %>% select(genus_species, eye_av, rootmass_av, n, Order, Suborder, Family, Subfamily, Genus, Species, mod_tiplabel,Adult_habitat, Activity_period, Mating_habitat, Life_history, Larval_habitat, Sex_dichromatism) 


##### cube root of mass vs. SVL #####

# Take mean of replicate specimens for each species (and remove specimens with missing data for two traits of interest)
av.svlmass <- frogs %>% 
  mutate_if(is.character, as.factor) %>% 
  filter(!(is.na(SVL_mm) | is.na(rootmass))) %>% 
  group_by(genus_species) %>%
  summarise(svl_av = mean(SVL_mm), rootmass_av = mean(rootmass), n = n())

## Merge data means with other column info
av.svlmass <- merge(av.svlmass, frogs[match(unique(frogs$genus_species), frogs$genus_species), ], by="genus_species", all.x = TRUE, all.y = FALSE)

#Keep averages and associated traits only
av.svlmass <- av.svlmass %>% select(genus_species, svl_av, rootmass_av, n, Order, Suborder, Family, Subfamily, Genus, Species, mod_tiplabel,Adult_habitat, Activity_period, Mating_habitat, Life_history, Larval_habitat, Sex_dichromatism) 

##### cornea diameter vs. eye diameter #####

# Take mean of replicate specimens for each species (and remove specimens with missing data for two traits of interest)
av.edcd <- frogs %>% 
  filter(!(is.na(cormean)|is.na(eyemean))) %>% 
  mutate_if(is.character, as.factor) %>% 
  filter(!is.na(eyemean) | !is.na(cormean)) %>% 
  group_by(genus_species) %>%
  summarise(eye_av = mean(eyemean), cornea_av = mean(cormean), n = n())

## Merge data means with other column info
av.edcd <- merge(av.edcd, frogs[match(unique(frogs$genus_species), frogs$genus_species), ], by="genus_species", all.x = TRUE, all.y = FALSE)

#Keep averages and associated traits only
av.edcd <- av.edcd %>% select(genus_species, eye_av, cornea_av, n, Order, Suborder, Family, Subfamily, Genus, Species, mod_tiplabel,Adult_habitat, Activity_period, Mating_habitat, Life_history, Larval_habitat, Sex_dichromatism) 

##### cornea diameter vs. snout vent length #####

# Take mean of replicate specimens for each species (and remove specimens with missing data for two traits of interest)
av.cdsvl <- frogs %>% 
  mutate_if(is.character, as.factor) %>% 
  filter(!(is.na(cormean) | is.na(SVL_mm))) %>% 
  group_by(genus_species) %>%
  summarise(cornea_av = mean(cormean), svl_av = mean(SVL_mm), n = n())

## Merge data means with other column info
av.cdsvl <- merge(av.cdsvl, frogs[match(unique(frogs$genus_species), frogs$genus_species), ], by="genus_species", all.x = TRUE, all.y = FALSE)

#Keep averages and associated traits only
av.cdsvl <- av.cdsvl %>% select(genus_species, cornea_av, svl_av, n, Order, Suborder, Family, Subfamily, Genus, Species, mod_tiplabel,Adult_habitat, Activity_period, Mating_habitat, Life_history, Larval_habitat, Sex_dichromatism) 


##### cornea diameter vs. cube root of mass #####

# Take mean of replicate specimens for each species (and remove specimens with missing data for two traits of interest)
av.cdmass <- frogs %>% 
  mutate_if(is.character, as.factor) %>% 
  filter(!(is.na(cormean) | is.na(rootmass))) %>% 
  group_by(genus_species) %>%
  summarise(cornea_av = mean(cormean), rootmass_av = mean(rootmass), n = n())

## Merge data means with other column info
av.cdmass <- merge(av.cdmass, frogs[match(unique(frogs$genus_species), frogs$genus_species), ], by="genus_species", all.x = TRUE, all.y = FALSE)

#Keep averages and associated traits only
av.cdmass <- av.cdmass %>% select(genus_species, cornea_av, rootmass_av, n, Order, Suborder, Family, Subfamily, Genus, Species, mod_tiplabel,Adult_habitat, Activity_period, Mating_habitat, Life_history, Larval_habitat, Sex_dichromatism) 
```

## Phylogenetic generalised least squares regressions

We fit PGLS models using the modified Feng et al. (2017) phylogeny and our averaged species data for morphological traits. 

### PGLS of log eye diameter vs. log cube root of mass

```{r pgls-edmass}

#fit PGLS model for log eye diameter vs. log cube root of mass using 'caper' package

#### check for best model of correlation structure ####

#Make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(frog.tree$tip.label, as.character(av.edmass$mod_tiplabel))

#Drop unwanted tips from phylogeny
tree.edit <- drop.tip(phy = frog.tree, tip = drops) 

#Make row names of the species the phylogeny tip labels
rownames(av.edmass) <- av.edmass$mod_tiplabel

#Reorder species data to match phylogeny order
dat <-ReorderData(tree.edit, av.edmass, taxa.names="row names")

#test whether BM or OU model is better fit for PGLS using the nlme package and AIC comparison

#Expected covariance under a pure Brownian model (Felsenstein 1985,	Martins and Hansen 1997)
pglsBM <- gls(log10(eye_av) ~ log10(rootmass_av), cor = corBrownian(phy = tree.edit), data = dat, method = "ML")

#Martins and Hansen's (1997) covariance structure (OU)
pglsOU<-gls(log10(eye_av) ~ log10(rootmass_av), cor = corMartins(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

#Pagel's “lambda” Correlation Structure: The correlation structure from the present model is derived from the Brownian motion model by multiplying the off-diagonal elements (i.e., the covariances) by lambda. The variances are thus the same than for a Brownian motion model.
pglsPagel<-gls(log10(eye_av) ~ log10(rootmass_av), cor = corPagel(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

anova(pglsBM, pglsOU, pglsPagel)

#### prep data ####

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edmass, data.names = av.edmass$mod_tiplabel)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edmass.comp <- comparative.data(phy = frog.tree, data = av.edmass, 
                            names.col = mod_tiplabel, vcv = TRUE, 
                            na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edmass.comp$dropped$tips #phylogeny
edmass.comp$dropped$unmatched.rows #dataset

#rescale tree to fit OU model with alpha = 0.5
frog.treeOU <- geiger::rescale(frog.tree, model = "OU", 0.5)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edmass.compOU <- comparative.data(phy = frog.treeOU, data = av.edmass, 
                            names.col = mod_tiplabel, vcv = TRUE, 
                            na.omit = FALSE, warn.dropped = TRUE)


#for eye diameter vs. cube root of mass using the OU tree
pgls_edmassOU <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.compOU, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#### run PGLS model ####

#for eye diameter vs. cube root of mass using the Maximum Liklihood estimate of lambda
pgls_edmass <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#### check model assumptions ####

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edmass)
par(mfrow = c(1, 1))

#### model outputs ####

#print model output 
summary(pgls_edmass)

#quick plot of pgls
plot(log10(eye_av) ~ log10(rootmass_av), data = av.edmass)
abline(pgls_edmass)

#Likelihood plot for Pagel's lambda from the PGLS model of eye diameter vs. the cuberoot of mass. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.edmass <- pgls.profile(pgls_edmass, "lambda")
plot(lambda.edmass)

```

For the PGLS of log eye diameter vs. log cube root of mass, Pagel's lambda was `r pgls.confint(pgls_edmass, "lambda")$opt` with 95% confidence intervals of `r pgls.confint(pgls_edmass, "lambda")$ci.val`, indicating that there is high phylogenetic signal in the residuals of our PGLS model fit (which is our measure of relative eye investment).

The diagnostic plots for the model show that the data conformed to most model assumptions, though we do have one species with the absolute value of studentized residual exceeding 3, which could disproportionately influence the regression. To check whether this is happening, we re-ran the model iteratively with outliers removed to check the robustness of parameter estimates. 

```{r pgls-edmass-outliers, include = FALSE}

# refit pgls after removing extreme outlier/s observed in previous fit

#### Iteration 1 ####

#extract the phylogenetic residuals from the original model
phyres.edmass1 <- residuals(pgls_edmass, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edmass1 <- phyres.edmass1/sqrt(var(phyres.edmass1))[1] 

#match species names with standardized residuals
rownames(stphyres.edmass1)<-rownames(pgls_edmass$residuals)

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edmass1)[(abs(stphyres.edmass1)>3)]
 
#remove these two species from data/tree
edmass.comp_1 <- edmass.comp[-which(abs(stphyres.edmass1)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edmass_1 <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.comp_1, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edmass_1, main = "PGLS eye diameter v root mass (- outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edmass_1 , "lambda"))

#print model output 
summary(pgls_edmass_1)

#### Iteration 2 ####

#extract the phylogenetic residuals from the original model
phyres.edmass2 <- residuals(pgls_edmass_1, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edmass2 <- phyres.edmass2/sqrt(var(phyres.edmass2))[1] 

#match species names with standardized residuals
rownames(stphyres.edmass2)<-rownames(pgls_edmass_1$residuals)

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edmass2)[(abs(stphyres.edmass2)>3)]
 
#remove these two species from data/tree
edmass.comp_2 <- edmass.comp_1[-which(abs(stphyres.edmass2)>3), ]

#re-run pgls analisys (without the outliers)
pgls_edmass_2 <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.comp_2, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edmass_2, main = "PGLS model for eye v root mass (- outliers 2)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edmass_2 , "lambda"))

#print model output 
summary(pgls_edmass_2)

#### Iteration 3 ####

#extract the phylogenetic residuals from the original model
phyres.edmass3 <- residuals(pgls_edmass_2, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edmass3 <- phyres.edmass3/sqrt(var(phyres.edmass3))[1] 

#match species names with standardized residuals
rownames(stphyres.edmass3)<-rownames(pgls_edmass_2$residuals)

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edmass3)[(abs(stphyres.edmass3)>3)]
 
#remove these two species from data/tree
edmass.comp_3 <- edmass.comp_2[-which(abs(stphyres.edmass3)>3), ]

#re-run pgls analisys (without the outliers)
pgls_edmass_3 <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.comp_3, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edmass_3, main = "PGLS model for eye v root mass (- outliers 3)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edmass_3 , "lambda"))

#print model output 
summary(pgls_edmass_3)

#### Iteration 4 ####

#extract the phylogenetic residuals from the original model
phyres.edmass4 <- residuals(pgls_edmass_3, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edmass4 <- phyres.edmass4/sqrt(var(phyres.edmass4))[1] 

#match species names with standardized residuals
rownames(stphyres.edmass4)<-rownames(pgls_edmass_3$residuals)

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edmass4)[(abs(stphyres.edmass4)>3)]
 
#remove these two species from data/tree
edmass.comp_4 <- edmass.comp_3[-which(abs(stphyres.edmass4)>3), ]

#re-run pgls analisys (without the outliers)
pgls_edmass_4 <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.comp_4, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edmass_4, main = "PGLS model for eye v root mass (- outliers 4)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edmass_4 , "lambda"))

#print model output 
summary(pgls_edmass_4)

#### Iteration 5 ####

#extract the phylogenetic residuals from the original model
phyres.edmass5 <- residuals(pgls_edmass_4, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edmass5 <- phyres.edmass5/sqrt(var(phyres.edmass5))[1] 

#match species names with standardized residuals
rownames(stphyres.edmass5)<-rownames(pgls_edmass_4$residuals)

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edmass5)[(abs(stphyres.edmass5)>3)]
 
#remove these two species from data/tree
edmass.comp_5 <- edmass.comp_4[-which(abs(stphyres.edmass5)>3), ]

#re-run pgls analisys (without the outliers)
pgls_edmass_5 <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.comp_5, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edmass_5, main = "PGLS model for eye v root mass (no outliers 5)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edmass_5 , "lambda"))

#print model output 
summary(pgls_edmass_5)

#### Iteration 6 ####

#extract the phylogenetic residuals from the original model
phyres.edmass6 <- residuals(pgls_edmass_5, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edmass6 <- phyres.edmass6/sqrt(var(phyres.edmass6))[1] 

#match species names with standardized residuals
rownames(stphyres.edmass6) <- rownames(pgls_edmass_5$residuals)

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edmass6)[(abs(stphyres.edmass6)>3)]
 
#remove these two species from data/tree
edmass.comp_6 <- edmass.comp_5[-which(abs(stphyres.edmass6)>3), ]

#re-run pgls analisys (without the outliers)
pgls_edmass_6 <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.comp_6, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edmass_6, main = "PGLS model for eye v root mass (no outliers)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edmass_6 , "lambda"))

#print model output 
summary(pgls_edmass_6)

#### Iteration 7 ####

#extract the phylogenetic residuals from the original model
phyres.edmass7 <- residuals(pgls_edmass_6, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edmass7 <- phyres.edmass7/sqrt(var(phyres.edmass7))[1] 

#match species names with standardized residuals
rownames(stphyres.edmass7)<-rownames(pgls_edmass_6$residuals)

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edmass7)[(abs(stphyres.edmass7)>3)]

#remove these two species from data/tree
edmass.comp_7 <- edmass.comp_6[-which(abs(stphyres.edmass7)>3), ]

#re-run pgls analisys (without the outliers)
pgls_edmass_7 <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.comp_7, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edmass_7, main = "PGLS model for eye v root mass (no outliers)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edmass_7 , "lambda"))

#print model output 
summary(pgls_edmass_7)

#### Iteration 8 ####

#extract the phylogenetic residuals from the original model
phyres.edmass8 <- residuals(pgls_edmass_7, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edmass8 <- phyres.edmass8/sqrt(var(phyres.edmass8))[1] 

#match species names with standardized residuals
rownames(stphyres.edmass8)<-rownames(pgls_edmass_7$residuals)

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edmass8)[(abs(stphyres.edmass8)>3)]

#remove these two species from data/tree
edmass.comp_8 <- edmass.comp_7[-which(abs(stphyres.edmass8)>3), ]

#re-run pgls analisys (without the outliers)
pgls_edmass_8 <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.comp_8, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edmass_8, main = "PGLS model for eye v root mass (no outliers)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edmass_8 , "lambda"))
 
#print model output 
summary(pgls_edmass_8)

#### Iteration 9 ####

#extract the phylogenetic residuals from the original model
phyres.edmass9 <- residuals(pgls_edmass_8, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edmass9 <- phyres.edmass9/sqrt(var(phyres.edmass9))[1] 

#match species names with standardized residuals
rownames(stphyres.edmass9)<-rownames(pgls_edmass_8$residuals)

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edmass9)[(abs(stphyres.edmass9)>3)]

#remove these two species from data/tree
edmass.comp_9 <- edmass.comp_8[-which(abs(stphyres.edmass9)>3), ]

#re-run pgls analisys (without the outliers)
pgls_edmass_9 <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.comp_9, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edmass_9, main = "PGLS model for eye v root mass (no outliers)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edmass_9 , "lambda"))

#print model output 
summary(pgls_edmass_9)

#### Iteration 10 ####

#extract the phylogenetic residuals from the original model
phyres.edmass10 <- residuals(pgls_edmass_9, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edmass10 <- phyres.edmass10/sqrt(var(phyres.edmass10))[1] 

#match species names with standardized residuals
rownames(stphyres.edmass10)<-rownames(pgls_edmass_9$residuals)

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edmass10)[(abs(stphyres.edmass10)>3)]

#remove these two species from data/tree
edmass.comp_10 <- edmass.comp_8[-which(abs(stphyres.edmass9)>3), ]

#re-run pgls analisys (without the outliers)
pgls_edmass_10 <- pgls(log10(eye_av) ~ log10(rootmass_av), 
               data = edmass.comp_10, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edmass_10, main = "PGLS model for eye v root mass (no outliers)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edmass_10 , "lambda"))

#print model output 
summary(pgls_edmass_10)
```

As iterative removal of phylogenetic outliers has no major effect on model parameter estimates, we include all species in our PGLS of log eye diameter vs. log cube root of mass. 

### PGLS of log eye diameter vs. log snout-vent length 

```{r pgls-edsvl}

#fit PGLS model for log eye diameter vs. log SVL using 'caper' package

#### check for best model of correlation structure ####

#Make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(frog.tree$tip.label, as.character(av.edsvl$mod_tiplabel))

#Drop unwanted tips from phylogeny
tree.edit <- drop.tip(phy = frog.tree, tip = drops) 

#Make row names of the species the phylogeny tip labels
rownames(av.edsvl) <- av.edsvl$mod_tiplabel

#Reorder species data to match phylogeny order
dat <-ReorderData(tree.edit, av.edsvl, taxa.names="row names")

#test whether BM or OU model is better fit for PGLS using the nlme package and AIC comparison

#Expected covariance under a pure Brownian model (Felsenstein 1985,	Martins and Hansen 1997)
pglsBM <- gls(log10(eye_av) ~ log10(svl_av), cor = corBrownian(phy = tree.edit), data = dat, method = "ML")

#Martins and Hansen's (1997) covariance structure (OU)
pglsOU<-gls(log10(eye_av) ~ log10(svl_av), cor = corMartins(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

#Pagel's “lambda” Correlation Structure: The correlation structure from the present model is derived from the Brownian motion model by multiplying the off-diagonal elements (i.e., the covariances) by lambda. The variances are thus the same than for a Brownian motion model.
pglsPagel<-gls(log10(eye_av) ~ log10(svl_av), cor = corPagel(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

anova(pglsBM, pglsOU, pglsPagel)

#### prep data ####

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edsvl, data.names = av.edsvl$mod_tiplabel)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edsvl.comp <- comparative.data(phy = frog.tree, data = av.edsvl, 
                            names.col = mod_tiplabel, vcv = TRUE, 
                            na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edsvl.comp$dropped$tips #phylogeny
edsvl.comp$dropped$unmatched.rows #dataset

#### run PGLS model ####

#for log eye diameter vs. log svl using the Maximum Liklihood estimate of lambda
pgls_edsvl <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#### check model assumptions ####

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl, main = "PGLS model for mean eye diameter vs. svl")
par(mfrow = c(1, 1))

#### model outputs ####

#print model output 
summary(pgls_edsvl)

#quick plot of pgls
plot(log10(eye_av) ~ log10(svl_av), data = av.edsvl)
abline(pgls_edsvl)

#Likelihood plot for Pagel's lambda from the PGLS model of eye diameter vs. the cuberoot of mass. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.edsvl <- pgls.profile(pgls_edsvl, "lambda")
plot(lambda.edsvl)

```

For the PGLS of log mean eye diameter vs. log snout-vent length, Pagel's lambda was `r pgls.confint(pgls_edsvl, "lambda")$opt` with 95% confidence intervals of `r pgls.confint(pgls_edsvl, "lambda")$ci.val`, indicating that there is high phylogenetic signal in the residuals of our PGLS model fit (which is our measure of relative eye investment).

The diagnostic plots show that the data conformed to most model assumptions, though we do have two species with studentized residuals exceeding ±3, which could disproportionately influence the regression. To check whether this is happening, we re-ran the model with these two points eliminated. 

```{r pgls-edsvl-outliers, include = FALSE}

# refit pgls after removing extreme outlier/s observed in previous fit

#### Iteration 1 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl1 <- residuals(pgls_edsvl, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl1 <- phyres.edsvl1/sqrt(var(phyres.edsvl1))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl1)<-rownames(pgls_edsvl$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl1)[(abs(stphyres.edsvl1)>3)]
 
#remove these two species from data/tree
edsvl.comp_1 <- edsvl.comp[-which(abs(stphyres.edsvl1)>3), ]

#re-run pgls analisys (without the outliers)
pgls_edsvl_1 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_1, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_1, main = "PGLS model for eye v svl (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_1 , "lambda"))

#print model output 
summary(pgls_edsvl_1)

#### Iteration 2 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl2 <- residuals(pgls_edsvl_1, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl2 <- phyres.edsvl2/sqrt(var(phyres.edsvl2))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl2)<-rownames(pgls_edsvl_1$residuals) #matches the residuals up with the species names

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl2)[(abs(stphyres.edsvl2)>3)]
 
#remove this species from data/tree
edsvl.comp_2 <- edsvl.comp_1[-which(abs(stphyres.edsvl2)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_2 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_2, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_2, main = "PGLS model for eye v svl (no outliers 2)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_2 , "lambda"))

#print model output 
summary(pgls_edsvl_2)

#### Iteration 3 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl3 <- residuals(pgls_edsvl_2, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl3 <- phyres.edsvl3/sqrt(var(phyres.edsvl3))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl3)<-rownames(pgls_edsvl_2$residuals) #matches the residuals up with the species names

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl3)[(abs(stphyres.edsvl3)>3)]
 
#remove this species from data/tree
edsvl.comp_3 <- edsvl.comp_2[-which(abs(stphyres.edsvl3)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_3 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_3, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_3, main = "PGLS model for eye v svl (no outliers 3)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_3 , "lambda"))

#print model output 
summary(pgls_edsvl_3)

#### Iteration 4 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl4 <- residuals(pgls_edsvl_3, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl4 <- phyres.edsvl4/sqrt(var(phyres.edsvl4))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl4)<-rownames(pgls_edsvl_3$residuals) #matches the residuals up with the species names

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl4)[(abs(stphyres.edsvl4)>3)]
 
#remove these two species from data/tree
edsvl.comp_4 <- edsvl.comp_3[-which(abs(stphyres.edsvl4)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_4 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_4, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_4, main = "PGLS model for eye v svl (no outliers 4)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_4 , "lambda"))

#print model output 
summary(pgls_edsvl_4)

#### Iteration 5 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl5 <- residuals(pgls_edsvl_4, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl5 <- phyres.edsvl5/sqrt(var(phyres.edsvl5))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl5)<-rownames(pgls_edsvl_4$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl5)[(abs(stphyres.edsvl5)>3)]
 
#remove these two species from data/tree
edsvl.comp_5 <- edsvl.comp_4[-which(abs(stphyres.edsvl5)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_5 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_5, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_5, main = "PGLS model for eye v svl (no outliers 5)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_5 , "lambda"))

#print model output 
summary(pgls_edsvl_5)

#### Iteration 6 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl6 <- residuals(pgls_edsvl_5, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl6 <- phyres.edsvl6/sqrt(var(phyres.edsvl6))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl6)<-rownames(pgls_edsvl_5$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl6)[(abs(stphyres.edsvl6)>3)]
 
#remove these two species from data/tree
edsvl.comp_6 <- edsvl.comp_5[-which(abs(stphyres.edsvl6)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_6 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_6, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_6, main = "PGLS model for eye v svl (no outliers 5)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_6 , "lambda"))

#print model output 
summary(pgls_edsvl_6)

#### Iteration 7 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl7 <- residuals(pgls_edsvl_6, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl7 <- phyres.edsvl7/sqrt(var(phyres.edsvl7))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl7)<-rownames(pgls_edsvl_6$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl7)[(abs(stphyres.edsvl7)>3)]
 
#remove these two species from data/tree
edsvl.comp_7 <- edsvl.comp_6[-which(abs(stphyres.edsvl7)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_7 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_7, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_7, main = "PGLS model for eye v svl (no outliers 7)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_7 , "lambda"))

#print model output 
summary(pgls_edsvl_7)

#### Iteration 8 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl8 <- residuals(pgls_edsvl_7, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl8 <- phyres.edsvl8/sqrt(var(phyres.edsvl8))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl8)<-rownames(pgls_edsvl_7$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl8)[(abs(stphyres.edsvl8)>3)]
 
#remove these two species from data/tree
edsvl.comp_8 <- edsvl.comp_7[-which(abs(stphyres.edsvl8)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_8 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_8, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_8, main = "PGLS model for eye v svl (no outliers 7)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_8 , "lambda"))

#print model output 
summary(pgls_edsvl_8)

#### Iteration 9 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl9 <- residuals(pgls_edsvl_8, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl9 <- phyres.edsvl9/sqrt(var(phyres.edsvl9))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl9)<-rownames(pgls_edsvl_8$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl9)[(abs(stphyres.edsvl9)>3)]
 
#remove these two species from data/tree
edsvl.comp_9 <- edsvl.comp_8[-which(abs(stphyres.edsvl9)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_9 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_9, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_9, main = "PGLS model for eye v svl (no outliers 9)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_9 , "lambda"))

#print model output 
summary(pgls_edsvl_9)

#### Iteration 10 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl10 <- residuals(pgls_edsvl_9, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl10 <- phyres.edsvl10/sqrt(var(phyres.edsvl10))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl10)<-rownames(pgls_edsvl_9$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl10)[(abs(stphyres.edsvl10)>3)]
 
#remove these two species from data/tree
edsvl.comp_10 <- edsvl.comp_9[-which(abs(stphyres.edsvl10)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_10 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_10, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_10, main = "PGLS model for eye v svl (no outliers 10)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_10 , "lambda"))

#print model output 
summary(pgls_edsvl_10)

#### Iteration 11 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl11 <- residuals(pgls_edsvl_10, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl11 <- phyres.edsvl11/sqrt(var(phyres.edsvl11))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl11)<-rownames(pgls_edsvl_10$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl11)[(abs(stphyres.edsvl11)>3)]
 
#remove these two species from data/tree
edsvl.comp_11 <- edsvl.comp_10[-which(abs(stphyres.edsvl11)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_11 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_11, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_11, main = "PGLS model for eye v svl (no outliers 11)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_11 , "lambda"))

#print model output 
summary(pgls_edsvl_11)

#### Iteration 12 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl12 <- residuals(pgls_edsvl_11, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl12 <- phyres.edsvl12/sqrt(var(phyres.edsvl12))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl12)<-rownames(pgls_edsvl_11$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl12)[(abs(stphyres.edsvl12)>3)]
 
#remove these two species from data/tree
edsvl.comp_12 <- edsvl.comp_11[-which(abs(stphyres.edsvl12)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_12 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_12, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_12, main = "PGLS model for eye v svl (no outliers 12)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_12 , "lambda"))

#print model output 
summary(pgls_edsvl_12)

#### Iteration 13 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl13 <- residuals(pgls_edsvl_12, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl13 <- phyres.edsvl13/sqrt(var(phyres.edsvl13))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl13)<-rownames(pgls_edsvl_12$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl13)[(abs(stphyres.edsvl13)>3)]
 
#remove these two species from data/tree
edsvl.comp_13 <- edsvl.comp_12[-which(abs(stphyres.edsvl13)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_13 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_13, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_13, main = "PGLS model for eye v svl (no outliers 13)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_13 , "lambda"))

#print model output 
summary(pgls_edsvl_13)

#### Iteration 14 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl14 <- residuals(pgls_edsvl_13, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl14 <- phyres.edsvl14/sqrt(var(phyres.edsvl14))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl14)<-rownames(pgls_edsvl_13$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl14)[(abs(stphyres.edsvl14)>3)]
 
#remove these two species from data/tree
edsvl.comp_14 <- edsvl.comp_13[-which(abs(stphyres.edsvl14)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_14 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_14, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_14, main = "PGLS model for eye v svl (no outliers 14)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_14 , "lambda"))

#print model output 
summary(pgls_edsvl_14)

#### Iteration 15 ####

#extract the phylogenetic residuals from the original model
phyres.edsvl15 <- residuals(pgls_edsvl_14, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edsvl15 <- phyres.edsvl15/sqrt(var(phyres.edsvl15))[1] 

#match species names with standardized residuals
rownames(stphyres.edsvl15)<-rownames(pgls_edsvl_14$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edsvl15)[(abs(stphyres.edsvl15)>3)]
 
#remove these two species from data/tree
edsvl.comp_15 <- edsvl.comp_14[-which(abs(stphyres.edsvl15)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edsvl_15 <- pgls(log10(eye_av) ~ log10(svl_av), 
               data = edsvl.comp_15, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edsvl_15, main = "PGLS model for eye v svl (no outliers 14)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edsvl_15 , "lambda"))

#print model output 
summary(pgls_edsvl_15)
```

Again, as iterative removal of phylogenetic outliers did not qualitatively change our results, we use the full species dataset for our PGLS of log eye diameter vs. log snout-vent length among anurans. 

### PGLS of log corneal diameter vs. log eye diameter

```{r pgls-fit-cded}
#fit PGLS model for log corneal diameter vs. log eye diameter using 'caper' package

#### prep data ####

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edcd, data.names = av.edcd$mod_tiplabel)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edcd.comp <- comparative.data(phy = frog.tree, data = av.edcd, 
                              names.col = mod_tiplabel, vcv = TRUE, 
                              na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edcd.comp$dropped$tips #phylogeny
edcd.comp$dropped$unmatched.rows #dataset

#### run PGLS model ####

#for log eye diameter vs. log svl using the Maximum Liklihood estimate of lambda
pgls_edcd <- pgls(log10(cornea_av) ~ log10(eye_av), 
                  data = edcd.comp, 
                  lambda = "ML", #uses Maximum Liklihood estimate of lambda
                  param.CI = 0.95)

#### check model assumptions ####

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edcd, main = "PGLS model for mean cornea diameter vs. mean eye diameter")
par(mfrow = c(1, 1))

#### model outputs ####

#print model output 
summary(pgls_edcd)

#quick plot of pgls
plot(log10(cornea_av) ~ log10(eye_av), data = av.edcd)
abline(pgls_edcd)

#Likelihood plot for Pagel's lambda from the PGLS model of eye diameter vs. the cuberoot of mass. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.edcd <- pgls.profile(pgls_edcd, "lambda")
plot(lambda.edcd)

```

For the PGLS of mean cornea diameter vs. mean eye diameter, Pagel's lambda was `r pgls.confint(pgls_edcd, "lambda")$opt` with 95% confidence intervals of `r pgls.confint(pgls_edcd, "lambda")$ci.val`, indicating that there is low phylogenetic signal in the residuals of our PGLS model fit (which is our measure of relative corneal investment).

The diagnostic plots show that the data conformed to most model assumptions, though we do have species with studentized residuals exceeding ±3, which could disproportionately influence the regression. We thus iteratively re-ran the model with phylogenetic outliers removed. 

```{r pgls-edcd-outliers, include = FALSE}

# refit pgls after removing extreme outlier/s observed in previous fit

#### Iteration 1 ####

#extract the phylogenetic residuals from the original model
phyres.edcd1 <- residuals(pgls_edcd, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edcd1 <- phyres.edcd1/sqrt(var(phyres.edcd1))[1] 

#match species names with standardized residuals
rownames(stphyres.edcd1)<-rownames(pgls_edcd$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edcd1)[(abs(stphyres.edcd1)>3)]
 
#remove these two species from data/tree
edcd.comp_1 <- edcd.comp[-which(abs(stphyres.edcd1)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edcd_1 <- pgls(log10(cornea_av) ~ log10(eye_av), 
               data = edcd.comp_1, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edcd_1, main = "PGLS model for cornea vs. eye (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edcd_1 , "lambda"))

#print model output 
summary(pgls_edcd_1)

#### Iteration 2 ####

#extract the phylogenetic residuals from the original model
phyres.edcd2 <- residuals(pgls_edcd_1, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edcd2 <- phyres.edcd2/sqrt(var(phyres.edcd2))[1] 

#match species names with standardized residuals
rownames(stphyres.edcd2)<-rownames(pgls_edcd_1$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edcd2)[(abs(stphyres.edcd2)>3)]
 
#remove these two species from data/tree
edcd.comp_2 <- edcd.comp_1[-which(abs(stphyres.edcd2)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edcd_2 <- pgls(log10(cornea_av) ~ log10(eye_av), 
               data = edcd.comp_2, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edcd_2, main = "PGLS model for cornea vs. eye (no outliers 2)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edcd_2 , "lambda"))

#print model output 
summary(pgls_edcd_2)

#### Iteration 3 ####

#extract the phylogenetic residuals from the original model
phyres.edcd3 <- residuals(pgls_edcd_2, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edcd3 <- phyres.edcd3/sqrt(var(phyres.edcd3))[1] 

#match species names with standardized residuals
rownames(stphyres.edcd3)<-rownames(pgls_edcd_2$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edcd3)[(abs(stphyres.edcd3)>3)]
 
#remove these two species from data/tree
edcd.comp_3 <- edcd.comp_2[-which(abs(stphyres.edcd3)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edcd_3 <- pgls(log10(cornea_av) ~ log10(eye_av), 
               data = edcd.comp_3, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edcd_3, main = "PGLS model for cornea vs. eye (no outliers 2)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edcd_3 , "lambda"))

#print model output 
summary(pgls_edcd_3)

#### Iteration 4 ####

#extract the phylogenetic residuals from the original model
phyres.edcd4 <- residuals(pgls_edcd_3, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edcd4 <- phyres.edcd4/sqrt(var(phyres.edcd4))[1] 

#match species names with standardized residuals
rownames(stphyres.edcd4)<-rownames(pgls_edcd_3$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edcd4)[(abs(stphyres.edcd4)>3)]
 
#remove these two species from data/tree
edcd.comp_4 <- edcd.comp_3[-which(abs(stphyres.edcd4)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edcd_4 <- pgls(log10(cornea_av) ~ log10(eye_av), 
               data = edcd.comp_4, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edcd_4, main = "PGLS model for cornea vs. eye (no outliers 2)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edcd_4 , "lambda"))

#print model output 
summary(pgls_edcd_4)

#### Iteration 5 ####

#extract the phylogenetic residuals from the original model
phyres.edcd5 <- residuals(pgls_edcd_4, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edcd5 <- phyres.edcd5/sqrt(var(phyres.edcd5))[1] 

#match species names with standardized residuals
rownames(stphyres.edcd5)<-rownames(pgls_edcd_4$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edcd5)[(abs(stphyres.edcd5)>3)]
 
#remove these two species from data/tree
edcd.comp_5 <- edcd.comp_4[-which(abs(stphyres.edcd5)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edcd_5 <- pgls(log10(cornea_av) ~ log10(eye_av), 
               data = edcd.comp_5, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edcd_5, main = "PGLS model for cornea vs. eye (no outliers 2)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edcd_5 , "lambda"))

#print model output 
summary(pgls_edcd_5)

#### Iteration 6 ####

#extract the phylogenetic residuals from the original model
phyres.edcd6 <- residuals(pgls_edcd_5, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edcd6 <- phyres.edcd6/sqrt(var(phyres.edcd6))[1] 

#match species names with standardized residuals
rownames(stphyres.edcd6)<-rownames(pgls_edcd_5$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edcd6)[(abs(stphyres.edcd6)>3)]
 
#remove these two species from data/tree
edcd.comp_6 <- edcd.comp_5[-which(abs(stphyres.edcd6)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edcd_6 <- pgls(log10(cornea_av) ~ log10(eye_av), 
               data = edcd.comp_6, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edcd_6, main = "PGLS model for cornea vs. eye (no outliers 2)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edcd_6, "lambda"))

#print model output 
summary(pgls_edcd_6)

#### Iteration 7 ####

#extract the phylogenetic residuals from the original model
phyres.edcd7 <- residuals(pgls_edcd_6, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edcd7 <- phyres.edcd7/sqrt(var(phyres.edcd7))[1] 

#match species names with standardized residuals
rownames(stphyres.edcd7)<-rownames(pgls_edcd_6$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edcd7)[(abs(stphyres.edcd7)>3)]
 
#remove these two species from data/tree
edcd.comp_7 <- edcd.comp_6[-which(abs(stphyres.edcd7)>3), ]

#re-run pgls analysis (without the outliers)
pgls_edcd_7 <- pgls(log10(cornea_av) ~ log10(eye_av), 
               data = edcd.comp_7, 
               lambda = "ML", bounds = list(lambda = c(1e-05, 1)),
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_edcd_7, main = "PGLS model for cornea vs. eye (no outliers 2)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_edcd_7, "lambda"))

#print model output 
summary(pgls_edcd_7)

#### Iteration 8 ####

#extract the phylogenetic residuals from the original model
phyres.edcd8 <- residuals(pgls_edcd_7, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.edcd8 <- phyres.edcd8/sqrt(var(phyres.edcd8))[1] 

#match species names with standardized residuals
rownames(stphyres.edcd8)<-rownames(pgls_edcd_7$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.edcd8)[(abs(stphyres.edcd8)>3)]
 
### No remaining phylogenetic outliers ###
```

### PGLS of log snout-vent length vs. log cube root of mass

```{r pgls-svlrm}

#fit PGLS model for log svl vs. log root mass using 'caper' package

#### prep data ####

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.svlmass, data.names = av.svlmass$mod_tiplabel)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
svlmass.comp <- comparative.data(phy = frog.tree, data = av.svlmass, 
                              names.col = mod_tiplabel, vcv = TRUE, 
                              na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
svlmass.comp$dropped$tips #phylogeny
svlmass.comp$dropped$unmatched.rows #dataset

#### run PGLS model ####

#for log eye diameter vs. log svl using the Maximum Liklihood estimate of lambda
pgls_svlmass <- pgls(log10(svl_av) ~ log10(rootmass_av), 
                  data = svlmass.comp, 
                  lambda = "ML", #uses Maximum Liklihood estimate of lambda
                  param.CI = 0.95)

#### check model assumptions ####

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_svlmass, main = "PGLS model for log mean svl vs. log mean cube root of mass")
par(mfrow = c(1, 1))

#### model outputs ####

#print model output 
summary(pgls_svlmass)

#quick plot of pgls
plot(log10(svl_av) ~ log10(rootmass_av), data = av.svlmass)
abline(pgls_svlmass)

#Likelihood plot for Pagel's lambda from the PGLS model of svl vs. the cuberoot of mass. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.svlmass <- pgls.profile(pgls_svlmass, "lambda")
plot(lambda.svlmass)

```

For the PGLS of mean snout-vent length vs. mean cube root of mass, Pagel's lambda was `r pgls.confint(pgls_svlmass, "lambda")$opt` with 95% confidence intervals of `r pgls.confint(pgls_svlmass, "lambda")$ci.val`, indicating that there is some phylogenetic signal in the residuals of our PGLS model fit. 

The diagnostic plots show that the data conformed to most model assumptions, though we do have species with studentized residuals exceeding ±3, which could disproportionately influence the regression. We thus iteratively re-ran the model with phylogenetic outliers removed. 

```{r pgls-svlmass-outliers, include = FALSE}

# refit pgls after removing extreme outlier/s observed in previous fit

#### Iteration 1 ####

#extract the phylogenetic residuals from the original model
phyres.svlmass1 <- residuals(pgls_svlmass, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.svlmass1 <- phyres.svlmass1/sqrt(var(phyres.svlmass1))[1] 

#match species names with standardized residuals
rownames(stphyres.svlmass1)<-rownames(pgls_svlmass$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.svlmass1)[(abs(stphyres.svlmass1)>3)]
 
#remove these two species from data/tree
svlmass.comp_1 <- svlmass.comp[-which(abs(stphyres.svlmass1)>3), ]

#re-run pgls analysis (without the outliers)
pgls_svlmass_1 <- pgls(log10(svl_av) ~ log10(rootmass_av), 
               data = svlmass.comp_1, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_svlmass_1, main = "PGLS model for svl vs. mass (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_svlmass_1 , "lambda"))

#print model output 
summary(pgls_svlmass_1)

#### Iteration 2 ####

#extract the phylogenetic residuals from the original model
phyres.svlmass2 <- residuals(pgls_svlmass_1, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.svlmass2 <- phyres.svlmass2/sqrt(var(phyres.svlmass2))[1] 

#match species names with standardized residuals
rownames(stphyres.svlmass2)<-rownames(pgls_svlmass_1$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.svlmass2)[(abs(stphyres.svlmass2)>3)]
 
#remove these two species from data/tree
svlmass.comp_2 <- svlmass.comp_1[-which(abs(stphyres.svlmass2)>3), ]

#re-run pgls analysis (without the outliers)
pgls_svlmass_2 <- pgls(log10(svl_av) ~ log10(rootmass_av), 
               data = svlmass.comp_2, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_svlmass_2, main = "PGLS model for svl vs. mass (no outliers 2)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_svlmass_2 , "lambda"))

#print model output 
summary(pgls_svlmass_2)

#### Iteration 3 ####

#extract the phylogenetic residuals from the original model
phyres.svlmass3 <- residuals(pgls_svlmass_2, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.svlmass3 <- phyres.svlmass3/sqrt(var(phyres.svlmass3))[1] 

#match species names with standardized residuals
rownames(stphyres.svlmass3)<-rownames(pgls_svlmass_2$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.svlmass3)[(abs(stphyres.svlmass3)>3)]
 
### No remaining phylogenetic outliers ###

```

### PGLS of log corneal diameter vs. log cube root of mass

```{r pgls-cdmass}

#fit PGLS model for log cornea diameter vs. log cube root of mass using 'caper' package

#### check for best model of correlation structure ####

#Make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(frog.tree$tip.label, as.character(av.cdmass$mod_tiplabel))

#Drop unwanted tips from phylogeny
tree.edit <- drop.tip(phy = frog.tree, tip = drops) 

#Make row names of the species the phylogeny tip labels
rownames(av.cdmass) <- av.cdmass$mod_tiplabel

#Reorder species data to match phylogeny order
data <-ReorderData(tree.edit, data, taxa.names="row names")

#test whether BM or OU model is better fit for PGLS using the nlme package and AIC comparison

#Expected covariance under a pure Brownian model (Felsenstein 1985,	Martins and Hansen 1997)
pglsBM <- gls(log10(cornea_av) ~ log10(rootmass_av), cor = corBrownian(phy = tree.edit), data = av.cdmass, method = "ML")

#Martins and Hansen's (1997) covariance structure (OU)
pglsOU<-gls(log10(cornea_av) ~ log10(rootmass_av), cor = corMartins(1, phy = tree.edit, fixed = F), data = av.cdmass, method = "ML")

#Pagel's “lambda” Correlation Structure: The correlation structure from the present model is derived from the Brownian motion model by multiplying the off-diagonal elements (i.e., the covariances) by lambda. The variances are thus the same than for a Brownian motion model.
pglsPagel<-gls(log10(cornea_av) ~ log10(rootmass_av), cor = corPagel(1, phy = tree.edit, fixed = F), data = av.cdmass, method = "ML")

anova(pglsBM, pglsOU, pglsPagel)

#### prep data ####

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.cdmass, data.names = av.cdmass$mod_tiplabel)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
cdmass.comp <- comparative.data(phy = frog.tree, data = av.cdmass, 
                            names.col = mod_tiplabel, vcv = TRUE, 
                            na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
cdmass.comp$dropped$tips #phylogeny
cdmass.comp$dropped$unmatched.rows #dataset

#### run PGLS model ####

#for cornea diameter vs. cube root of mass using the Maximum Liklihood estimate of lambda
pgls_cdmass <- pgls(log10(cornea_av) ~ log10(rootmass_av), 
               data = cdmass.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#### check model assumptions ####

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdmass, main = "PGLS model for mean cornea diameter vs. svl")
par(mfrow = c(1, 1))

#### model outputs ####

#print model output 
summary(pgls_cdmass)

#quick plot of pgls
plot(log10(cornea_av) ~ log10(rootmass_av), data = av.cdmass)
abline(pgls_cdmass)

#Likelihood plot for Pagel's lambda from the PGLS model of eye diameter vs. the cuberoot of mass. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.cdmass <- pgls.profile(pgls_cdmass, "lambda")
plot(lambda.cdmass)

```

For the PGLS of log cornea diameter vs. log cube root of mass, Pagel's lambda was `r pgls.confint(pgls_cdmass, "lambda")$opt` with 95% confidence intervals of `r pgls.confint(pgls_cdmass, "lambda")$ci.val`, indicating that there is high phylogenetic signal in the residuals of our PGLS model fit.

The diagnostic plots show that the data conformed to most model assumptions, though we do have species with studentized residuals exceeding ±3, which could disproportionately influence the regression. We thus iteratively re-ran the model with phylogenetic outliers removed. 

```{r pgls-cdmass-outliers, include = FALSE}

# refit pgls after removing extreme outlier/s observed in previous fit

#### Iteration 1 ####

#extract the phylogenetic residuals from the original model
phyres.cdmass1 <- residuals(pgls_cdmass, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdmass1 <- phyres.cdmass1/sqrt(var(phyres.cdmass1))[1] 

#match species names with standardized residuals
rownames(stphyres.cdmass1)<-rownames(pgls_cdmass$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdmass1)[(abs(stphyres.cdmass1)>3)]
 
#remove these two species from data/tree
cdmass.comp_1 <- cdmass.comp[-which(abs(stphyres.cdmass1)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdmass_1 <- pgls(log10(cornea_av) ~ log10(rootmass_av), 
               data = cdmass.comp_1, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdmass_1, main = "PGLS model for cd vs. mass (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdmass_1 , "lambda"))

#print model output 
summary(pgls_cdmass_1)

#### Iteration 2 ####

#extract the phylogenetic residuals from the original model
phyres.cdmass2 <- residuals(pgls_cdmass_1, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdmass2 <- phyres.cdmass2/sqrt(var(phyres.cdmass2))[1] 

#match species names with standardized residuals
rownames(stphyres.cdmass2)<-rownames(pgls_cdmass_1$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdmass2)[(abs(stphyres.cdmass2)>3)]
 
#remove these two species from data/tree
cdmass.comp_2 <- cdmass.comp_1[-which(abs(stphyres.cdmass2)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdmass_2 <- pgls(log10(cornea_av) ~ log10(rootmass_av), 
               data = cdmass.comp_2, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdmass_2, main = "PGLS model for cd vs. mass (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdmass_2 , "lambda"))

#print model output 
summary(pgls_cdmass_2)

#### Iteration 3 ####

#extract the phylogenetic residuals from the original model
phyres.cdmass3 <- residuals(pgls_cdmass_2, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdmass3 <- phyres.cdmass3/sqrt(var(phyres.cdmass3))[1] 

#match species names with standardized residuals
rownames(stphyres.cdmass3)<-rownames(pgls_cdmass_2$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdmass3)[(abs(stphyres.cdmass3)>3)]
 
#remove these two species from data/tree
cdmass.comp_3 <- cdmass.comp_2[-which(abs(stphyres.cdmass3)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdmass_3 <- pgls(log10(cornea_av) ~ log10(rootmass_av), 
               data = cdmass.comp_3, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdmass_3, main = "PGLS model for cd vs. mass (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdmass_3 , "lambda"))

#print model output 
summary(pgls_cdmass_3)

#### Iteration 4 ####

#extract the phylogenetic residuals from the original model
phyres.cdmass4 <- residuals(pgls_cdmass_3, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdmass4 <- phyres.cdmass4/sqrt(var(phyres.cdmass4))[1] 

#match species names with standardized residuals
rownames(stphyres.cdmass4)<-rownames(pgls_cdmass_3$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdmass4)[(abs(stphyres.cdmass4)>3)]
 
#remove these two species from data/tree
cdmass.comp_4 <- cdmass.comp_3[-which(abs(stphyres.cdmass4)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdmass_4 <- pgls(log10(cornea_av) ~ log10(rootmass_av), 
               data = cdmass.comp_4, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdmass_4, main = "PGLS model for cd vs. mass (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdmass_4 , "lambda"))

#print model output 
summary(pgls_cdmass_4)

#### Iteration 5 ####

#extract the phylogenetic residuals from the original model
phyres.cdmass5 <- residuals(pgls_cdmass_4, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdmass5 <- phyres.cdmass5/sqrt(var(phyres.cdmass5))[1] 

#match species names with standardized residuals
rownames(stphyres.cdmass5)<-rownames(pgls_cdmass_4$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdmass5)[(abs(stphyres.cdmass5)>3)]
 
#remove these two species from data/tree
cdmass.comp_5 <- cdmass.comp_4[-which(abs(stphyres.cdmass5)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdmass_5 <- pgls(log10(cornea_av) ~ log10(rootmass_av), 
               data = cdmass.comp_5, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdmass_5, main = "PGLS model for cd vs. mass (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdmass_5 , "lambda"))

#print model output 
summary(pgls_cdmass_5)

#### Iteration 6 ####

#extract the phylogenetic residuals from the original model
phyres.cdmass6 <- residuals(pgls_cdmass_5, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdmass6 <- phyres.cdmass6/sqrt(var(phyres.cdmass6))[1] 

#match species names with standardized residuals
rownames(stphyres.cdmass6)<-rownames(pgls_cdmass_5$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdmass6)[(abs(stphyres.cdmass6)>3)]
 
#remove these two species from data/tree
cdmass.comp_6 <- cdmass.comp_5[-which(abs(stphyres.cdmass6)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdmass_6 <- pgls(log10(cornea_av) ~ log10(rootmass_av), 
               data = cdmass.comp_6, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdmass_6, main = "PGLS model for cd vs. mass (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdmass_6 , "lambda"))

#print model output 
summary(pgls_cdmass_6)

#### Iteration 7 ####

#extract the phylogenetic residuals from the original model
phyres.cdmass7 <- residuals(pgls_cdmass_6, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdmass7 <- phyres.cdmass7/sqrt(var(phyres.cdmass7))[1] 

#match species names with standardized residuals
rownames(stphyres.cdmass7)<-rownames(pgls_cdmass_6$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdmass7)[(abs(stphyres.cdmass7)>3)]
 
#remove these two species from data/tree
cdmass.comp_7 <- cdmass.comp_6[-which(abs(stphyres.cdmass7)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdmass_7 <- pgls(log10(cornea_av) ~ log10(rootmass_av), 
               data = cdmass.comp_7, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdmass_7, main = "PGLS model for cd vs. mass (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdmass_7 , "lambda"))

#print model output 
summary(pgls_cdmass_7)

#### Iteration 8 ####

#extract the phylogenetic residuals from the original model
phyres.cdmass8 <- residuals(pgls_cdmass_7, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdmass8 <- phyres.cdmass8/sqrt(var(phyres.cdmass8))[1] 

#match species names with standardized residuals
rownames(stphyres.cdmass8)<-rownames(pgls_cdmass_7$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdmass8)[(abs(stphyres.cdmass8)>3)]
 
#remove these two species from data/tree
cdmass.comp_8 <- cdmass.comp_7[-which(abs(stphyres.cdmass8)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdmass_8 <- pgls(log10(cornea_av) ~ log10(rootmass_av), 
               data = cdmass.comp_8, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdmass_8, main = "PGLS model for cd vs. mass (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdmass_8 , "lambda"))

#print model output 
summary(pgls_cdmass_8)

#### Iteration 9 ####

#extract the phylogenetic residuals from the original model
phyres.cdmass9 <- residuals(pgls_cdmass_8, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdmass9 <- phyres.cdmass9/sqrt(var(phyres.cdmass9))[1] 

#match species names with standardized residuals
rownames(stphyres.cdmass9)<-rownames(pgls_cdmass_8$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdmass9)[(abs(stphyres.cdmass9)>3)]
  
### No remaining phylogenetic outliers ###
```


### PGLS of log cornea diameter vs. log snout-vent length 

```{r pgls-cdsvl}

#fit PGLS model for log cornea diameter vs. log SVL using 'caper' package

#### check for best model of correlation structure ####

#Make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(frog.tree$tip.label, as.character(av.cdsvl$mod_tiplabel))

#Drop unwanted tips from phylogeny
tree.edit <- drop.tip(phy = frog.tree, tip = drops) 

#Make row names of the species the phylogeny tip labels
rownames(av.cdsvl) <- av.cdsvl$mod_tiplabel

#Reorder species data to match phylogeny order
dat <-ReorderData(tree.edit, av.cdsvl, taxa.names="row names")

#test whether BM or OU model is better fit for PGLS using the nlme package and AIC comparison

#Expected covariance under a pure Brownian model (Felsenstein 1985,	Martins and Hansen 1997)
pglsBM <- gls(log10(cornea_av) ~ log10(svl_av), cor = corBrownian(phy = tree.edit), data = dat, method = "ML")

#Martins and Hansen's (1997) covariance structure (OU)
pglsOU<-gls(log10(cornea_av) ~ log10(svl_av), cor = corMartins(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

#Pagel's “lambda” Correlation Structure: The correlation structure from the present model is derived from the Brownian motion model by multiplying the off-diagonal elements (i.e., the covariances) by lambda. The variances are thus the same than for a Brownian motion model.
pglsPagel<-gls(log10(cornea_av) ~ log10(svl_av), cor = corPagel(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

anova(pglsBM, pglsOU, pglsPagel)

#### prep data ####

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.cdsvl, data.names = av.cdsvl$mod_tiplabel)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
cdsvl.comp <- comparative.data(phy = frog.tree, data = av.cdsvl, 
                            names.col = mod_tiplabel, vcv = TRUE, 
                            na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
cdsvl.comp$dropped$tips #phylogeny
cdsvl.comp$dropped$unmatched.rows #dataset

#### run PGLS model ####

#for log cornea diameter vs.log svl using the Maximum Liklihood estimate of lambda
pgls_cdsvl <- pgls(log10(cornea_av) ~ log10(svl_av), 
               data = cdsvl.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#### check model assumptions ####

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdsvl, main = "PGLS model for mean cornea diameter vs. svl")
par(mfrow = c(1, 1))

#### model outputs ####

#print model output 
summary(pgls_cdsvl)

#quick plot of pgls
plot(log10(cornea_av) ~ log10(svl_av), data = av.cdsvl)
abline(pgls_cdsvl)

#Likelihood plot for Pagel's lambda from the PGLS model of cornea diameter vs. SVL. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.cdsvl <- pgls.profile(pgls_cdsvl, "lambda")
plot(lambda.cdsvl)

```

For the PGLS of log mean cornea diameter vs. log snout-vent length, Pagel's lambda was `r pgls.confint(pgls_cdsvl, "lambda")$opt` with 95% confidence intervals of `r pgls.confint(pgls_cdsvl, "lambda")$ci.val`, indicating that there is high phylogenetic signal in the residuals of our PGLS model fit.

The diagnostic plots show that the data conformed to most model assumptions, though we do have species with studentized residuals exceeding ±3, which could disproportionately influence the regression. To check whether this is happening, we re-ran the model iteratively with phylogenetic outliers eliminated. 

```{r pgls-cdsvl-outliers, include = FALSE}

# refit pgls after removing phylogenetic outlier/s observed in previous fit

#### Iteration 1 ####

#extract the phylogenetic residuals from the original model
phyres.cdsvl1 <- residuals(pgls_cdsvl, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdsvl1 <- phyres.cdsvl1/sqrt(var(phyres.cdsvl1))[1] 

#match species names with standardized residuals
rownames(stphyres.cdsvl1)<-rownames(pgls_cdsvl$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdsvl1)[(abs(stphyres.cdsvl1)>3)]
 
#remove these two species from data/tree
cdsvl.comp_1 <- cdsvl.comp[-which(abs(stphyres.cdsvl1)>3), ]

#re-run pgls analisys (without the outliers)
pgls_cdsvl_1 <- pgls(log10(cornea_av) ~ log10(svl_av), 
               data = cdsvl.comp_1, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdsvl_1, main = "PGLS model for cornea v svl (no outliers 1)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdsvl_1 , "lambda"))

#print model output 
summary(pgls_cdsvl_1)

#### Iteration 2 ####

#extract the phylogenetic residuals from the original model
phyres.cdsvl2 <- residuals(pgls_cdsvl_1, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdsvl2 <- phyres.cdsvl2/sqrt(var(phyres.cdsvl2))[1] 

#match species names with standardized residuals
rownames(stphyres.cdsvl2)<-rownames(pgls_cdsvl_1$residuals) #matches the residuals up with the species names

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdsvl2)[(abs(stphyres.cdsvl2)>3)]
 
#remove this species from data/tree
cdsvl.comp_2 <- cdsvl.comp_1[-which(abs(stphyres.cdsvl2)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdsvl_2 <- pgls(log10(cornea_av) ~ log10(svl_av), 
               data = cdsvl.comp_2, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdsvl_2, main = "PGLS model for cornea v svl (no outliers 2)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdsvl_2 , "lambda"))

#print model output 
summary(pgls_cdsvl_2)

#### Iteration 3 ####

#extract the phylogenetic residuals from the original model
phyres.cdsvl3 <- residuals(pgls_cdsvl_2, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdsvl3 <- phyres.cdsvl3/sqrt(var(phyres.cdsvl3))[1] 

#match species names with standardized residuals
rownames(stphyres.cdsvl3)<-rownames(pgls_cdsvl_2$residuals) #matches the residuals up with the species names

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdsvl3)[(abs(stphyres.cdsvl3)>3)]
 
#remove this species from data/tree
cdsvl.comp_3 <- cdsvl.comp_2[-which(abs(stphyres.cdsvl3)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdsvl_3 <- pgls(log10(cornea_av) ~ log10(svl_av), 
               data = cdsvl.comp_3, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdsvl_3, main = "PGLS model for cornea v svl (no outliers 3)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdsvl_3 , "lambda"))

#print model output 
summary(pgls_cdsvl_3)

#### Iteration 4 ####

#extract the phylogenetic residuals from the original model
phyres.cdsvl4 <- residuals(pgls_cdsvl_3, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdsvl4 <- phyres.cdsvl4/sqrt(var(phyres.cdsvl4))[1] 

#match species names with standardized residuals
rownames(stphyres.cdsvl4)<-rownames(pgls_cdsvl_3$residuals) #matches the residuals up with the species names

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdsvl4)[(abs(stphyres.cdsvl4)>3)]
 
#remove these two species from data/tree
cdsvl.comp_4 <- cdsvl.comp_3[-which(abs(stphyres.cdsvl4)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdsvl_4 <- pgls(log10(cornea_av) ~ log10(svl_av), 
               data = cdsvl.comp_4, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdsvl_4, main = "PGLS model for cornea v svl (no outliers 4)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdsvl_4 , "lambda"))

#print model output 
summary(pgls_cdsvl_4)

#### Iteration 5 ####

#extract the phylogenetic residuals from the original model
phyres.cdsvl5 <- residuals(pgls_cdsvl_4, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdsvl5 <- phyres.cdsvl5/sqrt(var(phyres.cdsvl5))[1] 

#match species names with standardized residuals
rownames(stphyres.cdsvl5)<-rownames(pgls_cdsvl_4$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdsvl5)[(abs(stphyres.cdsvl5)>3)]
 
#remove these two species from data/tree
cdsvl.comp_5 <- cdsvl.comp_4[-which(abs(stphyres.cdsvl5)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdsvl_5 <- pgls(log10(cornea_av) ~ log10(svl_av), 
               data = cdsvl.comp_5, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdsvl_5, main = "PGLS model for cornea v svl (no outliers 5)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdsvl_5 , "lambda"))

#print model output 
summary(pgls_cdsvl_5)

#### Iteration 6 ####

#extract the phylogenetic residuals from the original model
phyres.cdsvl6 <- residuals(pgls_cdsvl_5, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdsvl6 <- phyres.cdsvl6/sqrt(var(phyres.cdsvl6))[1] 

#match species names with standardized residuals
rownames(stphyres.cdsvl6)<-rownames(pgls_cdsvl_5$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdsvl6)[(abs(stphyres.cdsvl6)>3)]
 
#remove these two species from data/tree
cdsvl.comp_6 <- cdsvl.comp_5[-which(abs(stphyres.cdsvl6)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdsvl_6 <- pgls(log10(cornea_av) ~ log10(svl_av), 
               data = cdsvl.comp_6, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdsvl_6, main = "PGLS model for cornea v svl (no outliers 5)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdsvl_6 , "lambda"))

#print model output 
summary(pgls_cdsvl_6)

#### Iteration 7 ####

#extract the phylogenetic residuals from the original model
phyres.cdsvl7 <- residuals(pgls_cdsvl_6, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdsvl7 <- phyres.cdsvl7/sqrt(var(phyres.cdsvl7))[1] 

#match species names with standardized residuals
rownames(stphyres.cdsvl7)<-rownames(pgls_cdsvl_6$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdsvl7)[(abs(stphyres.cdsvl7)>3)]
 
#remove these two species from data/tree
cdsvl.comp_7 <- cdsvl.comp_6[-which(abs(stphyres.cdsvl7)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdsvl_7 <- pgls(log10(cornea_av) ~ log10(svl_av), 
               data = cdsvl.comp_7, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdsvl_7, main = "PGLS model for cornea v svl (no outliers 7)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdsvl_7 , "lambda"))

#print model output 
summary(pgls_cdsvl_7)

#### Iteration 8 ####

#extract the phylogenetic residuals from the original model
phyres.cdsvl8 <- residuals(pgls_cdsvl_7, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdsvl8 <- phyres.cdsvl8/sqrt(var(phyres.cdsvl8))[1] 

#match species names with standardized residuals
rownames(stphyres.cdsvl8)<-rownames(pgls_cdsvl_7$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdsvl8)[(abs(stphyres.cdsvl8)>3)]
 
#remove these two species from data/tree
cdsvl.comp_8 <- cdsvl.comp_7[-which(abs(stphyres.cdsvl8)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdsvl_8 <- pgls(log10(cornea_av) ~ log10(svl_av), 
               data = cdsvl.comp_8, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdsvl_8, main = "PGLS model for cornea v svl (no outliers 7)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdsvl_8 , "lambda"))

#print model output 
summary(pgls_cdsvl_8)

#### Iteration 9 ####

#extract the phylogenetic residuals from the original model
phyres.cdsvl9 <- residuals(pgls_cdsvl_8, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdsvl9 <- phyres.cdsvl9/sqrt(var(phyres.cdsvl9))[1] 

#match species names with standardized residuals
rownames(stphyres.cdsvl9)<-rownames(pgls_cdsvl_8$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdsvl9)[(abs(stphyres.cdsvl9)>3)]
 
#remove these two species from data/tree
cdsvl.comp_9 <- cdsvl.comp_8[-which(abs(stphyres.cdsvl9)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdsvl_9 <- pgls(log10(cornea_av) ~ log10(svl_av), 
               data = cdsvl.comp_9, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdsvl_9, main = "PGLS model for cornea v svl (no outliers 9)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdsvl_9 , "lambda"))

#print model output 
summary(pgls_cdsvl_9)

#### Iteration 10 ####

#extract the phylogenetic residuals from the original model
phyres.cdsvl10 <- residuals(pgls_cdsvl_9, phylo = TRUE)

#standardize residuals by dividing by the square root of their variance
stphyres.cdsvl10 <- phyres.cdsvl10/sqrt(var(phyres.cdsvl10))[1] 

#match species names with standardized residuals
rownames(stphyres.cdsvl10)<-rownames(pgls_cdsvl_9$residuals) 

#pull the names of the outlier species (where standardized res beyond ±3)
rownames(stphyres.cdsvl10)[(abs(stphyres.cdsvl10)>3)]
 
#remove these two species from data/tree
cdsvl.comp_10 <- cdsvl.comp_9[-which(abs(stphyres.cdsvl10)>3), ]

#re-run pgls analysis (without the outliers)
pgls_cdsvl_10 <- pgls(log10(cornea_av) ~ log10(svl_av), 
               data = cdsvl.comp_10, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_cdsvl_10, main = "PGLS model for eye v svl (no outliers 10)")
par(mfrow = c(1, 1))

#lambda
plot(pgls.profile(pgls_cdsvl_10 , "lambda"))

#print model output 
summary(pgls_cdsvl_10)

```

Again, as iterative removal of phylogenetic outliers did not qualitatively change our results, we use the full species dataset for our PGLS of log cornea diameter vs. log snout-vent length among anurans. 

### Scaling via OLS and SMA for comparison

Because many previous papers on eye scaling and allometry have not employed phylogenetic regreressions, we also fit OLS and SMA models to our data for comparison. Our 4 SMA fits of eye size vs. body size (eye vs. svl, eye vs. mass, cornea vs. svl, cornea vs. mass) showed large negative residuals with high deviation from normality, so we used the option "robust = TRUE" in these 4 SMA fits. This fits a robust SMA using Huber's M estimation, which is less sensitive to outliers.

```{r ols-sma}

##### eye diameter vs. the cube root of mass #####

# OLS fit of eye diameter vs. cube root of mass
ols_edmass_av <- lm(formula = log10(eye_av) ~ log10(rootmass_av),
                   data = av.edmass)

# plot fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(ols_edmass_av)
par(mfrow = c(1,1)) #set plotting window back to one plot

# model summary
summary(ols_edmass_av)

# SMA fit of mean eye diameter vs. the cube root of mass
sma_edmass_av <- sma(formula = eye_av ~ rootmass_av, 
                     slope.test = 1, #tests for significant difference from isometry
                     robust = TRUE, #makes estimation robust to outliers
                     data = av.edmass, 
                     log = "xy",
                     method="SMA", 
                     alpha=0.05)

#plot fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(sma_edmass_av, main = "eye ~ mass")
plot(sma_edmass_av, which = "residual") #check assumptions of linearity and equal variance at all fitted values
plot(sma_edmass_av, which = "qq") #check assumption of normally distributed residuals
par(mfrow = c(1,1)) #set plotting window back to one plot

# model summary
summary(sma_edmass_av)


##### eye diameter vs. snout vent length #####

# OLS fit of eye diameter vs. SVL
ols_edsvl_av <- lm(formula = log10(eye_av) ~ log10(svl_av),
                   data = av.edsvl)

# plot fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(ols_edsvl_av)
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(ols_edsvl_av)

# SMA fit of eye diameter vs. SVL
sma_edsvl_av <- sma(formula = eye_av ~ svl_av, 
                    slope.test = 1, #tests for significant difference from isometry
                    robust = TRUE, #makes estimation robust to outliers
                    data = av.edsvl, 
                    log = "xy",
                    method = "SMA", 
                    alpha= 0.05)

# plot SMA fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(sma_edsvl_av, main = "eye ~ SVL")
plot(sma_edsvl_av, which = "residual") #check assumptions of linearity and equal variance at all fitted values
plot(sma_edsvl_av, which = "qq") #check assumption of normally distributed residuals
par(mfrow = c(1,1)) #set plotting window back to one plot

# model summary
summary(sma_edsvl_av)

##### cornea diameter vs. eye diameter #####

# OLS fit of cornea diameter vs. eye diameter
ols_edcd_av <- lm(formula = log10(cornea_av) ~ log10(eye_av),
                   data = av.edcd)

# plot fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(ols_edcd_av)
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(ols_edcd_av)

# SMA fit of mean cornea diameter vs. mean eye diameter
sma_edcd_av <- sma(formula = cornea_av ~ eye_av, 
                   data = av.edcd,
                   log = "xy",
                   method = "SMA", 
                   alpha = 0.05)

#plot fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(sma_edcd_av, main = "eye ~ cornea")
plot(sma_edcd_av, which = "residual") #check assumptions of linearity and equal variance at all fitted values
plot(sma_edcd_av, which = "qq") #check assumption of normally distributed residuals
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(sma_edcd_av)

##### SVL vs. cube root of mass #####

# OLS fit of SVL vs. mass
ols_svlmass_av <- lm(formula = log10(svl_av) ~ log10(rootmass_av),
                   data = av.svlmass)

# plot fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(ols_svlmass_av)
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(ols_svlmass_av)

# SMA fit of cube root of mass vs. SVL
sma_svlmass_av <- sma(formula = svl_av ~ rootmass_av,
                   data = av.svlmass, 
                   slope.test = 1, #tests for significant difference from isometry
                   log = "xy",
                   method = "SMA", 
                   alpha = 0.05)

#plot fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(sma_svlmass_av, main = "mass ~ SVL")
plot(sma_svlmass_av, which = "residual") #check assumptions of linearity and equal variance at all fitted values
plot(sma_svlmass_av, which = "qq") #check assumption of normally distributed residuals
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(sma_svlmass_av)

##### cornea diameter vs. mass #####

# OLS fit of cornea diameter vs. mass
ols_cdmass_av <- lm(formula = log10(cornea_av) ~ log10(rootmass_av),
                   data = av.cdmass)

# plot fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(ols_cdmass_av)
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(ols_cdmass_av)

# SMA for mean cornea diameter vs. cube root of mass
sma_cdmass_av <- sma(formula = cornea_av ~ rootmass_av,
                     robust = TRUE,
                     data = av.cdmass, 
                     log = "xy",
                     method = "SMA", 
                     alpha = 0.05)

#plot fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(sma_cdmass_av, main = "cornea ~ mass")
plot(sma_cdmass_av, which = "residual") #check assumptions of linearity and equal variance at all fitted values
plot(sma_cdmass_av, which = "qq") #check assumption of normally distributed residuals
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(sma_cdmass_av)

##### cornea diameter vs. SVL #####

# OLS fit of cornea diameter vs. SVL
ols_cdsvl_av <- lm(formula = log10(cornea_av) ~ log10(svl_av),
                   data = av.cdsvl)

# plot fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(ols_cdsvl_av)
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(ols_cdsvl_av)

# SMA of mean cornea diameter vs. snout vent length
sma_cdsvl_av <- sma(formula = cornea_av ~ svl_av,
                    robust = TRUE, #makes SMA robust to outliers
                    data = av.cdsvl,
                    log = "xy",
                    method = "SMA", 
                    alpha = 0.05)

#plot fit, residuals, qq
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(sma_cdsvl_av, main = "cornea ~ SVL")
plot(sma_cdsvl_av, which = "residual") #check assumptions of linearity and equal variance at all fitted values
plot(sma_cdsvl_av, which = "qq") #check assumption of normally distributed residuals
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(sma_cdsvl_av)

```

We then plotted the data (species averages) and PGLS fits (with OLS and SMA fits for comparison).

```{r pgls-plots, fig.height=4, fig.width=8}

#define a colorblind-friendly vector of colors for adult habitat
col_hab <- c("Aquatic" = "#0072B2",
              "Fossorial" = "#D55E00",
              "Ground-dwelling" = "#E69F00",
              "Scansorial" = "#009E73",
              "Semiaquatic" = "#56B4E9",
              "Subfossorial" = "#CC79A7")


#### plot for eye diameter vs. cube root of mass ####
plot_pgls.edmass <- ggplot(av.edmass, aes(x = rootmass_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.9, aes(color = Adult_habitat)) +
scale_color_manual(values = col_hab, name = "Adult habitat") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Eye diameter (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(pgls_edmass)[[2]], intercept = coef(pgls_edmass)[[1]], linetype = "solid") + #plots PGLS fit with solid line
geom_abline(slope = coef(sma_edmass_av)[[2]], intercept = coef(sma_edmass_av)[[1]], linetype = "dotted", col = "black") + #plots SMA fit with dotted line
   geom_abline(slope = coef(ols_edmass_av)[[2]], intercept = coef(ols_edmass_av)[[1]], linetype = "dashed", col = "black") #plots OLS fit with dashed line

#interactive plot
ggplotly(plot_pgls.edmass)

#### plot for eye diameter vs. SVL ####
plot_pgls.edsvl <- ggplot(av.edsvl, aes(x = svl_av, y = eye_av, text = genus_species)) +
  geom_point(alpha = 0.9, aes(color = Adult_habitat)) +
  scale_color_manual(values = col_hab, name = "Adult habitat") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Eye diameter (mm)") +
  scale_x_log10(name = "Snout-vent length (mm)") +
  geom_abline(slope = coef(pgls_edsvl)[[2]], intercept = coef(pgls_edsvl)[[1]], linetype = "solid") + #plots PGLS fit with solid line
 geom_abline(slope = coef(sma_edsvl_av)[[2]], intercept = coef(sma_edsvl_av)[[1]], linetype = "dotted") + #plots SMA fit with dotted line
   geom_abline(slope = coef(ols_edsvl_av)[[2]], intercept = coef(ols_edsvl_av)[[1]], linetype = "dashed") #plots OLS fit with dashed line

#interactive plot
ggplotly(plot_pgls.edsvl)

#### plot for cornea diameter vs. eye diameter ####
plot_pgls.edcd <- ggplot(av.edcd, aes(x = eye_av, y = cornea_av, text = genus_species)) +
  geom_point(alpha = 0.9, aes(color = Adult_habitat)) +
  scale_color_manual(values = col_hab, name = "Adult habitat") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Cornea diameter (mm)") +
  scale_x_log10(name = "Eye diameter (mm)") +
  geom_abline(slope = coef(pgls_edcd)[[2]], intercept = coef(pgls_edcd)[[1]], linetype = "solid") + #plots PGLS fit with solid line
 geom_abline(slope = coef(sma_edcd_av)[[2]], intercept = coef(sma_edcd_av)[[1]], linetype = "dotted") + #plots SMA fit with dotted line
   geom_abline(slope = coef(ols_edcd_av)[[2]], intercept = coef(ols_edcd_av)[[1]], linetype = "dashed") #plots OLS fit with dashed line

#interactive plot
ggplotly(plot_pgls.edcd)

#### plot for svl vs. mass ####
plot_pgls.svlmass <- ggplot(av.svlmass, aes(y = svl_av, x = rootmass_av, text = genus_species)) +
  geom_point(alpha = 0.9, aes(color = Adult_habitat)) +
    scale_color_manual(values = col_hab, name = "Adult habitat") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Snout-vent length (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(pgls_svlmass)[[2]], intercept = coef(pgls_svlmass)[[1]], linetype = "solid", col = "black") + #plots PGLS fit with solid line
 geom_abline(slope = coef(sma_svlmass_av)[[2]], intercept = coef(sma_svlmass_av)[[1]], linetype = "dotted") + #plots SMA fit with dotted line
   geom_abline(slope = coef(ols_svlmass_av)[[2]], intercept = coef(ols_svlmass_av)[[1]], linetype = "dashed") #plots OLS fit with dashed line

#interactive plot
ggplotly(plot_pgls.svlmass)

#### plot for cornea diameter vs. cube root of mass ####
plot_pgls.cdmass <- ggplot(av.cdmass, aes(x = rootmass_av, y = cornea_av, text = genus_species)) +
 geom_point(alpha = 0.9, aes(color = Adult_habitat)) +
   scale_color_manual(values = col_hab, name = "Adult habitat") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Cornea diameter (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(pgls_cdmass)[[2]], intercept = coef(pgls_cdmass)[[1]], linetype = "solid") + #plots PGLS fit with solid line
geom_abline(slope = coef(sma_cdmass_av)[[2]], intercept = coef(sma_cdmass_av)[[1]], linetype = "dotted", col = "black") + #plots SMA fit with dotted line
   geom_abline(slope = coef(ols_cdmass_av)[[2]], intercept = coef(ols_cdmass_av)[[1]], linetype = "dashed", col = "black") #plots OLS fit with dashed line

#interactive plot
ggplotly(plot_pgls.cdmass)

#### plot for cornea diameter vs. SVL ####
plot_pgls.cdsvl <- ggplot(av.cdsvl, aes(x = svl_av, y = cornea_av, text = genus_species)) +
  geom_point(alpha = 0.9, aes(color = Adult_habitat)) +
  scale_color_manual(values = col_hab, name = "Adult habitat") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Cornea diameter (mm)") +
  scale_x_log10(name = "Snout-vent length (mm)") +
  geom_abline(slope = coef(pgls_cdsvl)[[2]], intercept = coef(pgls_cdsvl)[[1]], linetype = "solid") + #plots PGLS fit with solid line
 geom_abline(slope = coef(sma_cdsvl_av)[[2]], intercept = coef(sma_cdsvl_av)[[1]], linetype = "dotted") + #plots SMA fit with dotted line
   geom_abline(slope = coef(ols_cdsvl_av)[[2]], intercept = coef(ols_cdsvl_av)[[1]], linetype = "dashed") #plots OLS fit with dashed line

#interactive plot
ggplotly(plot_pgls.cdsvl)
```

We used three of these plots to construct Figure 2 of the main text. 

```{r figure2-pgls, fig.cap = "Log-log plots of data and PGLS (solid line) fits for averaged species data. OLS (dashed) and SMA (dotted) fits also showed for comparison. (A) eye diameter vs. the cube root of mass, (B) eye diameter vs. SVL, and (C) cornea diameter vs. eye diameter", fig.width=6, fig.height=2, include = FALSE}

#make panels 
fig.a <- plot_pgls.edmass +
theme(legend.position = "none")

fig.b <- plot_pgls.edsvl +
theme(legend.position = "none")

fig.c <- plot_pgls.edcd +
theme(legend.position = "none")

# extract legend from one figure
leg <- get_legend(plot_pgls.edsvl + theme(legend.position = "bottom"))

#use cowplot package to nicely plot the graphs together with a shared common legend.
plots <- plot_grid(fig.a, fig.b, fig.c, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("A", "B", "C"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 1) #number of columns in grids

#construct full figure with panels and legend
plot_grid(plots, leg, nrow=2, rel_heights = c(1, 0.2), align = 'vh')

pdf("../Manuscript/Figures/figure2-pgls.pdf", width=9, height=3)
plot_grid(plots, leg, nrow=2, rel_heights = c(1, 0.2), align = 'vh')
dev.off()
```

We also included plots of all scaling relationships as well as displayed our full (unaveraged) specimen dataset in a supplemental figure. 

```{r figure-supp-scaling}

#plot for eye diameter vs. cube root of mass #
edmass_supp <- plot_pgls.edmass  #start with pgls plot
edmass_supp$layers <- c(geom_point(data = frogs, aes(x = rootmass, y = eyemean), alpha = 0.25, col = "gray50", pch = 18), edmass_supp$layers)  #add full dataset behind averages layer

#plot for eye diameter vs. svl 
edsvl_supp <- plot_pgls.edsvl  #start with pgls plot
edsvl_supp$layers <- c(geom_point(data = frogs, aes(x = SVL_mm, y = eyemean), alpha = 0.25, col = "gray50", pch = 18), edsvl_supp$layers)  #add full dataset behind averages layer

#plot for cornea diameter vs. eye dianeter
edcd_supp <- plot_pgls.edcd  #start with pgls plot
edcd_supp$layers <- c(geom_point(data = frogs, aes(x = eyemean, y = cormean), alpha = 0.25, col = "gray50", pch = 18), edcd_supp$layers)  #add full dataset behind averages layer

#plot for svl vs. cube root of mass
svlmass_supp <- plot_pgls.svlmass  #start with pgls plot
svlmass_supp$layers <- c(geom_point(data = frogs, aes(x = rootmass, y = SVL_mm), alpha = 0.25, col = "gray50", pch = 18), svlmass_supp$layers)  #add full dataset behind averages layer

#plot for cornea diameter vs. cube root of mass
cdmass_supp <- plot_pgls.cdmass  #start with pgls plot
cdmass_supp$layers <- c(geom_point(data = frogs, aes(x = rootmass, y = cormean), alpha = 0.25, col = "gray50", pch = 18), cdmass_supp$layers)  #add full dataset behind averages layer

#plot for cornea diameter vs. svl
cdsvl_supp <- plot_pgls.cdsvl  #start with pgls plot
cdsvl_supp$layers <- c(geom_point(data = frogs, aes(x = SVL_mm, y = cormean), alpha = 0.25, col = "gray50", pch = 18), cdsvl_supp$layers)  #add full dataset behind averages layer

#make panels 
fig.a <- edmass_supp +
  theme(legend.position = "none")

fig.b <- edsvl_supp +
  theme(legend.position = "none")

fig.c <- edcd_supp +
  theme(legend.position = "none")

fig.d <- svlmass_supp +
  theme(legend.position = "none")

fig.e <- cdmass_supp +
  theme(legend.position = "none")

fig.f <- cdsvl_supp +
  theme(legend.position = "none")

# extract legend from one figure
leg <- get_legend(edmass_supp + theme(legend.position = "bottom"))

#use cowplot package to nicely plot the graphs together with a shared common legend.
plots <- plot_grid(fig.a, fig.b, fig.c, fig.d, fig.e, fig.f, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("A", "B", "C", "D", "E", "F"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 2) #number of columns in grids

#construct full figure with panels and legend
plot_grid(plots, leg, nrow=2, rel_heights = c(1, 0.1), align = 'vh')

pdf("../Manuscript/Figures/figure-supp-scaling.pdf", width=9, height=6)
plot_grid(plots, leg, nrow=2, rel_heights = c(1, 0.1), align = 'vh')
dev.off()
```

***

# Eye scaling in fresh specimens vs. museum preserved specimens

## Correlation between preserved specimen eye measurements and fresh specimen eye measurements

To test whether our use of external eye and corneal diameter measurements from muesum specimens were providing reasonable approximations, we compared our preserved museum specimen dataset to a small field-sampled dataset we collected on fresh frogs. 

```{r fresh-frogs}

# Fresh specimen data ------

#import data on fresh frog eye measurements to convert transverse to axial measurements
fresh.frogs <- data.frame(read.csv("../Data/fresh_data.csv", header = TRUE, na.strings=c("", "NA", " "))) #load raw data

#Number of species and individuals sampled 
fresh.counts <-ddply(fresh.frogs, .(fresh.frogs$Family, fresh.frogs$Genus, fresh.frogs$Species), nrow)
names(fresh.counts) <- c("Family", "Genus","Species","Sampled")

#create scrolling RMarkdown table of sampling
kable(fresh.counts[ , c("Family","Genus","Species","Sampled")], caption = "Species and sampling effort for morphological data from fresh specimens.") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top") %>%
  scroll_box(height = "500px")




# Axial diameter vs. transverse diameter -------

#linear model to predict axial diameter from external transverse diameter of eye
mod_edad <- lm(log10(AD_mm_dissected) ~ log10(ED_mm_insitu), fresh.frogs)

#model summary
summary(mod_edad)

#plot
plot_edad <- ggplot(fresh.frogs, aes(x = ED_mm_insitu, y = AD_mm_dissected, text = Genus)) +
  geom_point(alpha = 0.5) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_x_log10("External transverse diameter (mm)") +
  scale_y_log10("Dissected axial length (mm)") +
  geom_abline(slope = coef(mod_edad)[[2]], intercept = coef(mod_edad)[[1]], linetype = "solid", alpha = 1)  

#interactive plot
ggplotly(plot_edad)

# External transverse diameter vs. dissected transverse diameter ------------

#linear model to predict dissected transverse diameter from external transverse diameter of eye
mod_eddd <- lm(log10(ED_mm_dissected) ~ log10(ED_mm_insitu), fresh.frogs)

#model summary
summary(mod_eddd)

#plot
plot_eddd <- ggplot(fresh.frogs, aes(x = ED_mm_insitu, y = ED_mm_dissected, text = Genus)) +
  geom_point(alpha = 0.5) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_x_log10("External transverse diameter (mm)") +
  scale_y_log10("Dissected transverse diameter (mm)") +
  geom_abline(slope = coef(mod_eddd)[[2]], intercept = coef(mod_eddd)[[1]], linetype = "solid", alpha = 1)  

#interactive plot
ggplotly(plot_eddd)
```

## Eye scaling with body size in preserved vs. fresh specimens

Then we compared eye scaling in fresh frogs to eye scaling in preserved museum specimens. We did this testing for significant differences in SMA fits across our fresh specimen data vs. our preserved specimen data. 

```{r freshpres-comp}

# Comparison of eye scaling in preserved frogs vs. fresh frogs. 

#preserved dataset (species averages)
preserved <- frogs %>% 
  mutate_if(is.character, as.factor) %>% 
  filter(!(is.na(eyemean) | is.na(cormean) | is.na(SVL_mm) | is.na(rootmass))) %>% 
  mutate(group = as.factor("preserved specimens")) %>%
  group_by(Family, genus_species, group) %>%
  summarise(TD_external = mean(eyemean), 
            eyecomp = mean (eyemean), 
            cornea = mean(cormean),
            svl = mean(SVL_mm), 
            mass = mean(rootmass),
            n = n())
  
#fresh dataset (species averages)
fresh <- fresh.frogs %>%
  mutate(genus_species = as.factor(paste(fresh.frogs$Genus, fresh.frogs$Species, sep="_"))) %>%
  mutate(rootmass = (Mass_g)^(1/3), group = as.factor("fresh specimens")) %>%
  group_by(Family, genus_species, group) %>%
  summarise(TD_external = mean(ED_mm_insitu), 
            eyecomp = mean (ED_mm_dissected), 
            cornea = mean(CD_mm_dissected),
            svl = mean(SVL_mm), 
            mass = mean(rootmass),
            TD_dissected = mean(ED_mm_dissected),
            AD_dissected = mean(AD_mm_dissected),
            n = n())

#merged preserved and fresh datasets
fresh.pres <- merge(preserved, fresh, all = TRUE)

# External transverse diameter vs. cuberoot of mass --------

#Test whether SMA fits are different for fresh vs. preserved frog datasets
sma_exeye.mass <- sma(formula = TD_external ~ mass + group,
                      robust = TRUE,
                      data = fresh.pres,
                      log = "xy",
                      method = "SMA", 
                      alpha = 0.05)
#Model summary
summary(sma_exeye.mass)

#Quick plot
plot(sma_exeye.mass)

#Color palette
col_fp <- c("preserved specimens" = "goldenrod",
            "fresh specimens" = "black")
#Plot
plot_exeye.mass <- ggplot(fresh.pres, aes(x = mass, y = TD_external, text = genus_species)) +
  geom_point(aes(color = group), size=1, alpha = 0.5) +
  scale_color_manual(values = col_fp, breaks = c("fresh specimens", "preserved specimens"), labels = c("Fresh", "Preserved"), name = "Specimen type") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "External transverse eye diameter (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(sma_exeye.mass)[[2]], intercept = coef(sma_exeye.mass)[[1]], linetype = "dashed", color = col_fp, alpha = 1)   #plots SMA fits with dashed line

#Interactive plot
ggplotly(plot_exeye.mass)


# External/dissected transverse diameter vs. cuberoot of mass --------

#Test whether preserved external diameter vs. mass scales differently than fresh dissected diameter vs. mass
sma_diffeye.mass <- sma(formula = eyecomp ~ mass + group, 
                        robust = TRUE,
                        data = fresh.pres,
                        log = "xy",
                        method = "SMA", 
                        alpha = 0.05)
#Model summary
summary(sma_diffeye.mass)

#Quick plot
plot(sma_diffeye.mass)

#Plot
plot_compeye.mass <- ggplot(fresh.pres, aes(x = mass, y = eyecomp, text = genus_species)) +
  geom_point(aes(color = group), size=1, alpha = 0.5) +
  scale_color_manual(values = col_fp, breaks = c("fresh specimens", "preserved specimens"), labels = c("Fresh (dissected)", "Preserved (external)"), name = "Specimen type") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Transverse eye diameter (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(sma_diffeye.mass)[[2]], intercept = coef(sma_diffeye.mass)[[1]], linetype = "dashed", color = col_fp, alpha = 1)   #plots SMA fits with dashed line

#Interactive plot
ggplotly(plot_compeye.mass)

# External transverse diameter vs. SVL --------

#Test whether external transverse diameter vs. SVL scales differently in fresh and preserved specimens
sma_exeye.svl <- sma(formula = TD_external ~ svl * group,
                     robust = TRUE,
                     data = fresh.pres,
                     log = "xy",
                     method = "SMA", 
                     alpha = 0.05)

#Model summary
summary(sma_exeye.svl)

#Quick plot
plot(sma_exeye.svl)

#Plot
plot_exeye.svl <- ggplot(fresh.pres, aes(x = svl, y = TD_external, text = genus_species)) +
  geom_point(aes(color = group), size=1, alpha = 0.5) +
  scale_color_manual(values = col_fp, breaks = c("fresh specimens", "preserved specimens"), labels = c("Fresh", "Preserved"), name = "Specimen type") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "External transverse eye diameter (mm)") +
  scale_x_log10(name = "SVL (mm)") +
  geom_abline(slope = coef(sma_exeye.svl)[[2]], intercept = coef(sma_exeye.svl)[[1]], linetype = "dashed", color = col_fp, alpha = 1)   #plots SMA fits with dashed line

#Interactive plot
ggplotly(plot_exeye.svl)

# External/dissected transverse diameter vs. SVL --------

#Test whether preserved external diameter vs. SVL scales differently than fresh dissected diameter vs. SVL
sma_diffeye.svl <- sma(formula = eyecomp ~ svl + group, 
                       robust = TRUE,
                       data = fresh.pres,
                       log = "xy",
                       method = "SMA", 
                       alpha = 0.05)
#Model summary
summary(sma_diffeye.svl)

#Quick plot
plot(sma_diffeye.svl)

#Plot
plot_compeye.svl <- ggplot(fresh.pres, aes(x = svl, y = eyecomp, text = genus_species)) +
  geom_point(aes(color = group), size=1, alpha = 0.5) +
 scale_color_manual(values = col_fp, breaks = c("fresh specimens", "preserved specimens"), labels = c("Fresh (dissected)", "Preserved (external)"), name = "Specimen type") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Transverse eye diameter (mm)") +
  scale_x_log10(name = "SVL (mm)") +
  geom_abline(slope = coef(sma_diffeye.svl)[[2]], intercept = coef(sma_diffeye.svl)[[1]], linetype = "dashed", color = col_fp, alpha = 1)   #plots SMA fits with dashed line

```

Make a pdf figure of fresh frog scaling relationships for supplement. 

```{r fresh-fig}

#make panels 
fig.a <- plot_edad +
  theme(legend.position = "none", text = element_text(size=12))

fig.b <- plot_eddd +
  theme(legend.position = "none", text = element_text(size=12))

fig.c <- plot_exeye.mass +
  theme(legend.position = "none", text = element_text(size=12))

fig.d <- plot_exeye.svl +
  theme(legend.position = "none", text = element_text(size=12))

fig.e <- plot_compeye.mass +
  theme(legend.position = "none", text = element_text(size=12))

fig.f <- plot_compeye.svl +
  theme(legend.position = "none", text = element_text(size=12))

#plots
plots <- plot_grid(fig.a, fig.b, fig.c, fig.d, fig.e, fig.f, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("A", "B", "C", "D", "E", "F"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 3) #number of rows in gridss

# extract legends
leg.cd <- get_legend(plot_exeye.svl + theme(legend.position = "right"))
leg.ef <- get_legend(plot_compeye.svl + theme(legend.position = "right"))
legends <- plot_grid(NULL, leg.cd, leg.ef, align = 'vh', nrow = 3, ncol = 1)

#export pdf of figure
pdf("../Manuscript/Figures/figure-freshfrogs.pdf", width=10, height=12)
plot_grid(plots, legends, ncol = 2, rel_widths = c(1, 0.2), align = 'vh')
dev.off()
```

***

# Eye scaling comparison across vertebrates

After determinining how eyes scale with body size in frogs, we then wanted to compare to eye scaling in other vertebrates. Some papers that have examined eye scaling have looked eyes relative to body mass, and others relative to body length. Additionally, some use axial length and some transverse diameter of the eye. We assembled data on eye and body size across 1208 vertebrates from published literature and matched these to available phylogenies to account for non-indepenent evolutionary relationships. 

## Import published data on vertebrate eye and body size and phylogenetic relationships

```{r data-verts, fig.height = 50, fig.width = 10, include = FALSE}

#### Vertebrate relative eye size data ####

#Load eye size data assembled from literature
vert_morph <- data.frame(read.csv("../Data/other verts/vertebrate_eyesize.csv", header=TRUE, na.strings=c("", "NA", " ")))

#tidy/format data
vert_morph <- vert_morph %>%
  mutate(rootmass = (Body_mass_kg*1000)^(1/3)) %>%
  filter(!(is.na(VertLife_name)&is.na(TimeTree_tiplabel)&is.na(TimeTree_genus_sub))) %>%
  mutate_if(is.character, as.factor)


# Subset datasets by clade
mammals <- vert_morph[vert_morph$Group == "Mammals (Mammalia)", ]

birds <- vert_morph[vert_morph$Group == "Birds (Aves)", ]

squamates <- vert_morph[vert_morph$Group == "Reptiles (Squamata)", ]

sharks <- vert_morph[vert_morph$Group == "Sharks/rays (Elasmobranchii)", ]

fishes <- vert_morph[vert_morph$Group == "Fishes (Actinopterygii)", ]

# Find species means for analysis
mammal_avs <- mammals %>% 
  mutate_if(is.character, as.factor) %>% 
  group_by(VertLife_name) %>%
  summarise(AD_av = mean(Axial_diameter_mm),
            TD_av = mean(Trans_diameter_mm),
            CD_av = mean(Cornea_diameter_mm),
            mass_av = mean(rootmass), 
            svl_av = mean(SVL_mm),
            n = n())

birds_avs <- birds %>% 
  mutate_if(is.character, as.factor) %>% 
  group_by(VertLife_name) %>%
  summarise(AD_av = mean(Axial_diameter_mm),
            TD_av = mean(Trans_diameter_mm),
            CD_av = mean(Cornea_diameter_mm),
            mass_av = mean(rootmass), 
            svl_av = mean(SVL_mm),
            n = n())

squamate_avs <- squamates %>% 
  mutate_if(is.character, as.factor) %>% 
  group_by(VertLife_name, Subgroup) %>%
  summarise(AD_av = mean(Axial_diameter_mm),
            TD_av = mean(Trans_diameter_mm),
            CD_av = mean(Cornea_diameter_mm),
            mass_av = mean(rootmass), 
            svl_av = mean(SVL_mm),
            n = n())

shark_avs <- sharks %>% 
  mutate_if(is.character, as.factor) %>% 
  group_by(VertLife_name) %>%
  summarise(AD_av = mean(Axial_diameter_mm),
            TD_av = mean(Trans_diameter_mm),
            CD_av = mean(Cornea_diameter_mm),
            mass_av = mean(rootmass), 
            svl_av = mean(SVL_mm),
            n = n())

fishes_avs <- fishes %>% 
  mutate(tips = coalesce(TimeTree_tiplabel, TimeTree_genus_sub)) %>% 
  mutate_if(is.character, as.factor) %>% 
  group_by(tips) %>%
  summarise(AD_av = mean(Axial_diameter_mm),
            TD_av = mean(Trans_diameter_mm),
            CD_av = mean(Cornea_diameter_mm),
            mass_av = mean(rootmass), 
            svl_av = mean(SVL_mm),
            n = n())


#### Vertebrate group phylogenies ####

#Import phylogenies (downloaded from VertLife and TimeTree)

# Mammals-----

#read tree
mammal.phy <- read.tree(file = "../Data/other verts/vertebrate_trees/VertLife_mammal_tree/Vert_Life_Mammal_consensus_tree.newick") 

#check that tree is dichotomous
is.binary(mammal.phy)

#plot tree
plot.phylo(mammal.phy)

# Birds-----

#read tree
bird.phy <- read.tree(file = "../Data/other verts/vertebrate_trees/VertLife_bird_tree/Vert_Life_Bird_consensus_tree.newick")

#check that tree is dichotomous
is.binary(bird.phy)

#make dichotomous (randomly resolve polytomies)
bird.phy <- multi2di(bird.phy)

#plot tree
plot.phylo(bird.phy)

# Squamates-----

#read tree
squamate.phy <- read.tree(file = "../Data/other verts/vertebrate_trees/VertLifesquamate_tree/Vert_Life_Squamate_consensus_tree.newick")

#check that tree is dichotomous
is.binary(squamate.phy)

#make dichotomous (randomly resolve polytomies)
squamate.phy <- multi2di(squamate.phy)

#plot tree
plot.phylo(squamate.phy)

# Sharks-----

#read tree
shark.phy <- read.tree(file = "../Data/other verts/vertebrate_trees/VertLife_shark_tree/Vert_Life_Shark_consensus_tree.newick") 

#check that tree is dichotomous
is.binary(shark.phy)

#make dichotomous (randomly resolve polytomies)
shark.phy <- multi2di(shark.phy)

#plot tree
plot.phylo(shark.phy)

# Fishes-----

#read tree
fishes.phy <- read.newick(file = "../Data/other verts/vertebrate_trees/TimeTree_fishes.nwk") 

#check that tree is dichotomous
is.binary(fishes.phy)

#plot tree
plot.phylo(fishes.phy)

#remove distant branches from fish tree
#drops <- as.character(c("Polypterus_endlicherii", "Erpetoichthys_calabaricus", "Acipenser_ruthenus")) #create vector of taxa to drop from tree
drops <- as.character(c("Polypterus_endlicherii", "Erpetoichthys_calabaricus", "Acipenser_ruthenus", "Echidna_catenata", "Gymnothorax_fimbriatus", "Gymnothorax_moringa", "Gymnothorax_nudivomer", "Gymnothorax_richardsonii", "Gymnothorax_vicinus", "Muraena_lentiginosa", "Myrichthys_colubrinus", "Myrichthys_breviceps", "Ariosoma_balearicum", "Heteroconger_hassi")) #create vector of taxa to drop from tree

fishes_pruned <- drop.tip(phy = fishes.phy, tip = drops) 

#plot tree
plot.phylo(fishes_pruned)

```


## Interspecific scaling across vertebrate groups via PGLS

```{r vert-pgls}

# Mammal scaling ------------

## Axial length vs. mass ##

#remove rows from averaged dataframe not in phylogeny
mammal.almass <- mammal_avs %>% 
  mutate(tips = gsub(" ", "_", VertLife_name)) %>% 
  drop_na(AD_av) %>% 
  drop_na(mass_av) %>%
  select(tips, AD_av, mass_av)

#check that names match in dataframe and tree
name.check(phy = mammal.phy, data = mammal.almass, data.names = mammal.almass$tips)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
mammal.comp <- comparative.data(phy = mammal.phy, data = data.frame(mammal.almass), 
                            names.col = tips, vcv = TRUE, 
                            na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
mammal.comp$dropped$tips #phylogeny
mammal.comp$dropped$unmatched.rows #dataset

#run PGLS model
pgls_mammals <- pgls(log10(AD_av) ~ log10(mass_av), 
               data = mammal.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#check model assumptions with diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_mammals)
par(mfrow = c(1, 1))

#print model output 
summary(pgls_mammals)

#quick plot of pgls
plot(log10(AD_av) ~ log10(mass_av), data = mammal.almass)
abline(pgls_mammals)

#Likelihood plot for Pagel's lambda from the PGLS model of cornea diameter vs. SVL. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.mammals <- pgls.profile(pgls_mammals, "lambda")
plot(lambda.mammals)

#OLS fit for comparison
ols_mammals <- lm(formula = log10(AD_av) ~ log10(mass_av),
                   data = mammal.almass)

#model summary
summary(ols_mammals)

#plot data
plot_mammals <- ggplot(mammal.almass, aes(x = mass_av, y = AD_av,)) +
  geom_point(alpha = 0.4, pch = 1, size = 0.5) +
  theme_bw() +
  ggtitle("Mammals (Mammalia)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Axial eye length (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(pgls_mammals)[[2]], intercept = coef(pgls_mammals)[[1]]) +
  geom_abline(slope = coef(ols_mammals)[[2]], intercept = coef(ols_mammals)[[1]], linetype = "dashed") +
  geom_point(data = mammals, aes(x = rootmass, y = Axial_diameter_mm, col = Reference, shape = Subgroup, text = Dataset_name), alpha = 0.5, size = 1.5)

#interactive plot
ggplotly(plot_mammals)

# Bird scaling -------------

## Axial length vs. mass ##

#remove rows from averaged dataframe not in phylogeny
birds.almass <- birds_avs %>% 
  mutate(tips = gsub(" ", "_", VertLife_name)) %>% 
  drop_na(AD_av) %>% 
  drop_na(mass_av) %>%
  select(tips, AD_av, mass_av)

#check that names match in dataframe and tree
name.check(phy = bird.phy, data = birds.almass, data.names = birds.almass$tips)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
bird.comp <- comparative.data(phy = bird.phy, data = data.frame(birds.almass), names.col = tips, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
bird.comp$dropped$tips #phylogeny
bird.comp$dropped$unmatched.rows #dataset

#run PGLS model
pgls_birds <- pgls(log10(AD_av) ~ log10(mass_av),
               data = bird.comp,
               lambda = "ML", #ML estimate of lambda
               param.CI = 0.95)

#check model assumptions with diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_birds)
par(mfrow = c(1, 1))

#print model output
summary(pgls_birds)

#quick plot of pgls
plot(log10(AD_av) ~ log10(mass_av), data = birds.almass)
abline(pgls_birds)

#Likelihood plot for Pagel's lambda from the PGLS model of cornea diameter vs. SVL. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.birds <- pgls.profile(pgls_birds, "lambda")
plot(lambda.birds)

#OLS fit for comparison
ols_birds <- lm(formula = log10(AD_av) ~ log10(mass_av),
                   data = birds.almass)

#model summary
summary(ols_birds)

#plot data
plot_birds <- ggplot(birds.almass, aes(x = mass_av, y = AD_av,)) +
  geom_point(alpha = 0.4, pch = 1, size = 0.5) +
  theme_bw() +
  ggtitle("Birds (Aves)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Axial length (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(pgls_birds)[[2]], intercept = coef(pgls_birds)[[1]]) +
  geom_abline(slope = coef(ols_birds)[[2]], intercept = coef(ols_birds)[[1]], linetype = "dashed") +
  geom_point(data = birds, aes(x = rootmass, y = Axial_diameter_mm, col = Reference, shape = Subgroup, text = Dataset_name), alpha = 0.5, size = 1.5)

#interactive plot
ggplotly(plot_birds)


# Squamate scaling -----------

## Axial length vs. SVL ##

#remove rows from averaged dataframe not in phylogeny
squamate.adsvl <- squamate_avs %>% 
  ungroup() %>%
  mutate(tips = gsub(" ", "_", VertLife_name)) %>% 
  drop_na(AD_av) %>% 
  drop_na(svl_av) %>%
  select(tips, AD_av, svl_av) 

#check that names match in dataframe and tree
name.check(phy = squamate.phy, data = squamate.adsvl, data.names = squamate.adsvl$tips)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
squamate.comp.adsvl <- comparative.data(phy = squamate.phy, data = data.frame(squamate.adsvl), names.col = tips, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
squamate.comp.adsvl$dropped$tips #phylogeny
squamate.comp.adsvl$dropped$unmatched.rows #dataset

#run PGLS model
pgls_squamate.adsvl <- pgls(log10(AD_av) ~ log10(svl_av), 
               data = squamate.comp.adsvl, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#check model assumptions with diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_squamate.adsvl)
par(mfrow = c(1, 1))

#print model output 
summary(pgls_squamate.adsvl)

#quick plot of pgls
plot(log10(AD_av) ~ log10(svl_av), data = squamate.adsvl)
abline(pgls_squamate.adsvl)

#Likelihood plot for Pagel's lambda from the PGLS model of cornea diameter vs. SVL. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.squamate.adsvl <- pgls.profile(pgls_squamate.adsvl, "lambda")
plot(lambda.squamate.adsvl)

#OLS fit for comparison
ols_squamate.adsvl <- lm(formula = log10(AD_av) ~ log10(svl_av),
                   data = squamate.adsvl)

#model summary
summary(ols_squamate.adsvl)

#plot data
plot_squamates.adsvl <- ggplot(squamate.adsvl, aes(x = svl_av, y = AD_av,)) +
  geom_point(alpha = 0.4, pch = 1, size = 0.5) +
  theme_bw() +
  ggtitle("Squamates (Squamata)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Axial length (mm)") +
  scale_x_log10(name = "Snout-vent length (mm))") +
  geom_abline(slope = coef(pgls_squamate.adsvl)[[2]], intercept = coef(pgls_squamate.adsvl)[[1]]) +
  geom_abline(slope = coef(ols_squamate.adsvl)[[2]], intercept = coef(ols_squamate.adsvl)[[1]], linetype = "dashed") +
  geom_point(data = squamates[squamates["Reference"] == "Hall 2008" & squamates["Subgroup"] == "lizards", ], aes(x = SVL_mm, y = Axial_diameter_mm, text = Dataset_name, col = Reference, shape = Subgroup), alpha = 0.5, size = 1.5) +
  guides(color = guide_legend(order = 1),
         shape = guide_legend(order = 2))

#interactive plot
ggplotly(plot_squamates.adsvl)

## Transverse diameter vs. SVL ##

#remove rows from averaged dataframe not in phylogeny
squamate.tdsvl <- squamate_avs %>% 
  mutate(tips = gsub(" ", "_", VertLife_name)) %>% 
  drop_na(TD_av) %>% 
  drop_na(svl_av) %>%
  select(tips, TD_av, svl_av, Subgroup) %>%
  droplevels()

#check that names match in dataframe and tree
name.check(phy = squamate.phy, data = squamate.tdsvl, data.names = squamate.tdsvl$tips)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
squamate.comp.tdsvl <- comparative.data(phy = squamate.phy, data = data.frame(squamate.tdsvl), names.col = tips, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
squamate.comp.tdsvl$dropped$tips #phylogeny
squamate.comp.tdsvl$dropped$unmatched.rows #dataset

#run PGLS model
pgls_squamate.tdsvl <- pgls(log10(TD_av) ~ log10(svl_av) * Subgroup, 
               data = squamate.comp.tdsvl, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#check model assumptions with diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_squamate.tdsvl)
par(mfrow = c(1, 1))

#print model output 
summary(pgls_squamate.tdsvl)

#quick plot of pgls
plot(log10(TD_av) ~ log10(svl_av), data = squamate.tdsvl)
abline(pgls_squamate.tdsvl)

#Likelihood plot for Pagel's lambda
lambda.squamate.tdsvl <- pgls.profile(pgls_squamate.tdsvl, "lambda")
plot(lambda.squamate.tdsvl)

#OLS fit for comparison
ols_squamate.tdsvl <- lm(formula = log10(TD_av) ~ log10(svl_av) * Subgroup,
                   data = squamate.tdsvl)

#model summary
summary(ols_squamate.tdsvl)

#make raw data dataframe for plots
squam.raw.tdsvl <- squamates %>%
  drop_na(Trans_diameter_mm) %>% 
  drop_na(SVL_mm)

#plot data
plot_squamates.tdsvl <- ggplot(squamate.tdsvl, aes(x = svl_av, y = TD_av, shape = Subgroup)) +
  geom_point(alpha = 0.4, pch = 1, size = 0.5) +
  theme_bw() +
  ggtitle("Squamates (Gekkomorpha & Colubridae)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Transverse eye diameter (mm)") +
  scale_x_log10(name = "Snout-vent length (mm)") +
  geom_abline(slope = coef(pgls_squamate.tdsvl)[[2]], intercept = coef(pgls_squamate.tdsvl)[[1]]) +
  geom_abline(slope = coef(pgls_squamate.tdsvl)[[2]] + coef(pgls_squamate.tdsvl)[[4]], 
              intercept = coef(pgls_squamate.tdsvl)[[1]] + coef(pgls_squamate.tdsvl)[[3]]) +
  geom_abline(slope = coef(ols_squamate.tdsvl)[[2]], intercept = coef(ols_squamate.tdsvl)[[1]], linetype = "dashed") +
  geom_abline(slope = coef(ols_squamate.tdsvl)[[2]] + coef(ols_squamate.tdsvl)[[4]], 
              intercept = coef(pgls_squamate.tdsvl)[[1]] + coef(pgls_squamate.tdsvl)[[3]], linetype = "dashed") +
  geom_point(data = squam.raw.tdsvl, aes(x = SVL_mm, y = Trans_diameter_mm, col = Reference, shape = Subgroup, text = Dataset_name), alpha = 0.5, size = 1.5) +
  guides(color = guide_legend(order = 1),
         shape = guide_legend(order = 2))


#interactive plot
ggplotly(plot_squamates.tdsvl)

## Axial diameter vs. mass ##

#remove rows from averaged dataframe not in phylogeny
squamate.admass <- squamate_avs %>% 
  mutate(tips = gsub(" ", "_", VertLife_name)) %>% 
  drop_na(AD_av) %>% 
  drop_na(mass_av) %>%
  select(tips, AD_av, mass_av)

#check that names match in dataframe and tree
name.check(phy = squamate.phy, data = squamate.admass, data.names = squamate.admass$tips)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
squamate.comp.admass <- comparative.data(phy = squamate.phy, data = data.frame(squamate.admass), names.col = tips, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
squamate.comp.admass$dropped$tips #phylogeny
squamate.comp.admass$dropped$unmatched.rows #dataset

#run PGLS model
pgls_squamate.admass <- pgls(log10(AD_av) ~ log10(mass_av), 
               data = squamate.comp.admass, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#check model assumptions with diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_squamate.admass)
par(mfrow = c(1, 1))

#print model output 
summary(pgls_squamate.admass)

## Insufficient data (n = 8) to sufficiently estimate slope. 

# Shark scaling -------

## Axial diameter vs. mass ##

#remove rows from averaged dataframe not in phylogeny
shark.admass <- shark_avs %>% 
  mutate(tips = gsub(" ", "_", VertLife_name)) %>% 
  drop_na(AD_av) %>% 
  drop_na(mass_av) %>%
  select(tips, AD_av, mass_av)

#check that names match in dataframe and tree
name.check(phy = shark.phy, data = shark.admass, data.names = shark.admass$tips)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
shark.comp <- comparative.data(phy = shark.phy, data = data.frame(shark.admass), names.col = tips, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
shark.comp$dropped$tips #phylogeny
shark.comp$dropped$unmatched.rows #dataset

#run PGLS model
pgls_shark.admass <- pgls(log10(AD_av) ~ log10(mass_av),
               data = shark.comp, 
               lambda = "ML",
               param.CI = 0.95)

#check model assumptions with diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_shark.admass)
par(mfrow = c(1, 1))

#print model output 
summary(pgls_shark.admass)

#quick plot of pgls
plot(log10(AD_av) ~ log10(mass_av), data = shark.admass)
abline(pgls_shark.admass)

#Likelihood plot for Pagel's lambda
lambda.shark.admass <- pgls.profile(pgls_shark.admass, "lambda")
plot(lambda.shark.admass)

#OLS fit for comparison
ols_shark.admass <- lm(formula = log10(AD_av) ~ log10(mass_av),
                   data = shark.admass)

#model summary
summary(ols_shark.admass)

#plot data
plot_sharks <- ggplot(shark.admass, aes(x = mass_av, y = AD_av,)) +
  geom_point(alpha = 0.4, pch = 1, size = 0.5) +
  theme_bw() +
  ggtitle("Sharks and rays (Elasmobranchii)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Axial eye length (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(pgls_shark.admass)[[2]], intercept = coef(pgls_shark.admass)[[1]]) +
  geom_abline(slope = coef(ols_shark.admass)[[2]], intercept = coef(ols_shark.admass)[[1]], linetype = "dashed") +
  geom_point(data = sharks, aes(x = rootmass, y = Axial_diameter_mm, col = Reference, shape = Subgroup, text = Dataset_name), alpha = 0.5, size = 1.5)

#interactive plot
ggplotly(plot_sharks)

#Fish scaling ------------

## Axial diameter vs. mass ##

#remove rows from averaged dataframe not in phylogeny
fishes.admass <- fishes_avs %>% 
  drop_na(AD_av) %>% 
  drop_na(mass_av) %>%
  select(tips, AD_av, mass_av) %>%
  mutate(tips = as.character(tips))

#check that names match in dataframe and tree
name.check(phy = fishes_pruned, data = fishes.admass, data.names = fishes.admass$tips)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
fishes.comp.admass <- comparative.data(phy = fishes_pruned, data = data.frame(fishes.admass), names.col = tips, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
fishes.comp.admass$dropped$tips #phylogeny
fishes.comp.admass$dropped$unmatched.rows #dataset

#run PGLS model
pgls_fishes.admass <- pgls(log10(AD_av) ~ log10(mass_av),
               data = fishes.comp.admass, 
               lambda = "ML", bounds = list(lambda = c(1e-05, 1)),
               param.CI = 0.95)

#check model assumptions with diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_fishes.admass)
par(mfrow = c(1, 1))

#print model output 
summary(pgls_fishes.admass)

#quick plot of pgls
plot(log10(AD_av) ~ log10(mass_av), data = fishes.admass)
abline(pgls_fishes.admass)

#Likelihood plot for Pagel's lambda
lambda.fishes.admass <- pgls.profile(pgls_fishes.admass, "lambda")
plot(lambda.fishes.admass)

#OLS fit for comparison
ols_fishes.admass <- lm(formula = log10(AD_av) ~ log10(mass_av),
                   data = fishes.admass)

#model summary
summary(ols_fishes.admass)

#make raw data dataframe for plots
fish.raw.admass <- fishes %>%
  drop_na(Axial_diameter_mm) %>% 
  drop_na(rootmass) %>%
  filter(TimeTree_tiplabel %in% fishes_pruned$tip.label | TimeTree_genus_sub %in% fishes_pruned$tip.label)

#plot data
plot_fishes.admass <- ggplot(fishes.admass, aes(x = mass_av, y = AD_av,)) +
  geom_point(alpha = 0.4, pch = 1, size = 0.5) +
  theme_bw() +
  ggtitle("Fishes (Actinopterygii)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Axial length (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(pgls_fishes.admass)[[2]], intercept = coef(pgls_fishes.admass)[[1]]) +
  geom_abline(slope = coef(ols_fishes.admass)[[2]], intercept = coef(ols_fishes.admass)[[1]], linetype = "dashed") +
  geom_point(data = fish.raw.admass, aes(x = rootmass, y = Axial_diameter_mm, col = Reference, shape = Subgroup, text = Dataset_name), alpha = 0.5, size = 1.5)

#interactive plot
ggplotly(plot_fishes.admass)

## Transverse diameter vs. mass ##

#remove rows from averaged dataframe not in phylogeny
fishes.tdmass <- fishes_avs %>% 
  drop_na(TD_av) %>% 
  drop_na(mass_av) %>%
  select(tips, TD_av, mass_av) %>%
  mutate(tips = as.character(tips))

#check that names match in dataframe and tree
name.check(phy = fishes_pruned, data = fishes.tdmass, data.names = fishes.tdmass$tips)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
fishes.comp.tdmass <- comparative.data(phy = fishes_pruned, data = data.frame(fishes.tdmass), names.col = tips, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
fishes.comp.tdmass$dropped$tips #phylogeny
fishes.comp.tdmass$dropped$unmatched.rows #dataset

#run PGLS model
pgls_fishes.tdmass <- pgls(log10(TD_av) ~ log10(mass_av),
               data = fishes.comp.tdmass, 
               lambda = "ML",
               param.CI = 0.95)

#check model assumptions with diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_fishes.tdmass)
par(mfrow = c(1, 1))

#print model output 
summary(pgls_fishes.tdmass)

#quick plot of pgls
plot(log10(TD_av) ~ log10(mass_av), data = fishes.tdmass)
abline(pgls_fishes.tdmass)

#Likelihood plot for Pagel's lambda
lambda.fishes.tdmass <- pgls.profile(pgls_fishes.tdmass, "lambda")
plot(lambda.fishes.tdmass)

#OLS fit for comparison
ols_fishes.tdmass <- lm(formula = log10(TD_av) ~ log10(mass_av),
                   data = fishes.tdmass)

#model summary
summary(ols_fishes.tdmass)

#make raw data dataframe for plots
fish.raw.tdmass <- fishes %>%
  drop_na(Trans_diameter_mm) %>% 
  drop_na(rootmass) %>%
  filter(TimeTree_tiplabel %in% fishes_pruned$tip.label | TimeTree_genus_sub %in% fishes_pruned$tip.label)

  
#plot data
plot_fishes.tdmass <- ggplot(fishes.tdmass, aes(x = mass_av, y = TD_av,)) +
  geom_point(alpha = 0.4, pch = 1, size = 0.5) +
  theme_bw() +
  ggtitle("Fishes (Teleostei)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Transverse eye (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(pgls_fishes.tdmass)[[2]], intercept = coef(pgls_fishes.tdmass)[[1]]) +
  geom_abline(slope = coef(ols_fishes.tdmass)[[2]], intercept = coef(ols_fishes.tdmass)[[1]], linetype = "dashed") +
  geom_point(data = fish.raw.tdmass, aes(x = rootmass, y = Trans_diameter_mm, col = Reference, shape = Subclade, text = Dataset_name), alpha = 0.5, size = 1.5)

#interactive plot
ggplotly(plot_fishes.tdmass)

#Frogs (convert transverse to axial diameter via correlation in fresh specimens) --------

# Frogs scaling axial length with mass

#subset frog data and add group name
frogs.almass <- av.edmass %>%
  mutate(eye.axial = 10^(coef(mod_edad)[[2]]*log10(eye_av)) + 10^(coef(mod_edad)[[1]])) %>%
  rename(tips = mod_tiplabel, mass_av = rootmass_av, AD_av = eye.axial) %>%
  mutate(Subgroup = "frogs & toads", Reference = "This study") %>%
  select(tips, mass_av, AD_av, Subgroup, Reference) %>%
  mutate(tips = as.character(tips))

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = frogs.almass, data.names = frogs.almass$tips)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
frogs.comp.admass <- comparative.data(phy = frog.tree, data = data.frame(frogs.almass), names.col = tips, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
frogs.comp.admass$dropped$tips #phylogeny
frogs.comp.admass$dropped$unmatched.rows #dataset

#run PGLS model
pgls_frogs.admass <- pgls(log10(AD_av) ~ log10(mass_av),
               data = frogs.comp.admass, 
               lambda = "ML",
               param.CI = 0.95)

#check model assumptions with diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_frogs.admass)
par(mfrow = c(1, 1))

#print model output 
summary(pgls_frogs.admass)

#quick plot of pgls
plot(log10(AD_av) ~ log10(mass_av), data = frogs.almass)
abline(pgls_frogs.admass)

#Likelihood plot for Pagel's lambda
lambda.frogs.admass <- pgls.profile(pgls_frogs.admass, "lambda")
plot(lambda.frogs.admass)

#OLS fit for comparison
ols_frogs.admass <- lm(formula = log10(AD_av) ~ log10(mass_av),
                   data = frogs.almass)

#model summary
summary(ols_frogs.admass)

#plot data
plot_frogs.admass <- ggplot(frogs.almass, aes(x = mass_av, y = AD_av, text = tips, shape = Subgroup, col = Reference)) +
  geom_point(alpha = 0.5, size = 1.5) +
  theme_bw() +
  ggtitle("Anuran amphibians (Anura)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Axial length (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(pgls_frogs.admass)[[2]], intercept = coef(pgls_frogs.admass)[[1]]) +
  geom_abline(slope = coef(ols_frogs.admass)[[2]], intercept = coef(ols_frogs.admass)[[1]], linetype = "dashed") 

#interactive plot
ggplotly(plot_frogs.admass)

# Frogs scaling axial length with SVL

#subset frog data and add group name
frogs.alsvl <- av.edsvl %>%
  mutate(eye.axial = coef(mod_edad)[[2]]*eye_av + coef(mod_edad)[[1]]) %>%
  rename(tips = mod_tiplabel, AD_av = eye.axial) %>%
  mutate(Subgroup = "frogs & toads", Reference = "This study") %>%
  select(tips, svl_av, AD_av, Subgroup, Reference) %>%
  mutate(tips = as.character(tips))

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = frogs.alsvl, data.names = frogs.alsvl$tips)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
frogs.comp.alsvl <- comparative.data(phy = frog.tree, data = data.frame(frogs.alsvl), names.col = tips, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
frogs.comp.alsvl$dropped$tips #phylogeny
frogs.comp.alsvl$dropped$unmatched.rows #dataset

#run PGLS model
pgls_frogs.alsvl <- pgls(log10(AD_av) ~ log10(svl_av),
               data = frogs.comp.alsvl, 
               lambda = "ML",
               param.CI = 0.95)

#check model assumptions with diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_frogs.alsvl)
par(mfrow = c(1, 1))

#print model output 
summary(pgls_frogs.alsvl)

#quick plot of pgls
plot(log10(AD_av) ~ log10(svl_av), data = frogs.alsvl)
abline(pgls_frogs.alsvl)

#Likelihood plot for Pagel's lambda
lambda.frogs.alsvl <- pgls.profile(pgls_frogs.alsvl, "lambda")
plot(lambda.frogs.alsvl)

#OLS fit for comparison
ols_frogs.alsvl <- lm(formula = log10(AD_av) ~ log10(svl_av),
                   data = frogs.alsvl)

#model summary
summary(ols_frogs.alsvl)

#plot data
plot_frogs.alsvl <- ggplot(frogs.alsvl, aes(x = svl_av, y = AD_av, text = tips, color = Reference, shape = Subgroup)) +
  geom_point(alpha = 0.5, size = 1.5) +
  theme_bw() +
  ggtitle("Anuran amphibians (Anura)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Axial eye length (mm)") +
  scale_x_log10(name = "SVL (mm)") +
  geom_abline(slope = coef(pgls_frogs.alsvl)[[2]], intercept = coef(pgls_frogs.alsvl)[[1]]) +
  geom_abline(slope = coef(ols_frogs.alsvl)[[2]], intercept = coef(ols_frogs.alsvl)[[1]], linetype = "dashed") 

#interactive plot
ggplotly(plot_frogs.alsvl)
```


```{r figure-supp-vertebrates, fig.height=15, fig.width=10}

#make panels 
fig.a <- plot_birds 

fig.b <- plot_mammals

fig.c <- plot_sharks

fig.d <- plot_fishes.admass
  
fig.e <-  plot_frogs.admass

fig.f <- plot_squamates.adsvl

fig.g <- plot_frogs.alsvl
  
fig.h <- plot_squamates.tdsvl

fig.i <- plot_fishes.tdmass

#use cowplot package to nicely plot the graphs together with a shared common legend.
plots <- plot_grid(fig.a, fig.b, fig.c, fig.d, fig.e, fig.f, fig.g, fig.h, fig.i, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 5) #number of columns in grids

#show figure
plots

#export figure
pdf("../Manuscript/Figures/figure-supp-verts.pdf", width=12, height=15)
plot_grid(plots)
dev.off()
```


Because these data came from various published papers and datsets, taxonomic groupings and measurements were not consistent across all comparisons. However, as we measured transverse eye diameter, snout-vent length, and mass across all of our museum specimens, and additionally had a smaller dataset of axial length, transverse eye diameter, and snout-vent length measurements from fresh, field-collected specimens, we were able to compare anurans to other vertebrate groups across multiple measures of eye scaling.
 

```{r scaling-verts-axial, fig.height = 4, fig.width = 9}

# Scaling of eye axial diameter with mass across vertebrates. 

#add group names to datasets
frogs.almass <- frogs.almass %>%
  mutate(group = Subgroup) %>%
  select(tips, mass_av, AD_av, group)
mammal.almass <- mutate(mammal.almass, group = "mammals")
birds.almass <- mutate(birds.almass, group = "birds")
shark.admass <- mutate(shark.admass, group = "sharks & rays")
fishes.admass <- mutate(fishes.admass, group = "ray-finned fishes")

#merge datasets
vert.almass <- rbind(frogs.almass, mammal.almass, birds.almass, shark.admass, fishes.admass)

#see number of each group
table(vert.almass$group)

#make color vector for groups
col_almass <- c("mammals" = "#b15928", 
               "birds" = "#e41a1c",
               "sharks & rays" = "#ff7f00",
               "ray-finned fishes" = "#377eb8",
               "frogs & toads" = "#33a02c")

#make shape vector for groups
sh_almass <- c("mammals" = 22, 
               "birds" = 24,
               "sharks & rays" = 23,
               "ray-finned fishes" = 25,
               "frogs & toads" = 21)

#plot
plot_verts.almass <- ggplot(vert.almass, aes(x = mass_av, y = AD_av, text = tips)) +
  geom_point(aes(color = group, fill = group, shape = group), size = 1, alpha = 0.5) +
  scale_color_manual(values = col_almass, 
                     name = "", 
                     breaks = c("frogs & toads","birds","sharks & rays","ray-finned fishes","mammals"),
                     labels = c(expression(paste("frogs & toads"^"1")), 
                              expression(paste("birds"^"2,3")),
                              expression(paste("sharks & rays"^"4")),
                              expression(paste("ray-finned fishes"^"2,5")),
                              expression(paste("mammals"^"2,6")))) +
  scale_fill_manual(values = col_almass, 
                    name = "",
                    breaks = c("frogs & toads","birds","sharks & rays","ray-finned fishes","mammals"),
                     labels = c(expression(paste("frogs & toads"^"1")), 
                              expression(paste("birds"^"2,3")),
                              expression(paste("sharks & rays"^"4")),
                              expression(paste("ray-finned fishes"^"2,5")),
                              expression(paste("mammals"^"2,6")))) +
  scale_shape_manual(values = sh_almass, 
                     name = "", 
                     breaks = c("frogs & toads","birds","sharks & rays","ray-finned fishes","mammals"),
                     labels = c(expression(paste("frogs & toads"^"1")), 
                              expression(paste("birds"^"2,3")),
                              expression(paste("sharks & rays"^"4")),
                              expression(paste("ray-finned fishes"^"2,5")),
                              expression(paste("mammals"^"2,6")))) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Axial length (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(pgls_frogs.admass)[[2]], intercept = coef(pgls_frogs.admass)[[1]], col = col_almass["frogs & toads"]) +
  geom_abline(slope = coef(pgls_mammals)[[2]], intercept = coef(pgls_mammals)[[1]], col = col_almass["mammals"]) +
  geom_abline(slope = coef(pgls_birds)[[2]], intercept = coef(pgls_birds)[[1]], col = col_almass["birds"]) +
  geom_abline(slope = coef(pgls_shark.admass)[[2]], intercept = coef(pgls_shark.admass)[[1]], col = col_almass["sharks & rays"]) +
  geom_abline(slope = coef(pgls_fishes.admass)[[2]], intercept = coef(pgls_fishes.admass)[[1]], col = col_almass["ray-finned fishes"])
  
#Interactive plot
ggplotly(plot_verts.almass)
```

Then we looked at axial eye diameter vs. snout-vent length in frogs and lizards. 

```{r scaling-verts-adsvl}

# Scaling of eye axial diameter with snout-vent length across frogs and lizards. 

#add group names to other datasets
squamate.adsvl <- squamate.adsvl %>% 
  mutate(group = "lizards") %>% 
  select(tips, AD_av, svl_av, group)
frogs.alsvl <- frogs.alsvl %>%
  mutate(group = Subgroup) %>%
  select(tips, AD_av, svl_av, group)

#merge datasets
vert.alsvl <- rbind(frogs.alsvl, squamate.adsvl)

#see number of each group
table(vert.alsvl$group)

#make color vector for groups
col_alsvl <- c("lizards" = "#6a3d9a",
               "frogs & toads" = "#33a02c")

#make shape vector for groups
sh_alsvl <- c("lizards" = 22,
               "frogs & toads" = 21)

#plot
plot_verts.alsvl <- ggplot(vert.alsvl, aes(x = svl_av, y = AD_av, text = tips)) +
  geom_point(aes(color = group, fill = group, shape = group), size = 1, alpha = 0.5) +
  scale_color_manual(values = col_alsvl, 
                     name = "", 
                     breaks = c("frogs & toads", "lizards"),
                     labels = c(expression(paste("frogs & toads"^"1")), 
                              expression(paste("lizards"^"7")))) +
  scale_fill_manual(values = col_alsvl, 
                    name = "",
                   breaks = c("frogs & toads", "lizards"),
                     labels = c(expression(paste("frogs & toads"^"1")), 
                              expression(paste("lizards"^"7")))) +
  scale_shape_manual(values = sh_alsvl, 
                     name = "", 
                     breaks = c("frogs & toads", "lizards"),
                     labels = c(expression(paste("frogs & toads"^"1")), 
                              expression(paste("lizards"^"7")))) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Aaxial length (mm)") +
  scale_x_log10(name = "Snout-vent length (mm)") +
  geom_abline(slope = coef(pgls_frogs.alsvl)[[2]], intercept = coef(pgls_frogs.alsvl)[[1]], col = col_alsvl["frogs & toads"]) +
  geom_abline(slope = coef(pgls_squamate.adsvl)[[2]], intercept = coef(pgls_squamate.adsvl)[[1]], col = col_alsvl["lizards"])
  
#Interactive plot
ggplotly(plot_verts.alsvl)
```

Then we looked at transverse eye diameter versus the cube root of mass across frogs and acanthomorph fishes. 

```{r scaling-verts-transverse, fig.height = 4, fig.width = 8, fig.cap = ""}

# Scaling of eye transverse diameter with cuberoot of mass across frogs and fishes. 

#add group names to datasets
frogs.edmass <- av.edmass %>%
  mutate(group = "frogs & toads", tips = genus_species, mass_av = rootmass_av, TD_av = eye_av) %>%
  select(tips, TD_av, mass_av, group)
fishes.tdmass <- fishes.tdmass %>% 
  mutate(group = "teleost fishes") %>% 
  select(tips, TD_av, mass_av, group)

#merge datasets
vert.tdmass <- rbind(frogs.edmass, fishes.tdmass)

#see number of each group
table(vert.tdmass$group)

#make color vector for groups
col_tdmass <- c("teleost fishes" = "#377eb8",
               "frogs & toads" = "#33a02c")

#make shape vector for groups
sh_tdmass <- c("teleost fishes" = 25,
               "frogs & toads" = 21)

#plot
plot_verts.tdmass <- ggplot(vert.tdmass, aes(x = mass_av, y = TD_av, text = tips)) +
  geom_point(aes(color = group, fill = group, shape = group), size = 1, alpha = 0.5) +
  scale_color_manual(values = col_tdmass, 
                     name = "", 
                     breaks = c("frogs & toads", "teleost fishes"),
                     labels = c(expression(paste("frogs & toads"^"1")), 
                              expression(paste("teleost fishes"^"5,8")))) +
  scale_fill_manual(values = col_tdmass, 
                    name = "",
                    breaks = c("frogs & toads", "teleost fishes"),
                    labels = c(expression(paste("frogs & toads"^"1")), 
                              expression(paste("teleost fishes"^"5,8")))) +
  scale_shape_manual(values = sh_tdmass, 
                     name = "", 
                     breaks = c("frogs & toads", "teleost fishes"),
                     labels = c(expression(paste("frogs & toads"^"1")), 
                              expression(paste("teleost fishes"^"5,8")))) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Eye diameter (mm)") +
  scale_x_log10(name = "Cube root of mass (g)") +
  geom_abline(slope = coef(pgls_edmass)[[2]], intercept = coef(pgls_edmass)[[1]], col = col_tdmass["frogs & toads"]) +
  geom_abline(slope = coef(pgls_fishes.tdmass)[[2]], intercept = coef(pgls_fishes.tdmass)[[1]], col = col_tdmass["teleost fishes"])
  
#Interactive plot
ggplotly(plot_verts.tdmass)
```

Finally, we looked at transverse eye diameter versus snout-vent length across frogs, geckos, and snakes. 

```{r scaling-verts-length, fig.height=4, fig.width=8}

# Scaling of eye transverse diameter with SVL across frogs, snakes, and geckos. 

#add group names to datasets
frogs.edsvl <- av.edsvl %>%
  mutate(group = "frogs & toads", tips = genus_species, TD_av = eye_av) %>%
  select(tips, TD_av, svl_av, group)
squamate.tdsvl <- squamate.tdsvl %>% 
  ungroup() %>%
  mutate(group = Subgroup) %>% 
  select(tips, TD_av, svl_av, group)

#merge datasets
vert.tdsvl <- rbind(frogs.edsvl, squamate.tdsvl)

#see number of each group
table(vert.tdsvl$group)

#make color vector for groups
col_tdsvl <- c("geckos" = "#6a3d9a",
               "colubrid snakes" = "#cab2d6", 
               "frogs & toads" = "#33a02c")

#make shape vector for groups
sh_tdsvl <- c("geckos" = 22,
               "colubrid snakes" = 22, 
               "frogs & toads" = 21)

#plot
plot_verts.tdsvl <- ggplot(vert.tdsvl, aes(x = svl_av, y = TD_av, text = tips)) +
  geom_point(aes(color = group, fill = group, shape = group), size = 1, alpha = 0.5) +
  scale_color_manual(values = col_tdsvl, 
                     name = "", 
                     breaks = c("frogs & toads", "colubrid snakes", "geckos"),
                     labels = c(expression(paste("frogs & toads"^1)), 
                              expression(paste("colubrid snakes"^9)),
                              expression(paste("geckos"^10)))) +
  scale_fill_manual(values = col_tdsvl, 
                    name = "", 
                     breaks = c("frogs & toads", "colubrid snakes", "geckos"),
                     labels = c(expression(paste("frogs & toads"^1)), 
                              expression(paste("colubrid snakes"^9)),
                              expression(paste("geckos"^10)))) +
  scale_shape_manual(values = sh_tdsvl, 
                     name = "", 
                      breaks = c("frogs & toads", "colubrid snakes", "geckos"),
                     labels = c(expression(paste("frogs & toads"^1)), 
                              expression(paste("colubrid snakes"^9)),
                              expression(paste("geckos"^10)))) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text.align = 0) +
  scale_y_log10(name = "Eye diameter (mm)") +
  scale_x_log10(name = "Snout-vent length (mm)") +
  geom_abline(slope = coef(pgls_edsvl)[[2]], intercept = coef(pgls_edsvl)[[1]], col = col_tdsvl["frogs & toads"]) +
 geom_abline(slope = coef(pgls_squamate.tdsvl)[[2]], intercept = coef(pgls_squamate.tdsvl)[[1]], col = col_tdsvl["colubrid snakes"]) +
  geom_abline(slope = coef(pgls_squamate.tdsvl)[[2]] + coef(pgls_squamate.tdsvl)[[4]], 
              intercept = coef(pgls_squamate.tdsvl)[[1]] + coef(pgls_squamate.tdsvl)[[3]], col = col_tdsvl["geckos"]) 
 
  
#Interactive plot
ggplotly(plot_verts.tdsvl)
```

```{r verts-figure5, fig.width=14, fig.height=6}

#make panels 
fig.a <- plot_verts.almass +
  theme(legend.position = "none", text = element_text(size=10))

fig.b <- plot_verts.alsvl +
  theme(legend.position = "none", text = element_text(size=10))

fig.c <- plot_verts.tdmass +
  theme(legend.position = "none", text = element_text(size=10))

fig.d <- plot_verts.tdsvl +
  theme(legend.position = "none", text = element_text(size=10))

# extract legends from figures
leg.a <- get_legend(plot_verts.almass + 
                      theme(legend.position = "right",
                            legend.justification = "top"))

leg.b <- get_legend(plot_verts.alsvl + theme(legend.position = "right", 
                    legend.justification = "top"))

leg.c <- get_legend(plot_verts.tdmass + theme(legend.position = "right", 
                    legend.justification = "top"))

leg.d <- get_legend(plot_verts.tdsvl + theme(legend.position = "right", 
                    legend.justification = "top"))

#panels A&C
plot.ac <- plot_grid(fig.a, leg.a, fig.c, leg.c,
                     align = 'vh',
                     axis = 't',
                     labels = c("A","", "C",""), 
                     hjust = -0.5, 
                     vjust = 1,
                     ncol = 2,
                     rel_widths = c(1,0.4))

#panels B&D
plot.bd <- plot_grid(fig.b, leg.b, fig.d, leg.d,
                     align = 'vh',
                     axis = 't',
                     labels = c("B","", "D",""), 
                     hjust = -0.5, 
                     vjust = 1,
                     ncol = 2,
                     rel_widths = c(1,0.4))

#use cowplot package to nicely plot the graphs together with a shared common legend.
plots <- plot_grid(plot.ac, plot.bd, 
                   align = 'vh', 
                   hjust = -1, 
                   vjust = 1,
                   ncol = 2,
                   rel_widths = c(1,1))

#export pdf of figure
pdf("../Manuscript/Figures/figure5-verts.pdf", width = 9.8, height = 5)
plot_grid(plots)
dev.off()
```


***


# Relative eye size and eye investment in frogs

To investigate how relative eye size differed across species with different ecological traits, we looked at the residuals from PGLS fits of eye diameter vs. body size. 

First, we added residuals of PGLS fits to our datasets by species. 

```{r pgls-residuals}

#Add residuals from PGLS fits to dataframes

#Residuals for PGLS of eye diameter ~ cube root of mass -------

#extract pgls residuals 
pglsres.edmass <- residuals(pgls_edmass) 

#name residuals
colnames(pglsres.edmass) <- "pglsres.edmass" 

#add rownames to original dataframe
rownames(av.edmass) <- av.edmass$mod_tiplabel 

#merge residuals with original data by rowname
av.edmass.phy <- merge(av.edmass, pglsres.edmass, by = "row.names") 

#Residuals for PGLS of eye diameter ~ SVL ------------

#extract pgls residuals
pglsres.edsvl <- residuals(pgls_edsvl) 

#name residuals
colnames(pglsres.edsvl) <- "pglsres.edsvl"

#add rownames to original dataframe
rownames(av.edsvl) <- av.edsvl$mod_tiplabel 

#merge residuals with original data by rowname
av.edsvl.phy <- merge(av.edsvl, pglsres.edsvl, by = "row.names") 

#residuals for PGLS of cornea diameter ~ eye diameter -------------

#extract pgls residuals
pglsres.edcd <- residuals(pgls_edcd) 

#name residuals
colnames(pglsres.edcd) <- "pglsres.edcd" 

#add rownames to original dataframe
rownames(av.edcd) <- av.edcd$mod_tiplabel 

#merge residuals with original data by rowname
av.edcd.phy <- merge(av.edcd, pglsres.edcd, by = "row.names") 
```

## Phylogenetic distribution of absolute and relative eye size

We then plotted absolute eye size and relative eye investment (represented by the residual of the eye size vs. body size PGLS) onto the anuran phylogeny. 

```{r plot-phytools, fig.height = 15, fig.width = 8}

# color vectors -------

#create vector of colors for activity period
col_act <- c("Both" = "deeppink3",
             "Diurnal" = "darkgoldenrod1", 
             "Nocturnal" = "blueviolet",
             "Unknown" = "gray")

# create vector of colors for life history
col_mat <- c("Ground" = "wheat4", 
         "Lentic water" = "royalblue1", 
         "Lotic water" = "skyblue1",
         "Plants" = "springgreen3",
         "Unknown" = "gray")


# create vector of colors for life history
col_lif <- c("Free-living larvae" = "darkturquoise", 
         "No free-living larvae" = "black", 
         "Unknown" = "gray")

# create vector of colors for larval habitat
col_larv <- c("Lentic water" = "royalblue1", 
         "Lotic water" = "skyblue1",
         "No larvae" = "lightcoral",
         "Obscured" = "darkslateblue",
         "On land" = "lightpink4",
         "Unknown" = "gray")

# create vector of colors for sex dichromatism
col_dich <- c("Absent" = "black", 
         "Present" = "chartreuse2",
         "Unknown" = "gray")



# Prep data and phylogeny -----

#subset data for eye diameter and mass
edmass.bars <- av.edmass.phy %>%
  mutate(oldtip = mod_tiplabel, 
         newtip = genus_species, 
         abseye = eye_av, 
         inveye = pglsres.edmass, 
         hab = Adult_habitat,
         act = Activity_period, 
         mat = Mating_habitat,
         lif = Life_history,
         larv = Larval_habitat,
         dich = Sex_dichromatism,
         family = Family) %>%
  select(oldtip, newtip, abseye, inveye, hab, act, mat, lif, larv, dich, family)

# set row names in dataset to match the tip labels in the tree
row.names(edmass.bars) <- edmass.bars$newtip

#drop phylogeny tips not in dataset)
tree_edmass <- drop.tip(frog.tree, frog.tree$tip.label[!(frog.tree$tip.label %in% edmass.bars$oldtip)])

#ladderize tree
tree_edmass <- ladderize(tree_edmass)

# Make Genus_species tip labels based on  Amphibian Species of the World names (and no family at end)
sp.tips <- data.frame(old = av.edmass$mod_tiplabel, new = av.edmass$genus_species)

# replace tip labels in phylogeny
tree_edmass <- sub.taxa.label(tree_edmass, sp.tips)

#check that phylogeny and data match exactly
name.check(tree_edmass, edmass.bars)

#resort trait dataset to the order of tree tip labels
edmass.bars <- edmass.bars[tree_edmass$tip.label, ] 

#make trait vector for absolute eye size
aveye <- as.vector(edmass.bars$abseye) 

#add tip label names to vector
names(aveye) <- edmass.bars$newtip 

#make trait vector for eye investment (PGLS residuals)
inv.mass <- as.vector(edmass.bars$inveye) #residuals of pgls
names(inv.mass) <- edmass.bars$newtip

#make trait vector of habitats
habs.edmass <- as.vector(edmass.bars$hab) 

#make vector of colors corresponding to phylogeny tips
tipcols.hab <- unname(col_hab[habs.edmass]) 


# Phylogeny with absolute eye diameters -----

#plot tree with eye size bars ant tip labels
plotTree.wBars(tree_edmass, aveye, 
               scale = 0.04, 
               tip.labels = TRUE, 
               offset = 0.35,
               ftype = "bi",
               fsize = 0.8,
               col = tipcols.hab)

#add legend for habitat states
legend(x = "left", legend = c("Aquatic", "Fossorial", "Ground-dwelling", "Scans orial", "Semiaquatic", "Subfossorial"), pch = 22, pt.cex= 2, pt.bg = col_hab, cex = 1, bty = "n", horiz = F)

# Phylogeny with relative eye size (compared to mass) -------

#plot tree with relative eye size bars
plotTree.wBars(tree_edmass, inv.mass, 
               scale = 0.7, 
               tip.labels = FALSE, 
               col = tipcols.hab,
               plot = FALSE, 
               add = TRUE)

```

Figure 3 was compiled in Illustrator, but the component parts of it are exported here. 

```{r figure3-component1}

# Separate panels for Figure 3

#export pdf of phylogeny, habitat ASR, and absolute eye size
pdf(file = "../Manuscript/Figures/Figure-3A-size.pdf", height=25, width=8)

#call plot with phylo, tip labels, and absolute eye diameters
plotTree.wBars(tree_edmass, aveye, 
               scale = 0.04, 
               width = 1,
               tip.labels = FALSE, 
               offset = 0.4,
               col = tipcols.hab)

#add legend for habitat states
legend(x  ="left", legend = c("Aquatic (Aq)", "Fossorial (Fo)", "Ground-dwelling (Gr)", "Scansorial (Sc)", "Semiaquatic (Se)", "Subfossorial (Su)"), pch = 22, pt.cex= 2, pt.bg = col_hab, cex = 1, bty = "n",horiz = F)

dev.off()

#export pdf of bars for relative eye size
pdf(file = "../Manuscript/Figures/Figure-3A-investment.pdf", height=25, width=8)

#call bars without plot or tiplabels (add to previous)
plotTree.wBars(tree_edmass, inv.mass, 
               scale = 1.5, 
               width = 1,
               tip.labels = FALSE, 
               col = tipcols.hab,
               plot = FALSE)

dev.off()

#export pdf with family names
pdf(file = "../Manuscript/Figures/Fig-4a-fams.pdf", height=25, width=8)

#call plot with phylo, tip labels, and absolute eye diameters
plot.phylo(tree_edmass,
           type = "phylogram", 
           show.tip.label = TRUE,
           tip.color = "white",
           cex = 0.7, #text size
           no.margin = TRUE, 
           use.edge.length = TRUE,
           edge.width = 1,
           label.offset = .02)

tiplabels(text = edmass.bars$family,
          adj = 0,
          bg = "white",
          frame = "none",
          cex = 0.72,
          offset = 0.04)

dev.off()
```

```{r supp-habitat, fig.height = 15, fig.width = 8, results = "hide"}

# Phylogeny for adult habitat

#export pdf of phylogeny, habitat ASR, and absolute eye size
pdf(file = "../Manuscript/Figures/Figure-S4A-size.pdf", height=20, width=8)


# Phylogeny with absolute eye diameters -------

#plot tree with eye size bars ant tip labels
plotTree.wBars(tree_edmass, aveye, 
               scale = 0.04, 
               tip.labels = TRUE, 
               offset = 0.4,
               width = 1,
               ftype = "bi",
               fsize = 0.6,
               col = tipcols.hab)

tiplabels(pch = 21, col = tipcols.hab, bg = tipcols.hab, cex = 1, offset = 0.02)

#add legend for habitat states
legend(x  ="left", legend = c("Aquatic", "Fossorial", "Ground-dwelling", "Scansorial", "Semiaquatic", "Subfossorial"), pch = 22, pt.cex= 2, pt.bg = col_hab, cex = 1, bty = "n",horiz = F)

dev.off()

# Relative eye size component of figure
pdf(file = "../Manuscript/Figures/Figure-S4A-investment.pdf", height=20, width=8)

#add bars for eye investment (relative to mass)
plotTree.wBars(tree_edmass, inv.mass, 
               scale = 0.6, 
               width = 1,
               tip.labels = FALSE, 
               col = tipcols.hab,
               plot = FALSE)
dev.off()

```

```{r supp-activity, fig.height = 15, fig.width = 8, results = "hide"}

# Phylogeny for activity pattern

#make trait vector of activity pattern
act.edmass <- as.vector(edmass.bars$act) 

#make vector of colors corresponding to phylogeny tips
tipcols.act <- unname(col_act[act.edmass]) 


#Phylogeny with absolute eye diameters for activity pattern -------

#export pdf of phylogeny and absolute eye size
pdf(file = "../Manuscript/Figures/Figure-S7A-size.pdf", height = 20, width = 8)

#plot tree with eye size bars ant tip labels
plotTree.wBars(tree_edmass, aveye, 
               scale = 0.04, 
               width = 1,
               tip.labels = TRUE, 
               offset = 0.4,
               ftype = "bi",
               fsize = 0.6,
               col = tipcols.act)

tiplabels(pch = 21, col = tipcols.act, bg = tipcols.act, cex = 1, offset = 0.02)


#add legend for habitat states
legend(x = "left", legend = c("Both", "Diurnal", "Nocturnal"), pch = 22, pt.cex = 2, pt.bg = col_act, cex = 1, bty = "n", horiz = F)

dev.off()

# Add bars for relative eye size ----

#export as pdf
pdf(file = "../Manuscript/Figures/Figure-S7A-investment.pdf", height=20, width=8)

#add bars for eye investment (relative to mass)
plotTree.wBars(tree_edmass, inv.mass, 
               scale = 0.6, 
               width = 1,
               tip.labels = FALSE, 
               col = tipcols.act,
               plot = FALSE)
dev.off()

```

```{r supp-mating, fig.height = 15, fig.width = 8, results = "hide"}

# Phylogeny for activity pattern

#make trait vector of activity pattern
mat.edmass <-as.vector(edmass.bars$mat) 

#make vector of colors corresponding to phylogeny tips
tipcols.mat <- unname(col_mat[mat.edmass]) 

# Phylogeny with absolute eye diameters for mating habitat ----

#export pdf of phylogeny, mating habitat and absolute eye size
pdf(file = "../Manuscript/Figures/Figure-S5A-size.pdf", height = 20, width = 8)

#plot tree with eye size bars ant tip labels
plotTree.wBars(tree_edmass, aveye, 
               scale = 0.04, 
               width = 1,
               tip.labels = TRUE, 
               offset = 0.4,
               ftype = "bi",
               fsize = 0.6,
               col = tipcols.mat)

tiplabels(pch = 21, col = tipcols.mat, bg = tipcols.mat, cex = 1, offset = 0.02)


#add legend for habitat states
legend(x="left",legend = c("Ground", "Lentic water", "Lotic water", "Plants"), pch = 22, pt.cex= 2, pt.bg = col_mat, cex = 1, bty = "n",horiz = F)

dev.off()

# Bars for relative eye size ----

pdf(file = "../Manuscript/Figures/Figure-S5A-investment.pdf", height=20, width=8)

#add bars for eye investment (relative to mass)
plotTree.wBars(tree_edmass, inv.mass, 
               scale = 0.6, 
               width = 1,
               tip.labels = FALSE, 
               col = tipcols.mat,
               plot = FALSE)
dev.off()
```

```{r supp-dich, fig.height = 15, fig.width = 8, results = "hide"}

# Phylogeny for sexual dichromatism

#make trait vector of activity pattern
dich.edmass <-as.vector(edmass.bars$dich) 

#make vector of colors corresponding to phylogeny tips
tipcols.dich <- unname(col_dich[dich.edmass]) 


# Phylogeny with absolute eye diameters for sex dichromatism -----

#export pdf of phylogeny, sexual dichromatism, and absolute eye size
pdf(file = "../Manuscript/Figures/Figure-S6A-size.pdf", height = 20, width = 8)

#plot tree with eye size bars ant tip labels
plotTree.wBars(tree_edmass, aveye, 
               scale = 0.04, 
               width = 1,
               tip.labels = TRUE, 
               offset = 0.4,
               ftype = "bi",
               fsize = 0.6,
               col = tipcols.dich)

tiplabels(pch = 21, col = tipcols.dich, bg = tipcols.dich, cex = 1, offset = 0.02)

#add legend for states
legend(x= "left",legend = c("Absent", "Present"), pch = 22, pt.cex = 2, pt.bg = col_dich, cex = 1, bty = "n",horiz = F)

dev.off()

# Add bars for relative eye size ----

pdf(file = "../Manuscript/Figures/Figure-S6A-investment.pdf", height=20, width=8)

#add bars for eye investment (relative to mass)
plotTree.wBars(tree_edmass, inv.mass, 
               scale = 0.6, 
               width = 1,
               tip.labels = FALSE, 
               col = tipcols.dich,
               plot = FALSE)
dev.off()

```

```{r supp-lif, fig.height = 15, fig.width = 8, results = "hide"}

# Phylogeny for life history

#make trait vector of life history
life.edmass <- as.vector(edmass.bars$lif) 

#make vector of colors corresponding to phylogeny tips
tipcols.life <- unname(col_lif[life.edmass]) 


# Phylogeny with absolute eye diameters for life history -----

#export pdf of phylogeny and absolute eye size
pdf(file = "../Manuscript/Figures/Figure-S8A-size.pdf", height=20, width=8)

#plot tree with eye size bars ant tip labels
plotTree.wBars(tree_edmass, aveye, 
               scale = 0.04, 
               width = 1,
               tip.labels = TRUE, 
               offset = 0.4,
               ftype = "bi",
               fsize = 0.6,
               col = tipcols.life)

tiplabels(pch = 21, col = tipcols.life, bg = tipcols.life, cex = 1, offset = 0.02)

#add legend for habitat states
legend(x="left",legend = c("Free-living larvae", "No free-living larvae"), pch = 22, pt.cex= 2, pt.bg = col_lif, cex = 1, bty = "n",horiz = F)

dev.off()

# Add bars for relative eye size ---

pdf(file = "../Manuscript/Figures/Figure-S8A-investment.pdf", height=20, width=8)

#add bars for eye investment (relative to mass)
plotTree.wBars(tree_edmass, inv.mass, 
               scale = 0.6, 
               width = 1,
               tip.labels = FALSE, 
               col = tipcols.life,
               plot = FALSE)
dev.off()

```

```{r supp-larv, fig.height = 15, fig.width = 8, results = "hide"}

# Phylogeny for larval habitat

#make trait vector of larval habitat
larv.edmass <- as.vector(edmass.bars$larv)

#make vector of colors corresponding to phylogeny tips
tipcols.larv <- unname(col_larv[larv.edmass]) 

#### Phylogeny with absolute eye diameters  ####

#export pdf of phylogeny and absolute eye size
pdf(file = "../Manuscript/Figures/Figure-S9A-size.pdf", height=20, width=8)

#plot tree with eye size bars ant tip labels
plotTree.wBars(tree_edmass, aveye, 
               scale = 0.04, 
               width = 1,
               tip.labels = TRUE, 
               offset = 0.4,
               ftype = "bi",
               fsize = 0.6,
               col = tipcols.larv)

tiplabels(pch = 21, col = tipcols.larv, bg = tipcols.larv, cex = 1, offset = 0.02)


#add legend for habitat states
legend(x="left", legend = c("Lentic water", "Lotic water", "No larvae", "Obscured", "Terrestrial"), pch = 22, pt.cex= 2, pt.bg = col_larv, cex = 1, bty = "n", horiz = F)

dev.off()

# Add bars for relative eye size ----

pdf(file = "../Manuscript/Figures/Figure-S9A-investment.pdf", height=20, width=8)

#add bars for eye investment (relative to mass)
plotTree.wBars(tree_edmass, inv.mass, 
               scale = 0.6, 
               width = 1,
               tip.labels = FALSE, 
               col = tipcols.larv,
               plot = FALSE)
dev.off()

```


## Eye size and investment across ecological variables 

We compared the distribution of absolute and phylogenetically corrected relative eye sizes across states for each of the ecological traits we examined. 

```{r trait-stats-dataframe}

#Combine results from assessments of eye v. mass, eye v. svl, and cornea v. eye for stats (get all the pgls residuals from the various comparisons into the same dataframe)

edmass.sub <- av.edmass.phy %>%
  select(mod_tiplabel, eye_av, pglsres.edmass)

edsvl.sub <- av.edsvl.phy %>%
  select(mod_tiplabel, eye_av, pglsres.edsvl)

edcd.sub <- av.edcd.phy %>%
  select(mod_tiplabel, cornea_av, pglsres.edcd)

#merge subset dataframes with dplyr
all.res <- edmass.sub %>%
  full_join(edsvl.sub, by = "mod_tiplabel") %>%
  full_join(edcd.sub, by = "mod_tiplabel")

#add traits to dataframe
all.res.traits <- all.res %>%
  left_join(traits, by = "mod_tiplabel")
all.res.traits <- rename(all.res.traits, "eye_av" = "eye_av.x")
```

We made boxplots to show how eye size and investment varied across states for each ecological trait. 

```{r trait-boxplots}

#### Boxplots of absolute eye diameter by trait states ####

#create color vectors for states

#create vector of colors for activity period
col_act <- c("Both" = "deeppink3",
             "Diurnal" = "darkgoldenrod1", 
             "Nocturnal" = "blueviolet",
             "Unknown" = "gray")

# create vector of colors for life history
col_mat <- c("Ground" = "wheat4", 
         "Lentic water" = "royalblue1", 
         "Lotic water" = "skyblue1",
         "Plants" = "springgreen3",
         "Unknown" = "gray")


# create vector of colors for life history
col_lif <- c("Free-living larvae" = "darkturquoise", 
         "No free-living larvae" = "black", 
         "Unknown" = "gray")

# create vector of colors for larval habitat
col_larv <- c("Lentic water" = "royalblue1", 
         "Lotic water" = "skyblue1",
         "No larvae" = "lightcoral",
         "Obscured" = "darkslateblue",
         "On land" = "lightpink4",
         "Unknown" = "gray")

# create vector of colors for sex dichromatism
col_dich <- c("Absent" = "black", 
         "Present" = "chartreuse2",
         "Unknown" = "gray")

#boxplot of eye size by adult habitat
hab.abs <- ggplot(data = filter(all.res.traits, Adult_habitat != "Unknown" & !is.na(eye_av)), 
                  aes(x = reorder(Adult_habitat, eye_av, FUN=mean), 
                      y = eye_av)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Adult_habitat, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_hab) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_x_discrete(breaks=c("Aquatic","Semiaquatic","Fossorial", "Subfossorial", "Ground-dwelling", "Scansorial"),labels=c("Aq.", "Semiaq.", "Foss.", "Subfoss.", "Ground", "Scans.")) +
  ylab("Eye diameter (mm)") +
  xlab("Adult habitat") +
  theme(legend.position = "none")

#boxplot of eye size by activity period
acper.abs <- ggplot(data = filter(all.res.traits, Activity_period != "Unknown" & !is.na(eye_av)), aes(x = reorder(Activity_period, eye_av, FUN=mean), y = eye_av)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Activity_period, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_act) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  ylab("Eye diameter (mm)") +
  xlab("Activity period") +
  theme(legend.position = "none")

#boxplot of eye size by mating habitat
mating.abs <- ggplot(data = filter(all.res.traits, Mating_habitat != "Unknown" & !is.na(eye_av)), 
                aes(x = reorder(Mating_habitat, eye_av, FUN=mean), y = eye_av)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  scale_color_manual(values = col_mat) +
  geom_jitter(aes(color = Mating_habitat, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  ylab("Mean eye diameter (mm)") +
  xlab("Mating habitat") +
  theme(legend.position = "none")

#boxplot of eye size by life history
life.abs <- ggplot(data = filter(all.res.traits, Life_history != "Unknown" & !is.na(eye_av)), 
                aes(x = reorder(Life_history, eye_av, FUN=mean), y = eye_av)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Life_history, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_lif) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  ylab("Mean eye diameter (mm)") +
  xlab("Life history") +
  theme(legend.position = "none")

#boxplot of eye size by larval habitat
larval.abs <- ggplot(data = filter(all.res.traits, Larval_habitat != "Unknown" & !is.na(eye_av)), 
                aes(x = reorder(Larval_habitat, eye_av, FUN=mean), y = eye_av)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Larval_habitat, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_larv) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  ylab("Mean eye diameter (mm)") +
  xlab("Larval habitat") +
  theme(legend.position = "none")

#boxplot of eye size by sexual dichromatism
dichrom.abs <- ggplot(data = filter(all.res.traits, Sex_dichromatism != "Unknown" & !is.na(eye_av)), 
                aes(x = reorder(Sex_dichromatism, eye_av, FUN=mean), y = eye_av)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Sex_dichromatism, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_dich) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  ylab("Mean eye diameter (mm)") +
  xlab("Sex dichromatism") +
  theme(legend.position = "none")


#### Boxplots of relative eye size (by mass) by trait states ####

#boxplot of eye investment (residuals of eye diameter v. cube root of mass PGLS) by adult habitat
hab.res.mass <- ggplot(data = filter(all.res.traits, Adult_habitat != "Unknown" & !is.na(pglsres.edmass)), 
              aes(x = reorder(Adult_habitat, pglsres.edmass, FUN = mean), y = pglsres.edmass)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Adult_habitat, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_hab, name = "Adult habitat") +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(axis.text = element_text(angle = 0, size = 9)) +
 scale_x_discrete(breaks=c("Aquatic", "Fossorial", "Subfossorial", "Ground-dwelling", "Scansorial", "Semiaquatic"), labels=c("Aq.", "Foss.", "Subfoss.","Ground", "Scans.", "Semiaq.")) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (mass)") +
  xlab("Adult habitat") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. cube root of mass PGLS) by activity period
acper.res.mass <- ggplot(data = filter(all.res.traits, Activity_period != "Unknown" & !is.na(pglsres.edmass)), 
                aes(x = reorder(Activity_period, pglsres.edmass, FUN=mean), y = pglsres.edmass)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Activity_period, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_act) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (mass)") +
  xlab("Activity period") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. cube root of mass PGLS) by mating habitat
mating.res.mass <- ggplot(data = filter(all.res.traits, Mating_habitat != "Unknown" & !is.na(pglsres.edmass)), 
                aes(x = reorder(Mating_habitat, pglsres.edmass, FUN=mean), y = pglsres.edmass)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  scale_color_manual(values = col_mat) +
  geom_jitter(aes(color = Mating_habitat, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (mass)") +
  xlab("Mating habitat") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. cube root of mass PGLS) by life history
life.res.mass <- ggplot(data = filter(all.res.traits, Life_history != "Unknown" & !is.na(pglsres.edmass)), 
                aes(x = reorder(Life_history, pglsres.edmass, FUN=mean), y = pglsres.edmass)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) + 
  geom_jitter(aes(color = Life_history, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_lif) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (mass)") +
  xlab("Life history") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. cube root of mass PGLS) by larval habitat
larval.res.mass <- ggplot(data = filter(all.res.traits, Larval_habitat != "Unknown" & !is.na(pglsres.edmass)), 
                aes(x = reorder(Larval_habitat, pglsres.edmass, FUN=mean), y = pglsres.edmass)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Larval_habitat, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_larv) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                    labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (mass)") +
  xlab("Larval habitat") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. cube root of mass PGLS) by sexual dichromatism
dichrom.res.mass <- ggplot(data = filter(all.res.traits, Sex_dichromatism != "Unknown" & !is.na(pglsres.edmass)), 
                aes(x = reorder(Sex_dichromatism, pglsres.edmass, FUN=mean), y = pglsres.edmass)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Sex_dichromatism, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_dich) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (mass)") +
  xlab("Sex dichromatism") +
  theme(legend.position = "none")


#### Boxplots of relative eye size (by SVL) by trait states ####

#boxplot of eye investment (residuals of eye diameter v. SVL PGLS) by adult habitat
hab.res.svl <- ggplot(data = filter(all.res.traits, Adult_habitat != "Unknown" & !is.na(pglsres.edsvl)), 
              aes(x = reorder(Adult_habitat, pglsres.edsvl, FUN=mean), y = pglsres.edsvl)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Adult_habitat, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_hab, name = "Adult habitat") +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(axis.text = element_text(angle = 0, size = 9)) +
 scale_x_discrete(breaks=c("Aquatic","Semiaquatic","Fossorial", "Subfossorial", "Ground-dwelling", "Scansorial"), labels=c("Aq.", "Semiaq.", "Foss.", "Subfoss.", "Ground", "Scans.")) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (SVL)") +
  xlab("Adult habitat") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. SVL PGLS) by activity period
acper.res.svl <- ggplot(data = filter(all.res.traits, Activity_period != "Unknown" & !is.na(pglsres.edsvl)), 
                aes(x = reorder(Activity_period, pglsres.edsvl, FUN=mean), y = pglsres.edsvl)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Activity_period, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_act) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (SVL)") +
  xlab("Activity period") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. SVL PGLS) by mating habitat
mating.res.svl <- ggplot(data = filter(all.res.traits, Mating_habitat != "Unknown" & !is.na(pglsres.edsvl)), 
                aes(x = reorder(Mating_habitat, pglsres.edsvl, FUN=mean), y = pglsres.edsvl)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  scale_color_manual(values = col_mat) +
  geom_jitter(aes(color = Mating_habitat, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (SVL)") +
  xlab("Mating habitat") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. SVL PGLS) by life history
life.res.svl <- ggplot(data = filter(all.res.traits, Life_history != "Unknown" & !is.na(pglsres.edsvl)), 
                aes(x = reorder(Life_history, pglsres.edsvl, FUN=mean), y = pglsres.edsvl)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) + 
  geom_jitter(aes(color = Life_history, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_lif) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (SVL)") +
  xlab("Life history") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. SVL PGLS) by larval habitat
larval.res.svl <- ggplot(data = filter(all.res.traits, Larval_habitat != "Unknown" & !is.na(pglsres.edsvl)), 
                aes(x = reorder(Larval_habitat, pglsres.edsvl, FUN=mean), y = pglsres.edsvl)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Larval_habitat, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_larv) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (SVL)") +
  xlab("Larval habitat") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. SVL PGLS) by sexual dichromatism
dichrom.res.svl <- ggplot(data = filter(all.res.traits, Sex_dichromatism != "Unknown" & !is.na(pglsres.edsvl)), 
                aes(x = reorder(Sex_dichromatism, pglsres.edsvl, FUN=mean), y = pglsres.edsvl)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Sex_dichromatism, text = genus_species), 
              shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_dich) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  ylab("Eye investment (SVL)") +
  xlab("Sex dichromatism") +
  theme(legend.position = "none")

#### Boxplots of relative corneal size (to eye size) by trait states ####

#boxplot of corneal investment (residuals of cornea diameter v. eye diameter PGLS) by adult habitat
hab.res.cor <- ggplot(data = filter(all.res.traits, Adult_habitat != "Unknown" & !is.na(pglsres.edcd)), 
              aes(x = reorder(Adult_habitat, pglsres.edcd, FUN = mean), y = pglsres.edcd)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Adult_habitat, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_hab, name = "Adult habitat") +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(axis.text = element_text(angle = 0, size = 9)) +
 scale_x_discrete(breaks=c("Aquatic", "Fossorial", "Subfossorial", "Ground-dwelling", "Scansorial", "Semiaquatic"), labels=c("Aq.", "Foss.", "Subfoss.","Ground", "Scans.", "Semiaq.")) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0.0, 0.1, 0.2),
                     labels = c("0.63x","0.79x", "1.0x", "1.3x","1.6x")) +
  ylab("Corneal investment") +
  xlab("Adult habitat") +
  theme(legend.position = "none")

#boxplot of corneal investment (residuals of cornea diameter v. eye diameter PGLS) by activity period
acper.res.cor <- ggplot(data = filter(all.res.traits, Activity_period != "Unknown" & !is.na(pglsres.edcd)), 
                aes(x = reorder(Activity_period, pglsres.edcd, FUN=mean), y = pglsres.edcd)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Activity_period, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_act) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0.0, 0.1, 0.2),
                     labels = c("0.63x","0.79x", "1.0x", "1.3x","1.6x")) +
  ylab("Corneal investment") +
  xlab("Activity period") +
  theme(legend.position = "none")

#boxplot of corneal investment (residuals of cornea diameter v. eye diameter PGLS) by mating habitat
mating.res.cor <- ggplot(data = filter(all.res.traits, Mating_habitat != "Unknown" & !is.na(pglsres.edcd)), 
                aes(x = reorder(Mating_habitat, pglsres.edcd, FUN=mean), y = pglsres.edcd)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Mating_habitat, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_mat) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0.0, 0.1, 0.2),
                     labels = c("0.63x","0.79x", "1.0x", "1.3x","1.6x")) +
  ylab("Corneal investment") +
  xlab("Mating habitat") +
  theme(legend.position = "none")

#boxplot of corneal investment (residuals of cornea diameter v. eye diameter PGLS) by life history
life.res.cor <- ggplot(data = filter(all.res.traits, Life_history != "Unknown" & !is.na(pglsres.edcd)), 
                aes(x = reorder(Life_history, pglsres.edcd, FUN=mean), y = pglsres.edcd)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Life_history, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_lif) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0.0, 0.1, 0.2),
                     labels = c("0.63x","0.79x", "1.0x", "1.3x","1.6x")) +
  ylab("Corneal investment") +
  xlab("Life history") +
  theme(legend.position = "none")

#boxplot of corneal investment (residuals of cornea diameter v. eye diameter PGLS) by larval habitat
larval.res.cor <- ggplot(data = filter(all.res.traits, Larval_habitat != "Unknown" & !is.na(pglsres.edcd)), 
                aes(x = reorder(Larval_habitat, pglsres.edcd, FUN=mean), y = pglsres.edcd)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Larval_habitat, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_larv) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0.0, 0.1, 0.2),
                     labels = c("0.63x","0.79x", "1.0x", "1.3x","1.6x")) +
  ylab("Corneal investment") +
  xlab("Larval_habitat") +
  theme(legend.position = "none")

#boxplot of corneal investment (residuals of cornea diameter v. eye diameter PGLS) by sexual dichromatism
dichrom.res.cor <- ggplot(data = filter(all.res.traits, Sex_dichromatism != "Unknown" & !is.na(pglsres.edcd)), 
                aes(x = reorder(Sex_dichromatism, pglsres.edcd, FUN=mean), y = pglsres.edcd)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Sex_dichromatism, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_dich) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
 scale_y_continuous(breaks = c(-0.2, -0.1, 0.0, 0.1, 0.2),
                     labels = c("0.63x","0.79x", "1.0x", "1.3x","1.6x")) +
  ylab("Corneal investment") +
  xlab("Sex dichromatism") +
  theme(legend.position = "none")
```

Then we plotted all of the absolute eye sizes and residuals from both mass and SVL to look for differences in distributions among ecological states. 

```{r interactive-boxplots, eval = FALSE}

print(ggplotly(hab.abs))
ggplotly(hab.res.mass)
ggplotly(hab.res.svl)
ggplotly(hab.res.cor)

ggplotly(acper.abs)
ggplotly(acper.res.mass)
ggplotly(acper.res.svl)
ggplotly(acper.res.cor)

ggplotly(mating.abs)
ggplotly(mating.res.mass)
ggplotly(life.abs)
ggplotly(life.res.mass)
ggplotly(larval.abs)
ggplotly(larval.res.mass)
ggplotly(dichrom.abs)
ggplotly(dichrom.res.mass)
```

```{r fig-boxplots, fig.cap = "Absolute eye size and eye investment by trait categories across anurans, with investment determined by both SVL and mass. Black bars indicate the median and black diamonds the mean for each trait category.", fig.height = 30, fig.width = 15}

#edit ggplots to fit into panels of one figure
a <- hab.abs +
  ggtitle("Adult habitat (eye size)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

b <- hab.res.mass +
  ggtitle("Adult habitat (inv. mass)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

c <- hab.res.svl +
  ggtitle("Adult habitat (inv. svl)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

d <- hab.res.cor +
  ggtitle("Adult habitat (cornea inv.)") +
theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

e <- acper.abs +
  ggtitle("Adult activity period (eye size)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

f <- acper.res.mass +
  ggtitle("Adult activity period (inv. mass)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

g <- acper.res.svl +
  ggtitle("Adult activity period (inv. svl)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

h <- acper.res.cor +
  ggtitle("Adult activity period (cornea inv.)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

i <- mating.abs +
  ggtitle("Mating habitat (eye size)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

j <- mating.res.mass +
  ggtitle("Mating habitat (mass inv.)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

k <- mating.res.svl +
  ggtitle("Mating habitat (svl. inv)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

l <- mating.res.cor +
  ggtitle("Mating habitat (cornea inv.)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")
  
m <- life.abs +
  ggtitle("Life history strategy (eye size)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")
  
n <- life.res.mass +
  ggtitle("Life history strategy (inv. mass)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

o <- life.res.svl +
  ggtitle("Life history strategy (inv. svl)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

p <- life.res.cor +
  ggtitle("Life history strategy (cornea inv.)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

q <- larval.abs +
  ggtitle("Larval habitat (eye size)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

r <- larval.res.mass +
  ggtitle("Larval habitat (inv. mass)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

s <- larval.res.svl +
  ggtitle("Larval habitat (inv. svl)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

t <- larval.res.cor +
  ggtitle("Larval habitat (cornea inv.)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")
  
u <- dichrom.abs +
  ggtitle("Sexual dichromatism (eye size)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

v <- dichrom.res.mass +
  ggtitle("Sexual dichromatism (inv. mass)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

w <- dichrom.res.svl +
  ggtitle("Sexual dichromatism (inv. svl)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")

x <- dichrom.res.cor +
  ggtitle("Sexual dichromatism (cornea inv.)") +
  theme(legend.position = "none", text = element_text(size=10), plot.title = element_text(size=10)) +
  xlab("")
 

#use cowplot package to nicely plot the graphs together with a shared common legend.
plots <- plot_grid(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, #list of plots to arrange in grid
                    align = 'vh', #align horizontally and vertically
                    labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X"), #panel labels for figure
                    hjust = 0, #adjustment for panel labels
                    nrow = 6) #number of rows in grids
 

#plot figure
plot_grid(plots)

```

The boxplots for eye size and investment across adult habitats also went into Figure 3, so we export those components here. 

```{r fig3-bc, results = "hide"}

#export panels of eye size by habitat, eye investment by habitat (boxplots) 

#make panels
fig.b <- hab.abs + 
  scale_x_discrete(breaks=c("Aquatic", "Fossorial", "Subfossorial", "Ground-dwelling", "Scansorial", "Semiaquatic"), labels=c("Aq", "Fo", "Su","Gr", "Sc", "Se")) +
  theme(legend.position = "none", axis.title=element_text(size=14), axis.text = element_text(size = 12))+
  ylab("Eye diameter (mm)") +
  xlab("Adult habitat")

fig.c <- hab.res.mass +
  scale_x_discrete(breaks=c("Fossorial", "Aquatic", "Subfossorial", "Ground-dwelling", "Semiaquatic", "Scansorial"), labels=c("Fo", "Aq", "Su","Gr", "Se", "Sc")) +
  theme(legend.position = "none", axis.title=element_text(size=14), axis.text = element_text(size = 12)) +
  ylab("Eye investment") +
  xlab("Adult habitat") +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c(".25x", ".40x", ".63x", "1.0x", "1.6x"))
  

#use cowplot package to nicely plot the graphs together with a shared common legend.
bc <- plot_grid(fig.b, fig.c, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("B", "C"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 1) #number of rows in grids


#export pdf of figure
pdf(file = "../Manuscript/Figures/Figure-3BC.pdf", height=3.4, width=7)
plot_grid(bc)
dev.off()
```

```{r fig4-boxplots, results = "hide"}

#boxplots for mating habitat and activity period

#make panels
fig.a <- ggplot(data = filter(all.res.traits, Mating_habitat != "Unknown" & !is.na(pglsres.edmass)), 
                aes(x = reorder(Mating_habitat, pglsres.edmass, FUN=mean), y = pglsres.edmass)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  geom_jitter(aes(color = Adult_habitat), shape = 19, size = 1.3, alpha = 0.7, position = position_jitter(0.2)) +
  scale_color_manual(values = col_hab) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  theme(text = element_text(size=12), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 10)) +
  ylab("Eye investment") +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.63x", "1.0x", "1.6x")) +
  xlab("Mating habitat") +
  scale_x_discrete(breaks = c("Ground", "Lentic water","Lotic water","Plants"),
                  labels = c("Ground", "Lentic","Lotic","Plants")) +
  theme(legend.position = "none")

fig.b <- ggplot(data = filter(all.res.traits, Activity_period != "Unknown" & !is.na(pglsres.edmass)), 
                aes(x = reorder(Activity_period, pglsres.edmass, FUN=mean), y = pglsres.edmass)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  geom_jitter(aes(color = Adult_habitat, text = genus_species), shape = 19, size = 1.3, alpha = 0.7, position = position_jitter(0.2)) +
  scale_color_manual(values = col_hab) +
  stat_summary(fun.y = mean, colour="black", geom="point", 
               shape=18, size=3, show_guide = FALSE) +
  theme(text = element_text(size=12), panel.background = element_blank(), axis.line = element_line(colour = "black"), axis.text = element_text(angle = 0, size = 10), axis.text.y = element_blank(), axis.ticks.y = element_blank(), axis.line.y = element_blank(), axis.title.y = element_blank(),legend.position = "none") +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2)) +
  xlab("Activity Period")

# extract legends from habitat figure
leg <- get_legend(plot_pgls.edmass + theme(legend.position = "right"))
  
#use cowplot package to nicely plot the graphs together with a shared common legend.
plots <- plot_grid(fig.a, fig.b, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("A","B"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           ncol = 2, #number of rows in grids
           rel_widths = c(1.6,1.3))

#export pdf of figure
pdf(file = "../Manuscript/Figures/Figure-4.pdf", height = 3.5, width = 7)
plot_grid(plots, leg, ncol = 2, rel_widths = c(1, 0.22))
dev.off()
```

```{r suppfigs-traitboxplots, results = "hide"}

#export boxplots by trait

#Adult habitat
#make panels
fig.b <- hab.abs + 
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank())

fig.c <- hab.res.mass +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
fig.d <- hab.res.svl +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank())

fig.e <- hab.res.cor +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank())


#use cowplot package to nicely plot the graphs together with a shared common legend.
bcde <- plot_grid(fig.b, fig.c, fig.d, fig.e, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("B", "C", "D", "E"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 1) #number of rows in grids

#export pdf of figure
pdf(file = "../Manuscript/Figures/Figure-S4BCDE.pdf", height=3, width=12)
plot_grid(bcde)
dev.off()

#Activity pattern

#make panels
fig.b <- acper.abs + 
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank())

fig.c <- acper.res.mass +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
fig.d <- acper.res.svl +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
fig.e <- acper.res.cor +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
#use cowplot package to nicely plot the graphs together with a shared common legend.
bcde <- plot_grid(fig.b, fig.c, fig.d, fig.e, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("B", "C", "D", "E"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 1) #number of rows in grids

#export pdf of figure
pdf(file = "../Manuscript/Figures/Figure-S7BCDE.pdf", height=3, width=12)
plot_grid(bcde)
dev.off()

#Mating habitat

#make panels
fig.b <- mating.abs + 
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank())

fig.c <- mating.res.mass +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
fig.d <- mating.res.svl +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
fig.e <- mating.res.cor +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
#use cowplot package to nicely plot the graphs together with a shared common legend.
bcde <- plot_grid(fig.b, fig.c, fig.d, fig.e, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("B", "C", "D", "E"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 1) #number of rows in grids

#export pdf of figure
pdf(file = "../Manuscript/Figures/Figure-S5BCDE.pdf", height=3, width=12)
plot_grid(bcde)
dev.off()

#Sexual dichromatism

#make panels
fig.b <- dichrom.abs + 
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank())

fig.c <- dichrom.res.mass +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
fig.d <- dichrom.res.svl +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
fig.e <- dichrom.res.cor +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
#use cowplot package to nicely plot the graphs together with a shared common legend.
bcde <- plot_grid(fig.b, fig.c, fig.d, fig.e, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("B", "C", "D", "E"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 1) #number of rows in grids

#export pdf of figure
pdf(file = "../Manuscript/Figures/Figure-S6BCDE.pdf", height=3, width=12)
plot_grid(bcde)
dev.off()

#Life History

#make panels
fig.b <- life.abs + 
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank())

fig.c <- life.res.mass +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
fig.d <- life.res.svl +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
fig.e <- life.res.cor +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
#use cowplot package to nicely plot the graphs together with a shared common legend.
bcde <- plot_grid(fig.b, fig.c, fig.d, fig.e, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("B", "C", "D", "E"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 1) #number of rows in grids

#export pdf of figure
pdf(file = "../Manuscript/Figures/Figure-S8BCDE.pdf", height=3, width=12)
plot_grid(bcde)
dev.off()

#Larval habitat

#make panels
fig.b <- larval.abs + 
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank())

fig.c <- larval.res.mass +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
fig.d <- larval.res.svl +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
fig.e <- larval.res.cor +
  theme(legend.position = "none", axis.title=element_text(size=12), axis.text.x=element_blank(), axis.ticks.x = element_blank()) 
#use cowplot package to nicely plot the graphs together with a shared common legend.
bcde <- plot_grid(fig.b, fig.c, fig.d, fig.e, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("B", "C", "D", "E"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 1) #number of rows in grids

#export pdf of figure
pdf(file = "../Manuscript/Figures/Figure-S9BCDE.pdf", height=3, width=12)
plot_grid(bcde)
dev.off()
```

## Testing for differences in eye size across ecological states

Data for eye size and eye investment across ecological groups were not normally distributed, so we used non-parametric statistical tests for comparing our data between ecological groups. 

We used Kruskal-Wallis tests to test for differences in means across ecological groups, which is a non-parametric alternative to a one-way ANOVA test that uses ranks. It extends the two-sample Wilcoxon test to more than two groups. It’s recommended when the assumptions of one-way ANOVA test are not met, which was true across most of our data. As for the Wilcoxon test (or Mann-Whitney test) with two samples, this test converts the response values to ranks, and tests whether the ranks are distributed equally across the conditions, as would be expected under the null hypothesis.

When the Kruskal-Wallis test indicated that means differed across ecological groups and there were more than two groups, we used post-hoc Pairwise Wilcoxen Rank Sum Tests to caluclate pairwise comparisons between group levels with corrections for mulitple testing. This is a non-parametric alternative to a Tukey test that uses ranks to test for differences in means.

#### Adult habitat

```{r trait-stats-habitat}

#Habitat

# Dataframe with just habitat, eye size, residuals, with "Unknown" observations removed
hab.df <- all.res.traits %>% 
  select(mod_tiplabel, Adult_habitat, eye_av.y, pglsres.edmass, pglsres.edsvl, pglsres.edcd) %>% 
  rename(eye_av = eye_av.y) %>%
  mutate(Adult_habitat = as.factor(Adult_habitat)) %>%
  filter(Adult_habitat != "Unknown")

#Summary stats
hab.stats <- ddply(hab.df, ~Adult_habitat, summarise, 
                   mean.eye=mean(eye_av, na.rm = TRUE), 
                   sd.eye=sd(eye_av, na.rm = TRUE), 
                   n.eye=length(eye_av[!is.na(eye_av)]), 
                   mean.edmass=mean(pglsres.edmass, na.rm = TRUE),
                   sd.edmass=sd(pglsres.edmass, na.rm = TRUE),
                   n.edmass=length(pglsres.edmass[!is.na(pglsres.edmass)]), 
                   mean.edsvl=mean(pglsres.edsvl, na.rm = TRUE),
                   sd.edsvl=sd(pglsres.edsvl, na.rm = TRUE),
                   n.edsvl=length(pglsres.edsvl[!is.na(pglsres.edsvl)]),
                   mean.edcd=mean(pglsres.edcd, na.rm = TRUE),
                   sd.pglsres.edcd=sd(pglsres.edcd, na.rm = TRUE),
                   n.edcd=length(pglsres.edcd[!is.na(pglsres.edcd)]))

print(hab.stats)

# Test for differences in mean (absolute) eye size across habitat groups: Kruskal-Wallis test
kruskal.test(eye_av ~ Adult_habitat, data = hab.df) 

# Mulitple comparisons for differences in mean (absolute) eye size across habitat groups: Pairwise Wilcoxon rank sum tests
pairwise.wilcox.test(hab.df$eye_av, hab.df$Adult_habitat, p.adjust.method = "bonferroni")

# Test for differences relative eye size (by mass) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edmass ~ Adult_habitat, data = hab.df) 

# Mulitple comparisons for differences in eye investment (by mass) across habitat groups: Pairwise Wilcoxon rank sum tests
pairwise.wilcox.test(hab.df$pglsres.edmass, hab.df$Adult_habitat, p.adjust.method = "bonferroni")

# Test for differences in eye investment (by svl) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edsvl ~ Adult_habitat, data = hab.df) 

# Mulitple comparisons for differences in eye investment (by svl) across habitat groups: Pairwise Wilcoxon rank sum tests
pairwise.wilcox.test(hab.df$pglsres.edsvl, hab.df$Adult_habitat, p.adjust.method = "bonferroni")

# Test for differences in relative corneal size (compared to eye size) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edcd ~ Adult_habitat, data = hab.df) 

```

#### Activity period

```{r trait-stats-acper}

# Activity period

# Dataframe with just activity period, eye size, residuals, with "Unknown" observations removed
act.df <- all.res.traits %>% 
  select(mod_tiplabel, Activity_period, eye_av.y, pglsres.edmass, pglsres.edsvl, pglsres.edcd) %>% 
  rename(eye_av = eye_av.y) %>%
  mutate(Activity_period = as.factor(Activity_period)) %>%
  filter(Activity_period != "Unknown")

#Summary stats
act.stats <- ddply(act.df, ~Activity_period, summarise,
                   mean.eye=mean(eye_av, na.rm = TRUE), 
                   sd.eye=sd(eye_av, na.rm = TRUE), 
                   n.eye=length(eye_av[!is.na(eye_av)]), 
                   mean.edmass=mean(pglsres.edmass, na.rm = TRUE),
                   sd.edmass=sd(pglsres.edmass, na.rm = TRUE),
                   n.edmass=length(pglsres.edmass[!is.na(pglsres.edmass)]), 
                   mean.edsvl=mean(pglsres.edsvl, na.rm = TRUE),
                   sd.edsvl=sd(pglsres.edsvl, na.rm = TRUE),
                   n.edsvl=length(pglsres.edsvl[!is.na(pglsres.edsvl)]),
                   mean.edcd=mean(pglsres.edcd, na.rm = TRUE),
                   sd.pglsres.edcd=sd(pglsres.edcd, na.rm = TRUE),
                   n.edcd=length(pglsres.edcd[!is.na(pglsres.edcd)]))

print(act.stats)

# Test for differences in mean (absolute) eye size across activity period groups: Kruskal-Wallis test
kruskal.test(eye_av ~ Activity_period, data = act.df) 

# Mulitple comparisons for differences in mean (absolute) eye size across activity period groups: Pairwise Wilcoxon rank sum tests
pairwise.wilcox.test(act.df$eye_av, act.df$Activity_period, p.adjust.method = "bonferroni")

# Test for differences in eye investment (by mass) across activity period groups: Kruskal-Wallis test
kruskal.test(pglsres.edmass ~ Activity_period, data = act.df) 

# Test for differences in eye investment (by SVL) across activity period groups: Kruskal-Wallis test
kruskal.test(pglsres.edsvl ~ Activity_period, data = act.df) 

# Test for differences in relative corneal size (compared to eye size) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edcd ~ Activity_period, data = act.df) 

```

#### Mating habitat

```{r trait-stats-mating}

# Dataframe with just mating habitat, eye size, residuals, with "Unknown" observations removed
mat.df <- all.res.traits %>% 
  select(mod_tiplabel, Mating_habitat, eye_av.y, pglsres.edmass, pglsres.edsvl, pglsres.edcd) %>% 
  rename(eye_av = eye_av.y) %>%
  mutate(Mating_habitat = as.factor(Mating_habitat)) %>%
  filter(Mating_habitat != "Unknown")

#Summary stats
mat.stats <- ddply(mat.df, ~Mating_habitat, summarise,
                   mean.eye=mean(eye_av, na.rm = TRUE), 
                   sd.eye=sd(eye_av, na.rm = TRUE), 
                   n.eye=length(eye_av[!is.na(eye_av)]), 
                   mean.edmass=mean(pglsres.edmass, na.rm = TRUE),
                   sd.edmass=sd(pglsres.edmass, na.rm = TRUE),
                   n.edmass=length(pglsres.edmass[!is.na(pglsres.edmass)]), 
                   mean.edsvl=mean(pglsres.edsvl, na.rm = TRUE),
                   sd.edsvl=sd(pglsres.edsvl, na.rm = TRUE),
                   n.edsvl=length(pglsres.edsvl[!is.na(pglsres.edsvl)]),
                   mean.edcd=mean(pglsres.edcd, na.rm = TRUE),
                   sd.pglsres.edcd=sd(pglsres.edcd, na.rm = TRUE),
                   n.edcd=length(pglsres.edcd[!is.na(pglsres.edcd)]))

print(mat.stats)

# Test for differences in mean (absolute) eye size across mating habitat groups: Kruskal-Wallis test
kruskal.test(eye_av ~ Mating_habitat, data = mat.df) 

# Mulitple comparisons for differences in mean (absolute) eye size across mating habitat groups: Pairwise Wilcoxon rank sum tests
pairwise.wilcox.test(mat.df$eye_av, mat.df$Mating_habitat, p.adjust.method = "bonferroni")

# Test for differences in eye investment (by mass) across mating habitat: Kruskal-Wallis test
kruskal.test(pglsres.edmass ~ Mating_habitat, data = mat.df) 

# Mulitple comparisons for differences in eye investment (by mass) across mating habitat groups: Pairwise Wilcoxon rank sum tests
pairwise.wilcox.test(mat.df$pglsres.edmass, mat.df$Mating_habitat, p.adjust.method = "bonferroni")

# Test for differences in eye investment (by SVL) across mating habitat: Kruskal-Wallis test
kruskal.test(pglsres.edsvl ~ Mating_habitat, data = mat.df) 

# Mulitple comparisons for differences in eye investment (by SVL) across mating habitat groups: Pairwise Wilcoxon rank sum tests
pairwise.wilcox.test(mat.df$pglsres.edsvl, mat.df$Mating_habitat, p.adjust.method = "bonferroni")

# Test for differences in relative corneal size (compared to eye size) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edcd ~ Mating_habitat, data = mat.df) 
```

```{r mating-subset, include = FALSE}

#subset to only ground-dwelling species and then test again. 

#Mating habitat
mat.ground <- all.res.traits %>% 
  filter(Adult_habitat == "Ground-dwelling") %>%
  select(mod_tiplabel, Mating_habitat, eye_av.y, pglsres.edmass, pglsres.edsvl, pglsres.edcd) %>% 
  rename(eye_av = eye_av.y) %>%
  mutate(Mating_habitat = as.factor(Mating_habitat)) %>%
  filter(Mating_habitat != "Unknown")

# Test for differences in eye investment (by mass) across mating habitat: Kruskal-Wallis test
kruskal.test(pglsres.edmass ~ Mating_habitat, data = mat.ground) 

#Summary stats (ground dwelling only)
mat.stats.ground <- ddply(mat.ground, ~Mating_habitat, summarise,
                   mean.eye=mean(eye_av, na.rm = TRUE), 
                   sd.eye=sd(eye_av, na.rm = TRUE), 
                   n.eye=length(eye_av[!is.na(eye_av)]), 
                   mean.edmass=mean(pglsres.edmass, na.rm = TRUE),
                   sd.edmass=sd(pglsres.edmass, na.rm = TRUE),
                   n.edmass=length(pglsres.edmass[!is.na(pglsres.edmass)]), 
                   mean.edsvl=mean(pglsres.edsvl, na.rm = TRUE),
                   sd.edsvl=sd(pglsres.edsvl, na.rm = TRUE),
                   n.edsvl=length(pglsres.edsvl[!is.na(pglsres.edsvl)]),
                   mean.edcd=mean(pglsres.edcd, na.rm = TRUE),
                   sd.pglsres.edcd=sd(pglsres.edcd, na.rm = TRUE),
                   n.edcd=length(pglsres.edcd[!is.na(pglsres.edcd)]))

print(mat.stats.ground)

```

#### Life history strategy

```{r trait-stats-lifehistory}

#Mating habitat

# Dataframe with just mating habitat, eye size, residuals, with "Unknown" observations removed
lif.df <- all.res.traits %>% 
  select(mod_tiplabel, Life_history, eye_av.y, pglsres.edmass, pglsres.edsvl, pglsres.edcd) %>% 
  rename(eye_av = eye_av.y) %>%
  mutate(Life_history = as.factor(Life_history)) %>%
  filter(Life_history != "Unknown")

#Summary stats
lif.stats <- ddply(lif.df, ~Life_history, summarise,
                   mean.eye=mean(eye_av, na.rm = TRUE), 
                   sd.eye=sd(eye_av, na.rm = TRUE), 
                   n.eye=length(eye_av[!is.na(eye_av)]), 
                   mean.edmass=mean(pglsres.edmass, na.rm = TRUE),
                   sd.edmass=sd(pglsres.edmass, na.rm = TRUE),
                   n.edmass=length(pglsres.edmass[!is.na(pglsres.edmass)]), 
                   mean.edsvl=mean(pglsres.edsvl, na.rm = TRUE),
                   sd.edsvl=sd(pglsres.edsvl, na.rm = TRUE),
                   n.edsvl=length(pglsres.edsvl[!is.na(pglsres.edsvl)]),
                   mean.edcd=mean(pglsres.edcd, na.rm = TRUE),
                   sd.pglsres.edcd=sd(pglsres.edcd, na.rm = TRUE),
                   n.edcd=length(pglsres.edcd[!is.na(pglsres.edcd)]))

print(lif.stats)

# Test for differences in mean (absolute) eye size across mating habitat groups: Kruskal-Wallis test
kruskal.test(eye_av ~ Life_history, data = lif.df) 

# Test for differences in eye investment (by mass) across mating habitat: Kruskal-Wallis test
kruskal.test(pglsres.edmass ~ Life_history, data = lif.df) 

# Test for differences in eye investment (by SVL) across mating habitat: Kruskal-Wallis test
kruskal.test(pglsres.edsvl ~ Life_history, data = lif.df) 

# Test for differences in relative corneal size (compared to eye size) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edcd ~ Life_history, data = lif.df)
```

#### Larval habitat

```{r trait-stats-larhab}

#Larval habitat

# Dataframe with just mating habitat, eye size, residuals, with "Unknown" observations removed
lar.df <- all.res.traits %>% 
  select(mod_tiplabel, Larval_habitat, eye_av.y, pglsres.edmass, pglsres.edsvl, pglsres.edcd) %>% 
  rename(eye_av = eye_av.y) %>%
  mutate(Larval_habitat = as.factor(Larval_habitat)) %>%
  filter(Larval_habitat != "Unknown")

#Summary stats
lar.stats <- ddply(lar.df, ~Larval_habitat, summarise,
                   mean.eye=mean(eye_av, na.rm = TRUE), 
                   sd.eye=sd(eye_av, na.rm = TRUE), 
                   n.eye=length(eye_av[!is.na(eye_av)]), 
                   mean.edmass=mean(pglsres.edmass, na.rm = TRUE),
                   sd.edmass=sd(pglsres.edmass, na.rm = TRUE),
                   n.edmass=length(pglsres.edmass[!is.na(pglsres.edmass)]), 
                   mean.edsvl=mean(pglsres.edsvl, na.rm = TRUE),
                   sd.edsvl=sd(pglsres.edsvl, na.rm = TRUE),
                   n.edsvl=length(pglsres.edsvl[!is.na(pglsres.edsvl)]),
                   mean.edcd=mean(pglsres.edcd, na.rm = TRUE),
                   sd.pglsres.edcd=sd(pglsres.edcd, na.rm = TRUE),
                   n.edcd=length(pglsres.edcd[!is.na(pglsres.edcd)]))

print(lar.stats)

# Test for differences in mean (absolute) eye size across mating habitat groups: Kruskal-Wallis test
kruskal.test(eye_av ~ Larval_habitat, data = lar.df) 

# Test for differences in eye investment (by mass) across mating habitat: Kruskal-Wallis test
kruskal.test(pglsres.edmass ~ Larval_habitat, data = lar.df) 

# Test for differences in eye investment (by SVL) across mating habitat: Kruskal-Wallis test
kruskal.test(pglsres.edsvl ~ Larval_habitat, data = lar.df) 

# Test for differences in relative corneal size (compared to eye size) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edcd ~ Larval_habitat, data = lar.df) 
```

```{r trait-stats-larhab2}

#Redo larval habitat statistics after removing all species that do not have larvae (because that's just repeating the life history question again, this should be subset to only species with larvae)

# Dataframe with just larval habitat, eye size, residuals, with "Unknown" observations removed
lar.df2 <- all.res.traits %>% 
  select(mod_tiplabel, Larval_habitat, eye_av.y, pglsres.edmass, pglsres.edsvl, pglsres.edcd) %>% 
  rename(eye_av = eye_av.y) %>%
  mutate(Larval_habitat = as.factor(Larval_habitat)) %>%
  filter(Larval_habitat != "Unknown", Larval_habitat != "No larvae")

#Summary stats
lar.stats2 <- ddply(lar.df2, ~Larval_habitat, summarise,
                   mean.eye=mean(eye_av, na.rm = TRUE), 
                   sd.eye=sd(eye_av, na.rm = TRUE), 
                   n.eye=length(eye_av[!is.na(eye_av)]), 
                   mean.edmass=mean(pglsres.edmass, na.rm = TRUE),
                   sd.edmass=sd(pglsres.edmass, na.rm = TRUE),
                   n.edmass=length(pglsres.edmass[!is.na(pglsres.edmass)]), 
                   mean.edsvl=mean(pglsres.edsvl, na.rm = TRUE),
                   sd.edsvl=sd(pglsres.edsvl, na.rm = TRUE),
                   n.edsvl=length(pglsres.edsvl[!is.na(pglsres.edsvl)]),
                   mean.edcd=mean(pglsres.edcd, na.rm = TRUE),
                   sd.pglsres.edcd=sd(pglsres.edcd, na.rm = TRUE),
                   n.edcd=length(pglsres.edcd[!is.na(pglsres.edcd)]))

print(lar.stats2)

# Test for differences in mean (absolute) eye size across mating habitat groups: Kruskal-Wallis test
kruskal.test(eye_av ~ Larval_habitat, data = lar.df2) 

# Test for differences in eye investment (by mass) across mating habitat: Kruskal-Wallis test
kruskal.test(pglsres.edmass ~ Larval_habitat, data = lar.df2) 

# Test for differences in eye investment (by SVL) across mating habitat: Kruskal-Wallis test
kruskal.test(pglsres.edsvl ~ Larval_habitat, data = lar.df2) 

# Test for differences in relative corneal size (compared to eye size) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edcd ~ Larval_habitat, data = lar.df2) 
```

#### Sexual dichromatism

```{r trait-stats-sexdichrom}

#Sexual dichromatism

# Dataframe with just sex dichromatism, eye size, residuals, with "Unknown" observations removed
dic.df <- all.res.traits %>% 
  select(mod_tiplabel, Sex_dichromatism, eye_av.y, pglsres.edmass, pglsres.edsvl, pglsres.edcd) %>% 
  rename(eye_av = eye_av.y) %>%
  mutate(Sex_dichromatism = as.factor(Sex_dichromatism)) %>%
  filter(Sex_dichromatism != "Unknown")

#Summary stats
dic.stats <- ddply(dic.df, ~Sex_dichromatism, summarise,
                   mean.eye=mean(eye_av, na.rm = TRUE), 
                   sd.eye=sd(eye_av, na.rm = TRUE), 
                   n.eye=length(eye_av[!is.na(eye_av)]), 
                   mean.edmass=mean(pglsres.edmass, na.rm = TRUE),
                   sd.edmass=sd(pglsres.edmass, na.rm = TRUE),
                   n.edmass=length(pglsres.edmass[!is.na(pglsres.edmass)]), 
                   mean.edsvl=mean(pglsres.edsvl, na.rm = TRUE),
                   sd.edsvl=sd(pglsres.edsvl, na.rm = TRUE),
                   n.edsvl=length(pglsres.edsvl[!is.na(pglsres.edsvl)]),
                   mean.edcd=mean(pglsres.edcd, na.rm = TRUE),
                   sd.pglsres.edcd=sd(pglsres.edcd, na.rm = TRUE),
                   n.edcd=length(pglsres.edcd[!is.na(pglsres.edcd)]))


print(dic.stats)

# Test for differences in mean (absolute) eye size across sex dichromatism habitat groups: Kruskal-Wallis test
kruskal.test(eye_av ~ Sex_dichromatism, data = dic.df) 

# Test for differences in eye investment (by mass) across mating habitat: Kruskal-Wallis test
kruskal.test(pglsres.edmass ~ Sex_dichromatism, data = dic.df)  

# Test for differences in eye investment (by SVL) across mating habitat: Kruskal-Wallis test
kruskal.test(pglsres.edsvl ~ Sex_dichromatism, data = dic.df) 

# Test for differences in relative corneal size (compared to eye size) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edcd ~ Sex_dichromatism, data = dic.df)  
```

```{r dichrom-subset, include = FALSE}

#remove adult habitat groups that never exhibit sexual dichromatism (aquatic and fossorial) and see if the pattern of increased eye investment in sexually dichromatic species holds. 

#Subset sexual dichromatism data (exclude fossorial and aquatic species)
dich.sub <- all.res.traits %>% 
  filter(Adult_habitat != "Fossorial", Adult_habitat != "Aquatic") %>%
  select(mod_tiplabel, Sex_dichromatism, eye_av.y, pglsres.edmass, pglsres.edsvl, pglsres.edcd) %>% 
  rename(eye_av = eye_av.y) %>%
  mutate(Sex_dichromatism = as.factor(Sex_dichromatism)) %>%
  filter(Sex_dichromatism != "Unknown")

#Summary stats (ground dwelling only)
mat.stats.ground <- ddply(mat.ground, ~Mating_habitat, summarise,
                   mean.eye=mean(eye_av, na.rm = TRUE), 
                   sd.eye=sd(eye_av, na.rm = TRUE), 
                   n.eye=length(eye_av[!is.na(eye_av)]), 
                   mean.edmass=mean(pglsres.edmass, na.rm = TRUE),
                   sd.edmass=sd(pglsres.edmass, na.rm = TRUE),
                   n.edmass=length(pglsres.edmass[!is.na(pglsres.edmass)]), 
                   mean.edsvl=mean(pglsres.edsvl, na.rm = TRUE),
                   sd.edsvl=sd(pglsres.edsvl, na.rm = TRUE),
                   n.edsvl=length(pglsres.edsvl[!is.na(pglsres.edsvl)]),
                   mean.edcd=mean(pglsres.edcd, na.rm = TRUE),
                   sd.pglsres.edcd=sd(pglsres.edcd, na.rm = TRUE),
                   n.edcd=length(pglsres.edcd[!is.na(pglsres.edcd)]))

print(mat.stats.ground)

# Test for differences in eye size) across states: Kruskal-Wallis test
kruskal.test(eye_av ~ Sex_dichromatism, data = dich.sub) 

# Test for differences in eye investment (by mass) across states: Kruskal-Wallis test
kruskal.test(pglsres.edmass ~ Sex_dichromatism, data = dich.sub) 

# Test for differences in eye investment (by svl) across states: Kruskal-Wallis test
kruskal.test(pglsres.edsvl ~ Sex_dichromatism, data = dich.sub) 

```


## Testing for ecological correlates of eye size across the phylogeny

### PGLS of eye scaling with body mass with ecological factors as covariates

#### Adult habitat as covariate

```{r pgls-habitat}

# Check for best model of correlation structure ----

#remove rows from averaged dataframe not in phylogeny
av.edmass.hab <- av.edmass %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Adult_habitat)
rownames(av.edmass.hab) <- av.edmass.hab$mod_tiplabel

#Make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(frog.tree$tip.label, as.character(av.edmass.hab$mod_tiplabel))

#Drop unwanted tips from phylogeny
tree.edit <- drop.tip(phy = frog.tree, tip = drops) 

#Reorder species data to match phylogeny order
dat <-ReorderData(tree.edit, av.edmass.hab, taxa.names="row names")

#check that tree and data match
name.check(tree.edit, dat)

#test whether BM or OU model is better fit for PGLS using the nlme package and AIC comparison

#Expected covariance under a pure Brownian model (Felsenstein 1985,	Martins and Hansen 1997)
pglsBM <- gls(log10(eye_av) ~ log10(rootmass_av) * Adult_habitat, cor = corBrownian(phy = tree.edit), data = dat, method = "ML")

#Martins and Hansen's (1997) covariance structure (OU)
pglsOU <- gls(log10(eye_av) ~ log10(rootmass_av) * Adult_habitat, cor = corMartins(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

#Pagel's “lambda” Correlation Structure: The correlation structure from the present model is derived from the Brownian motion model by multiplying the off-diagonal elements (i.e., the covariances) by lambda. The variances are thus the same than for a Brownian motion model.
pglsPagel<-gls(log10(eye_av) ~ log10(rootmass_av) * Adult_habitat, cor = corPagel(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

anova(pglsBM, pglsOU, pglsPagel)

# Run ML lambda model in caper ----

#remove rows from averaged dataframe not in phylogeny
av.edmass.hab <- av.edmass %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Adult_habitat)
rownames(av.edmass.hab) <- av.edmass.hab$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edmass.hab)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edmass.hab.comp <- comparative.data(phy = frog.tree, data = av.edmass.hab, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edmass.hab.comp$dropped$tips #phylogeny
edmass.hab.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of mass using the Maximum Liklihood estimate of lambda
pgls_hab <- pgls(log10(eye_av) ~ log10(rootmass_av) * Adult_habitat, 
               data = edmass.hab.comp, 
               lambda = "ML", bounds = list(lambda = c(1e-05, 1)),
               #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_hab) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_hab), digits = 3, caption= "ANOVA Table for eye size ~ mass * habitat") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#intercept and slope
summary(pgls_hab)

#plot for data and PGLS fit for eye diameter vs. cube root of mass
plot_pgls.edmass.hab <- ggplot(av.edmass.hab, aes(x = rootmass_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Adult_habitat)) +
  scale_color_manual(values = col_hab) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "eye diameter (mm)") +
  scale_x_log10(name = "cube root of mass (g)") +
  geom_abline(slope = coef(pgls_hab)[[2]], intercept = coef(pgls_hab)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edmass.hab)

```

#### Activity pattern as covariate

```{r pgls-activity}

# Check for best model of correlation structure ----

#remove rows from averaged dataframe not in phylogeny
av.edmass.act <- av.edmass %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Activity_period)
rownames(av.edmass.act) <- av.edmass.act$mod_tiplabel

#Make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(frog.tree$tip.label, as.character(av.edmass.act$mod_tiplabel))

#Drop unwanted tips from phylogeny
tree.edit <- drop.tip(phy = frog.tree, tip = drops) 

#Reorder species data to match phylogeny order
dat <-ReorderData(tree.edit, av.edmass.act, taxa.names="row names")

#check that tree and data match
name.check(tree.edit, dat)

#test whether BM or OU model is better fit for PGLS using the nlme package and AIC comparison

#Expected covariance under a pure Brownian model (Felsenstein 1985,	Martins and Hansen 1997)
pglsBM <- gls(log10(eye_av) ~ log10(rootmass_av) * Activity_period, cor = corBrownian(phy = tree.edit), data = dat, method = "ML")

#Martins and Hansen's (1997) covariance structure (OU)
pglsOU <- gls(log10(eye_av) ~ log10(rootmass_av) * Activity_period, cor = corMartins(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

#Pagel's “lambda” Correlation Structure: The correlation structure from the present model is derived from the Brownian motion model by multiplying the off-diagonal elements (i.e., the covariances) by lambda. The variances are thus the same than for a Brownian motion model.
pglsPagel<-gls(log10(eye_av) ~ log10(rootmass_av) * Activity_period, cor = corPagel(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

anova(pglsBM, pglsOU, pglsPagel)

# Run ML lambda model in caper ----

#remove rows from averaged dataframe not in phylogeny
av.edmass.act <- av.edmass %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Activity_period)
rownames(av.edmass.act) <- av.edmass.act$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edmass.act)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edmass.act.comp <- comparative.data(phy = frog.tree, data = av.edmass.act, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edmass.act.comp$dropped$tips #phylogeny
edmass.act.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of mass using the Maximum Liklihood estimate of lambda
pgls_act <- pgls(log10(eye_av) ~ Activity_period * log10(rootmass_av) , 
               data = edmass.act.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_act) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(pgls_act)

#significance of relationships
kable(anova(pgls_act), digits = 3, caption= "ANOVA Table for eye size ~ mass * habitat") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

  
#plot for data and PGLS fit for eye diameter vs. cube root of mass
plot_pgls.edmass.act <- ggplot(av.edmass.act, aes(x = rootmass_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Activity_period)) +
  scale_color_manual(values = col_act) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "eye diameter (mm)") +
  scale_x_log10(name = "cube root of mass (g)") +
  geom_abline(slope = coef(pgls_act)[[2]], intercept = coef(pgls_act)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edmass.act)

```

#### Mating habitat as covariate

```{r pgls-mating}

# Check for best model of correlation structure ----

#remove rows from averaged dataframe not in phylogeny
av.edmass.mat <- av.edmass %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Mating_habitat)
rownames(av.edmass.mat) <- av.edmass.mat$mod_tiplabel

#Make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(frog.tree$tip.label, as.character(av.edmass.mat$mod_tiplabel))

#Drop unwanted tips from phylogeny
tree.edit <- drop.tip(phy = frog.tree, tip = drops) 

#Reorder species data to match phylogeny order
dat <-ReorderData(tree.edit, av.edmass.mat, taxa.names="row names")

#check that tree and data match
name.check(tree.edit, dat)

#test whether BM or OU model is better fit for PGLS using the nlme package and AIC comparison

#Expected covariance under a pure Brownian model (Felsenstein 1985,	Martins and Hansen 1997)
pglsBM <- gls(log10(eye_av) ~ log10(rootmass_av) * Mating_habitat, cor = corBrownian(phy = tree.edit), data = dat, method = "ML")

#Martins and Hansen's (1997) covariance structure (OU)
pglsOU <- gls(log10(eye_av) ~ log10(rootmass_av) * Mating_habitat, cor = corMartins(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

#Pagel's “lambda” Correlation Structure: The correlation structure from the present model is derived from the Brownian motion model by multiplying the off-diagonal elements (i.e., the covariances) by lambda. The variances are thus the same than for a Brownian motion model.
pglsPagel<-gls(log10(eye_av) ~ log10(rootmass_av) * Mating_habitat, cor = corPagel(1, phy = tree.edit, fixed = F), data = dat, method = "ML")

anova(pglsBM, pglsOU, pglsPagel)

# Run ML lambda model in caper ----

#remove rows from averaged dataframe not in phylogeny
av.edmass.mat <- av.edmass %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Mating_habitat)
rownames(av.edmass.mat) <- av.edmass.mat$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edmass.mat)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edmass.mat.comp <- comparative.data(phy = frog.tree, data = av.edmass.mat, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edmass.mat.comp$dropped$tips #phylogeny
edmass.mat.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of mass using the Maximum Liklihood estimate of lambda
pgls_mat <- pgls(log10(eye_av) ~ log10(rootmass_av) * Mating_habitat, 
               data = edmass.mat.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_mat) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_mat), digits = 3, caption= "ANOVA Table for eye size ~ mass * mating hab") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#intercept and slope
summary(pgls_mat)

#plot for data and PGLS fit for eye diameter vs. cube root of mass
plot_pgls.edmass.mat <- ggplot(av.edmass.mat, aes(x = rootmass_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Mating_habitat)) +
  scale_color_manual(values = col_mat) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "eye diameter (mm)") +
  scale_x_log10(name = "cube root of mass (g)") +
  geom_abline(slope = coef(pgls_mat)[[2]], intercept = coef(pgls_mat)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edmass.mat)

```

#### Life history strategy as covariate

```{r pgls-lifehist}

#remove rows from averaged dataframe not in phylogeny
av.edmass.lh <- av.edmass %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Life_history)
rownames(av.edmass.lh) <- av.edmass.lh$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edmass.lh)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edmass.lh.comp <- comparative.data(phy = frog.tree, data = av.edmass.lh, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edmass.lh.comp$dropped$tips #phylogeny
edmass.lh.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of mass using the Maximum Liklihood estimate of lambda
pgls_lh <- pgls(log10(eye_av) ~ log10(rootmass_av) * Life_history, 
               data = edmass.lh.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_lh) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_lh), digits = 3, caption= "ANOVA Table for eye size ~ mass * life history") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#intercept and slope
summary(pgls_lh)

#plot for data and PGLS fit for eye diameter vs. cube root of mass
plot_pgls.edmass.lh <- ggplot(av.edmass.lh, aes(x = rootmass_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Life_history)) +
  scale_color_manual(values = col_lif) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "eye diameter (mm)") +
  scale_x_log10(name = "cube root of mass (g)") +
  geom_abline(slope = coef(pgls_lh)[[2]], intercept = coef(pgls_lh)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edmass.lh)

```

#### Larval habitat as covariate

```{r pgls-larvhab}

#remove rows from averaged dataframe not in phylogeny
av.edmass.larhab <- av.edmass %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Larval_habitat)
rownames(av.edmass.larhab) <- av.edmass.larhab$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edmass.larhab)
  
#use caper function to comsbine phylogeny and data into one object (this function also matches species names in tree and dataset)
edmass.larhab.comp <- comparative.data(phy = frog.tree, data = av.edmass.larhab, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edmass.larhab.comp$dropped$tips #phylogeny
edmass.larhab.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of mass using the Maximum Liklihood estimate of lambda
pgls_larhab <- pgls(log10(eye_av) ~ log10(rootmass_av) * Larval_habitat, 
               data = edmass.larhab.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_larhab) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_larhab), digits = 3, caption= "ANOVA Table for eye size ~ mass * sex dichromatism") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#model summary
summary(pgls_larhab)

#plot for data and PGLS fit for eye diameter vs. cube root of mass
plot_pgls.edmass.larhab <- ggplot(av.edmass.larhab, aes(x = rootmass_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Larval_habitat)) +
  scale_color_manual(values = col_larv) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "eye diameter (mm)") +
  scale_x_log10(name = "cube root of mass (g)") +
  geom_abline(slope = coef(pgls_larhab)[[2]], intercept = coef(pgls_larhab)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edmass.larhab)

```

#### Sexual dichromatism as covariate

```{r pgls-sexdich}

#remove rows from averaged dataframe not in phylogeny
av.edmass.dich <- av.edmass %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Sex_dichromatism)
rownames(av.edmass.dich) <- av.edmass.dich$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edmass.dich)
  
#use caper function to comsbine phylogeny and data into one object (this function also matches species names in tree and dataset)
edmass.dich.comp <- comparative.data(phy = frog.tree, data = av.edmass.dich, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edmass.dich.comp$dropped$tips #phylogeny
edmass.dich.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of mass using the Maximum Liklihood estimate of lambda
pgls_dich <- pgls(log10(eye_av) ~ log10(rootmass_av) * Sex_dichromatism, 
               data = edmass.dich.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_dich) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_dich), digits = 3, caption= "ANOVA Table for eye size ~ mass * sex dichromatism") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#model summary
summary(pgls_dich)

#plot for data and PGLS fit for eye diameter vs. cube root of mass
plot_pgls.edmass.dich <- ggplot(av.edmass.dich, aes(x = rootmass_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Sex_dichromatism)) +
  scale_color_manual(values = col_dich) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "eye diameter (mm)") +
  scale_x_log10(name = "cube root of mass (g)") +
  geom_abline(slope = coef(pgls_dich)[[2]], intercept = coef(pgls_dich)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edmass.dich)

```

### PGLS of eye scaling with ecological factors with SVL as covariate

#### Adult habitat as covariate

```{r pgls-habitat-svl}

#remove rows from averaged dataframe not in phylogeny
av.edsvl.hab <- av.edsvl %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Adult_habitat)
rownames(av.edsvl.hab) <- av.edsvl.hab$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edsvl.hab)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edsvl.hab.comp <- comparative.data(phy = frog.tree, data = av.edsvl.hab, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edsvl.hab.comp$dropped$tips #phylogeny
edsvl.hab.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_hab.svl <- pgls(log10(eye_av) ~ log10(svl_av) * Adult_habitat, 
               data = edsvl.hab.comp, 
               lambda = "ML", bounds = list(lambda = c(1e-05, 1)),
               #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_hab.svl) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_hab.svl), digits = 3, caption= "ANOVA Table for eye size ~ svl * habitat") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#intercept and slope
summary(pgls_hab.svl)

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edsvl.hab <- ggplot(av.edsvl.hab, aes(x = svl_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Adult_habitat)) +
  scale_color_manual(values = col_hab) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Eye diameter (mm)") +
  scale_x_log10(name = "SVL (mm)") +
  geom_abline(slope = coef(pgls_hab.svl)[[2]], intercept = coef(pgls_hab.svl)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edsvl.hab)

```

#### Activity pattern as covariate

```{r pgls-activity-svl}

#remove rows from averaged dataframe not in phylogeny
av.edsvl.act <- av.edsvl %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Activity_period)
rownames(av.edsvl.act) <- av.edsvl.act$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edsvl.act)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edsvl.act.comp <- comparative.data(phy = frog.tree, data = av.edsvl.act, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edsvl.act.comp$dropped$tips #phylogeny
edsvl.act.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_act.svl <- pgls(log10(eye_av) ~ Activity_period * log10(svl_av) , 
               data = edsvl.act.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_act.svl) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(pgls_act.svl)

#significance of relationships
kable(anova(pgls_act.svl), digits = 3, caption= "ANOVA Table for eye size ~ svl * activity period") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edsvl.act <- ggplot(av.edsvl.act, aes(x = svl_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Activity_period)) +
  scale_color_manual(values = col_act) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Eye diameter (mm)") +
  scale_x_log10(name = "SVL (mm)") +
  geom_abline(slope = coef(pgls_act.svl)[[4]], intercept = coef(pgls_act.svl)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edsvl.act)

```

#### Mating habitat as covariate

```{r pgls-mating-svl}

#remove rows from averaged dataframe not in phylogeny
av.edsvl.mat <- av.edsvl %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Mating_habitat)
rownames(av.edsvl.mat) <- av.edsvl.mat$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edsvl.mat)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edsvl.mat.comp <- comparative.data(phy = frog.tree, data = av.edsvl.mat, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edsvl.mat.comp$dropped$tips #phylogeny
edsvl.mat.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_mat.svl <- pgls(log10(eye_av) ~ log10(svl_av) * Mating_habitat, 
               data = edsvl.mat.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_mat.svl) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_mat.svl), digits = 3, caption= "ANOVA Table for eye size ~ svl * mating hab") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#intercept and slope
summary(pgls_mat.svl)

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edsvl.mat <- ggplot(av.edsvl.mat, aes(x = svl_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Mating_habitat)) +
  scale_color_manual(values = col_mat) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Eye diameter (mm)") +
  scale_x_log10(name = "SVL (mm)") +
  geom_abline(slope = coef(pgls_mat.svl)[[2]], intercept = coef(pgls_mat.svl)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edsvl.mat)

```

#### Life history strategy as covariate

```{r pgls-lifehist-svl}

#remove rows from averaged dataframe not in phylogeny
av.edsvl.lh <- av.edsvl %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Life_history)
rownames(av.edsvl.lh) <- av.edsvl.lh$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edsvl.lh)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edsvl.lh.comp <- comparative.data(phy = frog.tree, data = av.edsvl.lh, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edsvl.lh.comp$dropped$tips #phylogeny
edsvl.lh.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_lh.svl <- pgls(log10(eye_av) ~ log10(svl_av) * Life_history, 
               data = edsvl.lh.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_lh.svl) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_lh.svl), digits = 3, caption= "ANOVA Table for eye size ~ svl * life history") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#intercept and slope
summary(pgls_lh.svl)

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edsvl.lh <- ggplot(av.edsvl.lh, aes(x = svl_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Life_history)) +
  scale_color_manual(values = col_lif) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "eye diameter (mm)") +
  scale_x_log10(name = "cube root of svl (g)") +
  geom_abline(slope = coef(pgls_lh.svl)[[2]], intercept = coef(pgls_lh.svl)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edsvl.lh)

```

#### Larval habitat as covariate

```{r pgls-larvha-svl}

#remove rows from averaged dataframe not in phylogeny
av.edsvl.larhab <- av.edsvl %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Larval_habitat)
rownames(av.edsvl.larhab) <- av.edsvl.larhab$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edsvl.larhab)
  
#use caper function to comsbine phylogeny and data into one object (this function also matches species names in tree and dataset)
edsvl.larhab.comp <- comparative.data(phy = frog.tree, data = av.edsvl.larhab, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edsvl.larhab.comp$dropped$tips #phylogeny
edsvl.larhab.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_larhab.svl <- pgls(log10(eye_av) ~ log10(svl_av) * Larval_habitat, 
               data = edsvl.larhab.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_larhab.svl) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_larhab.svl), digits = 3, caption= "ANOVA Table for eye size ~ svl * sex dichromatism") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#model summary
summary(pgls_larhab.svl)

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edsvl.larhab <- ggplot(av.edsvl.larhab, aes(x = svl_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Larval_habitat)) +
  scale_color_manual(values = col_larv) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Eye diameter (mm)") +
  scale_x_log10(name = "SVL (mm") +
  geom_abline(slope = coef(pgls_larhab.svl)[[2]], intercept = coef(pgls_larhab.svl)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edsvl.larhab)

```

#### Sexual dichromatism as covariate

```{r pgls-sexdich-svl}

#remove rows from averaged dataframe not in phylogeny
av.edsvl.dich <- av.edsvl %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Sex_dichromatism)
rownames(av.edsvl.dich) <- av.edsvl.dich$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edsvl.dich)
  
#use caper function to comsbine phylogeny and data into one object (this function also matches species names in tree and dataset)
edsvl.dich.comp <- comparative.data(phy = frog.tree, data = av.edsvl.dich, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edsvl.dich.comp$dropped$tips #phylogeny
edsvl.dich.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_dich.svl <- pgls(log10(eye_av) ~ log10(svl_av) * Sex_dichromatism, 
               data = edsvl.dich.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_dich.svl) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_dich.svl), digits = 3, caption= "ANOVA Table for eye size ~ svl * sex dichromatism") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#model summary
summary(pgls_dich.svl)

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edsvl.dich <- ggplot(av.edsvl.dich, aes(x = svl_av, y = eye_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Sex_dichromatism)) +
  scale_color_manual(values = col_dich) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Eye diameter (mm)") +
  scale_x_log10(name = "SVL (mm") +
  geom_abline(slope = coef(pgls_dich.svl)[[2]], intercept = coef(pgls_dich.svl)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edsvl.dich)

```

### PGLS of cornea scaling with eye diameter with ecological factors as covariates

#### Adult habitat as covariate

```{r pgls-habitat-cor}

#remove rows from averaged dataframe not in phylogeny
av.edcd.hab <- av.edcd %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Adult_habitat)
rownames(av.edcd.hab) <- av.edcd.hab$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edcd.hab)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edcd.hab.comp <- comparative.data(phy = frog.tree, data = av.edcd.hab, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edcd.hab.comp$dropped$tips #phylogeny
edcd.hab.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_hab.cor <- pgls(log10(cornea_av) ~ log10(eye_av) * Adult_habitat, 
               data = edcd.hab.comp, 
               lambda = "ML", bounds = list(lambda = c(1e-05, 1)),
               #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_hab.cor) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_hab.cor), digits = 3, caption= "ANOVA Table for cornea size ~ eye size * habitat") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#intercept and slope
summary(pgls_hab.cor)

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edcd.hab <- ggplot(av.edcd.hab, aes(x = eye_av, y = cornea_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Adult_habitat)) +
  scale_color_manual(values = col_hab) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Cornea diameter (mm)") +
  scale_x_log10(name = "Eye diameter (mm)") +
  geom_abline(slope = coef(pgls_hab.cor)[[2]], intercept = coef(pgls_hab.cor)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edcd.hab)

```

#### Activity pattern as covariate

```{r pgls-activity-cor}

#remove rows from averaged dataframe not in phylogeny
av.edcd.act <- av.edcd %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Activity_period)
rownames(av.edcd.act) <- av.edcd.act$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edcd.act)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edcd.act.comp <- comparative.data(phy = frog.tree, data = av.edcd.act, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edcd.act.comp$dropped$tips #phylogeny
edcd.act.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_act.cor <- pgls(log10(cornea_av) ~ Activity_period * log10(eye_av) , 
               data = edcd.act.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_act.cor) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#model summary
summary(pgls_act.cor)

#significance of relationships
kable(anova(pgls_act.cor), digits = 3, caption= "ANOVA Table for cornea size ~ eye size * habitat") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edcd.act <- ggplot(av.edcd.act, aes(x = eye_av, y = cornea_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Activity_period)) +
  scale_color_manual(values = col_act) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Cornea diameter (mm)") +
  scale_x_log10(name = "Eye diameter (mm)") +
  geom_abline(slope = coef(pgls_act.cor)[[4]], intercept = coef(pgls_act.cor)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edcd.act)

```

#### Mating habitat as covariate

```{r pgls-mating-cor}

#remove rows from averaged dataframe not in phylogeny
av.edcd.mat <- av.edcd %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Mating_habitat)
rownames(av.edcd.mat) <- av.edcd.mat$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edcd.mat)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edcd.mat.comp <- comparative.data(phy = frog.tree, data = av.edcd.mat, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edcd.mat.comp$dropped$tips #phylogeny
edcd.mat.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_mat.cor <- pgls(log10(cornea_av) ~ log10(eye_av) * Mating_habitat, 
               data = edcd.mat.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_mat.cor) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_mat.cor), digits = 3, caption= "ANOVA Table for cornea size ~ eye size * mating hab") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#intercept and slope
summary(pgls_mat.cor)

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edcd.mat <- ggplot(av.edcd.mat, aes(x = eye_av, y = cornea_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Mating_habitat)) +
  scale_color_manual(values = col_mat) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Cornea diameter (mm)") +
  scale_x_log10(name = "Eye diameter (mm)") +
  geom_abline(slope = coef(pgls_mat.cor)[[2]], intercept = coef(pgls_mat.cor)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edcd.mat)

```

#### Life history strategy as covariate

```{r pgls-lifehist-cor}

#remove rows from averaged dataframe not in phylogeny
av.edcd.lh <- av.edcd %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Life_history)
rownames(av.edcd.lh) <- av.edcd.lh$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edcd.lh)
  
#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
edcd.lh.comp <- comparative.data(phy = frog.tree, data = av.edcd.lh, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edcd.lh.comp$dropped$tips #phylogeny
edcd.lh.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_lh.cor <- pgls(log10(cornea_av) ~ log10(eye_av) * Life_history, 
               data = edcd.lh.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_lh.cor) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_lh.cor), digits = 3, caption= "ANOVA Table for cornea size ~ eye size * life history") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#intercept and slope
summary(pgls_lh.cor)

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edcd.lh <- ggplot(av.edcd.lh, aes(x = eye_av, y = cornea_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Life_history)) +
  scale_color_manual(values = col_lif) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Cornea diameter (mm)") +
  scale_x_log10(name = "cube root of svl (g)") +
  geom_abline(slope = coef(pgls_lh.cor)[[2]], intercept = coef(pgls_lh.cor)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edcd.lh)

```

####Larval habitat as covariate

```{r pgls-larvhab-cor}

#remove rows from averaged dataframe not in phylogeny
av.edcd.larhab <- av.edcd %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Larval_habitat)
rownames(av.edcd.larhab) <- av.edcd.larhab$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edcd.larhab)
  
#use caper function to comsbine phylogeny and data into one object (this function also matches species names in tree and dataset)
edcd.larhab.comp <- comparative.data(phy = frog.tree, data = av.edcd.larhab, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edcd.larhab.comp$dropped$tips #phylogeny
edcd.larhab.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_larhab.cor <- pgls(log10(cornea_av) ~ log10(eye_av) * Larval_habitat, 
               data = edcd.larhab.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_larhab.cor) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_larhab.cor), digits = 3, caption= "ANOVA Table for cornea size ~ eye size * sex dichromatism") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#model summary
summary(pgls_larhab.cor)

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edcd.larhab <- ggplot(av.edcd.larhab, aes(x = eye_av, y = cornea_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Larval_habitat)) +
  scale_color_manual(values = col_larv) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Cornea diameter (mm)") +
  scale_x_log10(name = "Eye diameter (mm") +
  geom_abline(slope = coef(pgls_larhab.cor)[[2]], intercept = coef(pgls_larhab.cor)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edcd.larhab)

```

#### Sexual dichromatism as covariate

```{r pgls-sexdich-cor}

#remove rows from averaged dataframe not in phylogeny
av.edcd.dich <- av.edcd %>%
  drop_na(mod_tiplabel) %>%
  drop_na(Sex_dichromatism)
rownames(av.edcd.dich) <- av.edcd.dich$mod_tiplabel

#check that names match in dataframe and tree
name.check(phy = frog.tree, data = av.edcd.dich)
  
#use caper function to comsbine phylogeny and data into one object (this function also matches species names in tree and dataset)
edcd.dich.comp <- comparative.data(phy = frog.tree, data = av.edcd.dich, names.col = mod_tiplabel, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
edcd.dich.comp$dropped$tips #phylogeny
edcd.dich.comp$dropped$unmatched.rows #dataset

#run PGLS model for eye diameter vs. cube root of svl using the Maximum Liklihood estimate of lambda
pgls_dich.cor <- pgls(log10(cornea_av) ~ log10(eye_av) * Sex_dichromatism, 
               data = edcd.dich.comp, 
               lambda = "ML", #uses Maximum Liklihood estimate of lambda
               param.CI = 0.95)

#evaluate model assumptions
par(mfrow = c(2,2)) #makes your plot window into 2x2 panels
plot(pgls_dich.cor) #plot the linear model
par(mfrow = c(1,1)) #set plotting window back to one plot

#significance of relationships
kable(anova(pgls_dich.cor), digits = 3, caption= "ANOVA Table for cornea size ~ eye size * sex dichromatism") %>%
  kable_styling(full_width = F) %>%
  collapse_rows(columns = 1, valign = "top")

#model summary
summary(pgls_dich.cor)

#plot for data and PGLS fit for eye diameter vs. cube root of svl
plot_pgls.edcd.dich <- ggplot(av.edcd.dich, aes(x = eye_av, y = cornea_av, text = genus_species)) +
 geom_point(alpha = 0.6, aes(color = Sex_dichromatism)) +
  scale_color_manual(values = col_dich) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "Cornea diameter (mm)") +
  scale_x_log10(name = "Eye diameter (mm") +
  geom_abline(slope = coef(pgls_dich.cor)[[2]], intercept = coef(pgls_dich.cor)[[1]], linetype = "solid") #plots PGLS fit with solid line

#interactive plot
ggplotly(plot_pgls.edcd.dich)

```

***

# Comparison with a previous study

## Cornea scaling in frogs compared to Huang et al. (2019)

We compared our results for cornea scaling from this study to a recently published study by Huang et al. (2019). Huang et al. measured corneal diameter (from photographs) and both mass and svl. 

First we imported Huang et al.'s data and back-calculated their raw corneal diameter measurements from their published, calculated "eye volume" data and an equation provided in their methods. We also subset our own corneal diameter data from both museum specimens and from freshly caught field specimens to compare. Finally, to examine whether differential familial sampling could explain differences in our results vs. Huang et al's results, we subset our museum specimen data to exclude families not sampled by Huang. 

```{r frog-comp-data}

# Scaling of eyes with mass and svl in frogs (this study vs. Huang et al. 2019)

#import frog data from Huang et al. 2019
huang <- data.frame(read.csv("../Data/other verts/anurans.csv", header=TRUE, na.strings=c("", "NA", " "))) 

#Huang et al. 2019 calculated eye volume based on corneal radius via the equation: eye size = 2*1.33*pi*b^3 ,where b is corneal radius. Thus, b = cube root of (.5eye/1.33pi), and corneal diameter is twice that. 
huang.sub <- huang %>%
  mutate(rootmass = (Mass_g)^(1/3)) %>%
  mutate(cornea_mm = (2*(((Eye_vol_mm3/2)/(1.33*pi))^(1/3)))) %>%
  rename(Name = Species, Mass = rootmass, Cornea = cornea_mm) %>%
  mutate(Group = as.factor("Huang et al. 2019")) %>%
  select(Family, Name, Mass, SVL_mm, Cornea, Group)

#subset preserved frog data and add group name
frogs.corneas <- frogs %>%
  rename(Name = genus_species, Mass = rootmass, Cornea = cormean) %>%
  mutate(Group = as.factor("Museum specimens")) %>%
  select(Family, Name, Mass, SVL_mm, Cornea, Group)

#subset fresh frog data and add group name
frogs.fresh <- fresh.frogs %>%
  mutate(rootmass = (Mass_g)^(1/3)) %>%
  rename(Name = Genus, Mass = rootmass, Cornea = CD_mm_dissected) %>%
  mutate(Group = as.factor("Fresh specimens")) %>%
  select(Family, Name, Mass, SVL_mm, Cornea, Group)

#create a subset of our museum data where the families sampled are limited to the families sampled by Huang et al. (2019)
frogs.famsub <- frogs.corneas[frogs.corneas$Family %in% huang.sub$Family, ] %>% #subset our frogs based on family match in Huang data
   mutate(Group = as.factor("Museum (Huang families)"))

#join datasets together
frogcomp <- rbind(huang.sub, frogs.corneas, frogs.fresh, frogs.famsub)
```

Then we used an SMA test in smatr to determine whether corneal scaling with body size was the same for Huang's data, our museum specimen data, our fresh specimen data, and our subset of museum specimen data that excluded families not sampled by Huang et al. 

```{r frog-comp-smas}

# SMA test whether there are any differences in slope or intercept for our preserved specimens, fresh specimens, or subset of families matching Huang et al. sampling for corneal diameter vs. the cube root of mass
sma_frpr.mass <- sma(formula = Cornea ~ Mass + Group, #test for slope AND elevation diff.
               robust = TRUE,
               data = filter(frogcomp, Group != "Huang et al. 2019"),
               log = "xy",
               method = "SMA", 
               alpha = 0.05)

summary(sma_frpr.mass)
plot(sma_frpr.mass)

#SMA test whether there are any differences in slope or intercept for our preserved specimens, fresh specimens, subset of families matching Huang et al. sampling, or Huang et al.'s data for corneal diameter vs. the cube root of mass
sma_frogcomp.mass <- sma(formula = Cornea ~ Mass + Group, #test for slope diff
               robust = TRUE,
               data = frogcomp,
               log = "xy",
               method = "SMA", 
               multcomp = TRUE,
               multcompmethod = "adjusted",
               alpha = 0.05)

plot(sma_frogcomp.mass)
summary(sma_frogcomp.mass)
multcompmatrix(sma_frogcomp.mass)


# SMA test whether there are any differences in slope or intercept for our preserved specimens, fresh specimens, or subset of families matching Huang et al. sampling for corneal diameter vs. SVL
sma_frpr.svl <- sma(formula = Cornea ~ SVL_mm + Group, #test for slope AND elevation diff.
               robust = TRUE,
               data = filter(frogcomp, Group != "Huang et al. 2019"),
               log = "xy",
               method = "SMA", 
               alpha = 0.05)

summary(sma_frpr.svl)
plot(sma_frpr.svl)

#SMA test whether there are any differences in slope or intercept for our preserved specimens, fresh specimens, subset of families matching Huang et al. sampling, or Huang et al.'s data for corneal diameter vs. SVL
sma_frogcomp.svl <- sma(formula = Cornea ~ SVL_mm + Group, #test for slope diff
               robust = TRUE,
               data = frogcomp,
               log = "xy",
               method = "SMA", 
               multcomp = TRUE,
               multcompmethod = "adjusted",
               alpha = 0.05)

plot(sma_frogcomp.svl)
summary(sma_frogcomp.svl)
multcompmatrix(sma_frogcomp.svl)
```

Finally, we created plots showing the data and SMA fits. 

```{r frog-comp-plots}
#make color vector for groups
col_frogcomp <- c("Huang et al. 2019" = "#F28E2B",
               "Museum specimens" = "#59A14F",
               "Fresh specimens" = "#B07AA1", 
               "Museum (Huang families)" = "#39737c")

sh_frogcomp <- c("Huang et al. 2019" = 22,
               "Museum specimens" = 21,
               "Fresh specimens" = 21, 
               "Museum (Huang families)" = 21)

#plot for cornea diameter v. cube root mass
plot_frogcomp.mass <- ggplot(frogcomp, aes(x = Mass, y = Cornea, text = Name)) +
  geom_point(aes(color = Group), alpha = 0.5) +
  scale_color_manual(values = col_frogcomp, 
                     name = "Frog dataset",
                     breaks = c("Museum specimens",
                                "Museum (Huang families)",
                                "Fresh specimens",
                                "Huang et al. 2019"),
                    labels = c("Museum - all",
                                "Museum - subset",
                                "Fresh",
                                "Huang et al.")) +
  scale_shape_manual(values = sh_frogcomp, 
                     name = "Frog dataset") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "cornea diameter (mm)") +
  scale_x_log10(name = "cube root of mass (g)") +
  geom_abline(slope = coef(sma_frogcomp.mass)[[2]], intercept = coef(sma_frogcomp.mass)[[1]], linetype = "dashed", alpha = 1, color = col_frogcomp)
              
ggplotly(plot_frogcomp.mass)

#plot for cornea diameter v. svl
plot_frogcomp.svl <- ggplot(frogcomp, aes(x = SVL_mm, y = Cornea, text = Name)) +
  geom_point(aes(color = Group), alpha = 0.5) +
  scale_color_manual(values = col_frogcomp, 
                     name = "Frog dataset",
                     breaks = c("Museum specimens",
                                "Museum (Huang families)",
                                "Fresh specimens",
                                "Huang et al. 2019"),
                    labels = c("Museum - all",
                                "Museum - subset",
                                "Fresh",
                                "Huang et al.")) +
  scale_shape_manual(values = sh_frogcomp, 
                     name = "Frog dataset") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  scale_y_log10(name = "cornea diameter (mm)") +
  scale_x_log10(name = "snout-vent length (mm)") +
  geom_abline(slope = coef(sma_frogcomp.svl)[[2]], intercept = coef(sma_frogcomp.svl)[[1]], linetype = "dashed", alpha = 1, color = col_frogcomp)
              
ggplotly(plot_frogcomp.svl)

```

Then exported our plots to a pdf figure. 

```{r frogcomp-figure, fig.width = 10, fig.height = 10}

#make panels 
fig.a <- plot_frogcomp.mass +
  theme(legend.position = "none", text = element_text(size=12))

fig.b <- plot_frogcomp.svl +
  theme(legend.position = "none", text = element_text(size=12))

#use cowplot package to nicely plot the graphs together with a shared common legend.
plots <- plot_grid(fig.a, fig.b, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("A", "B"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 2) #number of rows in grids

# extract legends from figures
leg <- get_legend(plot_frogcomp.svl + theme(legend.position = "right"))

#construct full figure with panels and legend
plot_grid(plots, leg, ncol = 2, rel_widths = c(1, 0.5), align = 'vh')

#export pdf of figure
pdf("../Manuscript/Figures/figure-frogcomp.pdf", width=8, height=8)
plot_grid(plots, leg, ncol = 2, rel_widths = c(1, 0.3), align = 'vh')
dev.off()
```

## Adult habitat comparison based on Huang categories

```{r trait-stats-huanghab}

# Dataframe with just habitat, eye size, residuals, with "Unknown" observations removed
hab.huang <- all.res.traits %>% 
  select(mod_tiplabel, Adult_habitat, eye_av.y, pglsres.edmass, pglsres.edsvl, pglsres.edcd) %>% 
  rename(eye_av = eye_av.y) %>%
  mutate(Adult_habitat = as.factor(Adult_habitat)) %>%
  filter(Adult_habitat != "Unknown") %>%
  mutate(huang_habs = recode(Adult_habitat, 
         "Aquatic" = "A-S",
         "Semiaquatic"= "A-S",
         "Fossorial" = "T-A", 
         "Ground-dwelling" = "T-A",
         "Scansorial" = "T-A",
         "Subfossorial" = "T-A"))

#Summary stats
hh.stats <- ddply(hab.huang, ~huang_habs, summarise, 
                   mean.eye=mean(eye_av, na.rm = TRUE), 
                   sd.eye=sd(eye_av, na.rm = TRUE), 
                   n.eye=length(eye_av[!is.na(eye_av)]), 
                   mean.edmass=mean(pglsres.edmass, na.rm = TRUE),
                   sd.edmass=sd(pglsres.edmass, na.rm = TRUE),
                   n.edmass=length(pglsres.edmass[!is.na(pglsres.edmass)]), 
                   mean.edsvl=mean(pglsres.edsvl, na.rm = TRUE),
                   sd.edsvl=sd(pglsres.edsvl, na.rm = TRUE),
                   n.edsvl=length(pglsres.edsvl[!is.na(pglsres.edsvl)]),
                   mean.edcd=mean(pglsres.edcd, na.rm = TRUE),
                   sd.pglsres.edcd=sd(pglsres.edcd, na.rm = TRUE),
                   n.edcd=length(pglsres.edcd[!is.na(pglsres.edcd)]))

print(hh.stats)

# Test for differences in mean (absolute) eye size across habitat groups: Kruskal-Wallis test
kruskal.test(eye_av ~ huang_habs, data = hab.huang) 

# Test for differences relative eye size (by mass) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edmass ~ huang_habs, data = hab.huang) 
# Test for differences in eye investment (by svl) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edsvl ~ huang_habs, data = hab.huang) 

# Test for differences in relative corneal size (compared to eye size) across habitat groups: Kruskal-Wallis test
kruskal.test(pglsres.edcd ~ huang_habs, data = hab.huang) 

# Make boxplots of trait distributions. Categories by huang's habitat classifications but colored by ours. 

#boxplot of eye size by adult habitat
hh.abs <- ggplot(data = filter(hab.huang, !is.na(eye_av)), aes(x = huang_habs, y = eye_av)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Adult_habitat), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_hab) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
    scale_x_discrete(breaks=c("A-S","T-A"),labels=c("Aq/S", "T/Ar")) +
  ylab("Eye diameter (mm)") +
  xlab("Binary habitat") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. cube root of mass PGLS) by huang habitat categories
hh.res.mass <- ggplot(data = filter(hab.huang, !is.na(pglsres.edmass)), aes(x = huang_habs, y = pglsres.edmass)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Adult_habitat), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_hab, name = "Adult habitat") +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_x_discrete(breaks=c("A-S","T-A"),
                   labels=c("Aq/S", "T/Ar")) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2),
                     labels = c("0.25x", "0.40x", "0.6x", "1x", "1.6x"), limits = c(-0.62, 0.22)) +
  ylab("Eye investment by mass") +
  xlab("Binary habitat") +
  theme(legend.position = "none")

#boxplot of eye investment (residuals of eye diameter v. SVL PGLS) by adult habitat
hh.res.svl <-  ggplot(data = filter(hab.huang, !is.na(pglsres.edsvl)), aes(x = huang_habs, y = pglsres.edsvl)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point", shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Adult_habitat), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_hab, name = "Adult habitat") +
    theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_x_discrete(breaks=c("A-S","T-A"),
                   labels=c("Aq/S", "T/Ar")) +
  scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0.0, 0.2), labels = c("0.25x", "0.40x", "0.6x", "1x", "1.6x"), limits = c(-0.62, 0.22)) +
  ylab("Eye investment by SVL") +
  xlab("Binary habitat") +
  theme(legend.position = "none")

#make figure for the supplemental information
fig.a <- hh.abs + 
  theme(legend.position = "none", axis.title=element_text(size=12))

fig.b <- hh.res.mass +
  theme(legend.position = "none", axis.title=element_text(size=12)) 
  
fig.c <- hh.res.svl +
  theme(legend.position = "none", axis.title=element_text(size=12)) 
  
#use cowplot package to nicely plot the graphs together with a shared common legend.
plots <- plot_grid(fig.a, fig.b, fig.c, #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           labels = c("A", "B", "C"), #panel labels for figure
           hjust = -1, #adjustment for panel labels
           nrow = 1) #number of rows in grids

# extract legends from figures
leg <- get_legend(hh.res.svl + theme_bw() + theme(legend.position = "right"))

#export pdf of figure
pdf(file = "../Manuscript/Figures/supp-huanghabs.pdf", height=3, width=10)
plot_grid(plots, leg, ncol = 2 ,rel_widths = c(1, 0.2))
dev.off()
```